---
title: "Alu"
output: html_notebook
---


```{r}

# small bam file for sample
#chr6:107,211,482-107,264,849
echo "chr6  107211482  107264849" ch6_sample.bed
samtools view filtered/WTCHG_538916_221156.bam -L ch6_sample.bed -b -o ch6_sample.bam
samtools index ch6_sample.bam



# Repeats

curl 'http://genome.ucsc.edu/cgi-bin/hgTables?hgsid=738896965_zTgYMjAIHdEHcvRZcCteap7Zqbpt&boolshad.hgta_printCustomTrackHeaders=0&hgta_ctName=tb_rmsk&hgta_ctDesc=table+browser+query+on+rmsk&hgta_ctVis=pack&hgta_ctUrl=&fbQual=whole&fbUpBases=200&fbDownBases=200&hgta_doGetBed=get+BED' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:67.0) Gecko/20100101 Firefox/67.0' -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' -H 'Accept-Language: en-GB,en;q=0.5' --compressed -H 'Connection: keep-alive' -H 'Referer: http://genome.ucsc.edu/cgi-bin/hgTables' -H 'Cookie: hguid.genome-euro=362527140_Mi0jIpneXLkgeplCtz5aa6JO13ha; hguid=673678055_VvqPPBb5iFgOJEF3WMG7M3t7iual' -H 'Upgrade-Insecure-Requests: 1' > Repeat_Masker.bed.gz
gunzip -d Repeat_Masker.bed.gz

grep -P 'chr[0-9XY]+\t' Repeat_Masker.bed | sort -k1,1 -k2,2n  > Repeat_MaskerSorted.bed

# split into Alu and not
grep 'Alu[a-zA-Z]' Repeat_MaskerSorted.bed > Repeat_MaskerSorted_Alu.bed
grep -v 'Alu[a-zA-Z]' Repeat_MaskerSorted.bed > Repeat_MaskerSorted_NotAlu.bed

# check Alu with no family name not present
grep -P 'Alu\t' Repeat_MaskerSorted_Alu.bed | head

awk '$3-$2 > 250 && $3-$2 < 350' Repeat_MaskerSorted_Alu.bed > Repeat_MaskerSorted_Alu_QC.bed
```

Over half of all ZcwST peaks overlap Alu, overrepresented in low enrichment peaks. L1 also shows a pattern of being depleted at high enrichment peaks. Other repeats don't overlap a large fraction of preaks.

```{r}
repeats <- fread("~/Dropbox/Downloads/Repeat_MaskerSorted.bed")

setnames(repeats, c("chr","CI_start","CI_stop","Name","Score","Strand"))

repeats$id <- seq_along(repeats$chr)

repeats <- prejoin(repeats, ext = 0)

# check order preserverd
stopifnot(sum(diff(repeats$id)<0)==0)
stopifnot(sum(diff(repeats$id)!=1)==0)

head(repeats[,.N,by=Name][order(-N)], 50)

fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[Name!="Alu"][grep("Alu",Name)][CI_stop-CI_start>250 & CI_stop-CI_start<350], x1="Zcw UT", y1="Alu Repeats", shift=1e8) # yes up to 70%
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("L1",Name)], x1="Zcw UT", y1="L1 Repeats", shift=1e8) # depeated
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("LTR",Name)], x1="Zcw UT", y1="LTR Repeats", shift=1e8) # not

fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("MIR",Name)], x1="Zcw UT", y1="MIR Repeats", shift=1e8) # not but higher in high enriched 12%
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("MIR",Name)], x1="Zcw UT", y1="MIR Repeats", shift=1e8) + scale_y_continuous()
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("L2",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not, 12%
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("\\(",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not 12%
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("\\(",Name)], x1="Zcw UT", y1="SimpleRepeats", shift=1e8) + scale_y_continuous()# 2% higher in low enriched
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("MER",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not 10%
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("MLT",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not 5%
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("FLAM",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not <5%
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("THE",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not <5%
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("L3",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not <5%
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("rich",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("Charlie",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("Tigger",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("MST",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("Mam",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("HERV",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("ERVL",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not
# fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[Name=="THE1B"], x1="Zcw UT", y1="Repeats", shift=1e8) # not
```

Non Alu repeats are not enriched relative to expected based on randomised positions.

```{r}

# Summary of those not enriched
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("\\(|MIR|L2|THE1B|MER|MLT|FLAM|THE|L3|rich|Charlie|Tigger|LTR|MST|Mam|HERV|ERVL",Name)], x1="Zcw UT", y1="Non Alu Repeats", shift=1e8)

# Summary of what's left
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[!grep("\\(|Alu|MIR|L2|L1|THE1B|MER|MLT|FLAM|THE|L3|rich|Charlie|Tigger|LTR|MST|Mam|HERV|ERVL",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # not




fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluJ",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # yes up to 0.2
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluS",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # yes up to 0.55
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluY",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # a bit
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[Name=="Alu"], x1="Zcw UT", y1="Repeats", shift=1e8) # no

fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluSx",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # yes up to 0.25
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluSz",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # yes up to 0.15
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluSq",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # yes
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluSp",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # yes
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluSc",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # yes
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluSg",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) # yes

fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluSx",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) + scale_y_continuous()
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluSz",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) + scale_y_continuous()
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeats[grep("AluSg",Name)], x1="Zcw UT", y1="Repeats", shift=1e8) + scale_y_continuous()
```

## Fraction of Alu

75% of all Alu repeats have a ZcwST peak, higher for those with better match to consensus.

```{r}

repeats$enrichment <- as.numeric(repeats$Score)
fraction_overlapN2(repeats[Name!="Alu"][grep("Alu",Name)][CI_stop-CI_start>250 & CI_stop-CI_start<350], zcw_peaks_cin, x1="Alu Repeats by SmithWaterman Score (Not ", y1="Zcw UT", shift=1e8) # 

# Only ~ 35% of ==Alu overlap Zcw
# fraction_overlapN2(repeats[Name=="Alu"], zcw_peaks_cin, x1="Zcw UT", y1="Repeats", shift=1e8)
```

## Prdm9 Peaks

As expected Prdm9 is enriched for THE1B - but only N term not C term. Neither set of Prdm9 peaks not enriched for Alu repeats.

```{r}
# P9 enriched in THE1B
fraction_overlapN2(HP9N_peaks[cov_input>=5], repeats[Name=="THE1B"], x1="HP9N", y1="THE1B", shift=1e8) + scale_y_continuous()
# not C term
fraction_overlapN2(HP9combo_peaks[cov_input>=5], repeats[Name=="THE1B"], x1="HP9 Cterm", y1="THE1B", shift=1e8) + scale_y_continuous()
# P9 Not enriched for Alu
fraction_overlapN2(HP9N_peaks[cov_input>=5], repeats[grep("Alu",Name)], x1="HP9N", y1="Alu Repeats", shift=1e8)
fraction_overlapN2(HP9combo_peaks[cov_input>=5], repeats[Name!="Alu"][grep("Alu",Name)], x1="HP9 Cterm HA + V5", y1="Alu Repeats", shift=1e8)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=5], repeats[Name!="Alu"][grep("Alu",Name)], x1="Zcw CVC", y1="Alu Repeats", shift=1e8)

# total = 5,607,738
# left = 230,099
# Alu = 1,209,364
repeats[!grep("\\(|Alu|MIR|L2|L1|THE1B|MER|MLT|FLAM|THE|L3|rich|Charlie|Tigger|LTR|MST|Mam|HERV|ERVL",Name)][,.N,by=Name][order(-N)][1:50]

repeats[grep("Alu",Name)][,.N,by=Name][order(-N)][1:50]

```

# CpG

```{r}
system("bedtools getfasta -s -fi motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa -bed Repeat_MaskerSorted_Alu_QC.bed -name > AluRepeats.fa")

AluRepeats <- seqinr::read.fasta("~/Downloads/AluRepeats.fa", forceDNAtolower = FALSE, as.string = TRUE)
AluRepeats <- unlist(AluRepeats)

repeatsAlu <- repeats[Name!="Alu"][grep("Alu",Name)][CI_stop-CI_start>250 & CI_stop-CI_start<350]
# check names match for ordering
all.equal(repeatsAlu[,paste0(Name,"(",Strand,")")], names(AluRepeats))
repeatsAlu$length <- nchar(AluRepeats)

CG_locations <- sapply(AluRepeats,function(x) stringr::str_locate_all(x,"cg")[[1]][,1])
repeatsAlu$CG_count <- sapply(CG_locations, length)



# location & frequency hist
hist(unlist(CG_locations[repeatsAlu$length==299]), breaks=1000)
hist(CG_count, breaks=50)

# % of Alu
repeatsAlu$enrichment <- repeatsAlu$CG_count
p <- fraction_overlapN2(repeatsAlu, zcw_peaks_cin, x1="Alu Repeats by CpG Count (Not ", y1="Zcw UT", shift=1e9) + 
  scale_x_continuous() +
  xlim(0,25) + # min of 200 data points required
  xlab("CpG Count") + 
  ylab("Fraction of Alu Repeats overlaping Zcwpw1 peaks (no Prdm9)") +
  theme(legend.position = "bottom")


###### % of Zcw

tmp <- overlapGam(zcw_peaks_cin[cov_input>=5], repeats[Name!="Alu"][grep("Alu",Name)][CI_stop-CI_start>250 & CI_stop-CI_start<350])
tmpR <- overlapGam(zcw_peaks_cin[cov_input>=5], repeats[Name!="Alu"][grep("Alu",Name)][CI_stop-CI_start>250 & CI_stop-CI_start<350], shift = 1e8)

tmp2 <- fraction_overlapN2(zcw_peaks_cin[cov_input>=5], 
                           repeats[Name!="Alu"][grep("Alu",Name)][CI_stop-CI_start>250 & CI_stop-CI_start<350],
                           x1="Zcw UT", y1="Alu Repeats", shift=1e8) # yes up to 70%

pi <- tmp2 + 
  geom_ribbon(data=tmp, aes(ymin = Overlap - 2*se, ymax = Overlap + 2*se), fill = "#E41A1C", colour="#E41A1C", alpha=0.5) +
  geom_ribbon(data=tmpR, aes(ymin = Overlap - 2*se, ymax = Overlap + 2*se), fill = "#377EB8", colour="#377EB8", alpha=0.5) +
  scale_x_log10(limits=c(0.3,5)) +
  theme(legend.position = "bottom") +
  xlab("Zcwpw1 Enrichment (no Prdm9)") +
  ylab("Fraction of Zcwpw1 peaks that overlap Alu repeats")

pmain <- plot_grid(pi + theme(legend.position = "none"), p + theme(legend.position = "none"), labels = "AUTO")

pdf("../results/Alu_by_CpG.pdf", width = 10, height = 5.5)
plot_grid(pmain, get_legend(p), ncol=1, rel_heights = c(1, 0.07))
dev.off()

# % of Zcw
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeatsAlu[enrichment==0], x1="Zcw UT", y1="Alu Repeats CpG=0", shift=1e8) + scale_y_continuous()
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeatsAlu[enrichment==1], x1="Zcw UT", y1="Alu Repeats CpG=1", shift=1e8) + scale_y_continuous()
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeatsAlu[enrichment==5], x1="Zcw UT", y1="Alu Repeats CpG=5", shift=1e8) + scale_y_continuous()
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeatsAlu[enrichment==10], x1="Zcw UT", y1="Alu Repeats CpG=10", shift=1e8) + scale_y_continuous()
fraction_overlapN2(zcw_peaks_cin[cov_input>=5], repeatsAlu[enrichment==20], x1="Zcw UT", y1="Alu Repeats CpG=20", shift=1e8) + scale_y_continuous()
```

```{r}
repeatsAlu$id <- seq_along(repeatsAlu$chr)

overlapingAluIDS <- foverlaps(zcw_peaks_cin, repeatsAlu, mult="first")$id
overlapingAluIDS2 <- overlapingAluIDS[!is.na(overlapingAluIDS)]

repeatsAlu$overlap <- FALSE
repeatsAlu[id %in% overlapingAluIDS2, overlap := TRUE]

# compare count of CpG
repeatsAlu$enrichment <- repeatsAlu$CG_count
ggplot(repeatsAlu, aes(enrichment, fill=overlap)) + geom_histogram(aes(y=..density..), position = "identity", alpha=0.7, bins=70) + scale_fill_brewer(palette = "Set1")
ggplot(repeatsAlu, aes(length, colour=overlap)) + geom_density()

# compare location of CpG
hist(unlist(CG_locations[repeatsAlu[overlap==TRUE]$id]), breaks=1000)
hist(unlist(CG_locations[repeatsAlu[overlap==FALSE]$id]), breaks=1000)

# compare proportion of Alu families
AluOverapProp <- merge(repeatsAlu[overlap==TRUE][,.(Overlap=.N/nrow(repeatsAlu[overlap==TRUE]), Number=.N),by=Name],
                       repeatsAlu[overlap==FALSE][,.(NonOverlap=.N/nrow(repeatsAlu[overlap==FALSE]), Number=.N),by=Name], by="Name")
AluOverapProp[, ratioOfProportions := (Overlap+0.01)/(NonOverlap+0.01)]
AluOverapProp[, Occurrences := Number.y + Number.x]

ggplot(AluOverapProp, aes(Name, ratioOfProportions, size=Occurrences)) +
  geom_hline(yintercept = 1, col="darkgrey") +
  geom_point() +
  coord_flip() +
  scale_size() +
  scale_y_log10()


# compare proportion of GC Counts
AluOverapProp <- merge(repeatsAlu[overlap==TRUE][,.(Overlap=.N/nrow(repeatsAlu[overlap==TRUE]), Number=.N),by=CG_count],
                       repeatsAlu[overlap==FALSE][,.(NonOverlap=.N/nrow(repeatsAlu[overlap==FALSE]), Number=.N),by=CG_count], by="CG_count")
AluOverapProp[, ratioOfProportions := (Overlap)/(NonOverlap)]
AluOverapProp[, Occurrences := Number.y + Number.x]

ggplot(AluOverapProp, aes(CG_count, ratioOfProportions, size=Occurrences)) +
  geom_hline(yintercept = 1, col="darkgrey") +
  geom_point() +
  scale_size() +
  scale_y_log10()

# AluOverapProp$Type <- reorder(AluOverapProp$Name, AluOverapProp$ratioOfProportions)
# ggplot(AluOverapProp, aes(Type, ratioOfProportions, size=count)) + geom_hline(yintercept = 1, col="darkgrey") + geom_point() + coord_flip() + scale_size() + scale_y_log10()


# p <- ggplot(AluOverapProp, aes(Overlap, NonOverlap, label=Name, colour=log(ratioOfProportions))) + 
#   #stat_smooth(method="lm", formula = y ~ 0 + x, se = F, col="grey", size=1) +
#   geom_abline() +
#   geom_point() + 
#   scale_x_log10() + scale_y_log10()
# 
# p
# p + scale_x_continuous() + scale_y_continuous() + geom_label_repel(data=AluOverapProp[ratioOfProportions<0.9 | ratioOfProportions>1.5])
# ggplot(AluOverapProp, aes(V1.x*V1.y, V1.x/V1.y, label=Name, colour=ratioOfProportions)) + geom_point() + geom_label_repel(data=AluOverapProp[ratioOfProportions<0.7 | ratioOfProportions>1.5]) + scale_x_log10() + scale_y_log10()
```

Cross check results
```{r}
repeatsAlu$enrichment <- repeatsAlu$Score
fraction_overlapN2(repeatsAlu[grep("AluJ",Name)], zcw_peaks_cin, x1="AluJ Repeats by SmithWaterman Score (Not ", y1="Zcw UT", shift=1e8) + scale_x_continuous()
fraction_overlapN2(repeatsAlu[grep("AluSz",Name)], zcw_peaks_cin, x1="AluSz Repeats by SmithWaterman Score (Not ", y1="Zcw UT", shift=1e8) + scale_x_continuous() 
fraction_overlapN2(repeatsAlu[grep("AluS[cqpgx]",Name)], zcw_peaks_cin, x1="AluS[cqpgx] Repeats by SmithWaterman Score (Not ", y1="Zcw UT", shift=1e8) + scale_x_continuous()
fraction_overlapN2(repeatsAlu[grep("AluY",Name)], zcw_peaks_cin, x1="AluY Repeats by SmithWaterman Score (Not ", y1="Zcw UT", shift=1e8) + scale_x_continuous()
```

```{r}
repeatsAlu[grep("AluJ",Name), class := "AluJ"]
repeatsAlu[grep("AluSz",Name), class := "AluSz"]
repeatsAlu[grep("AluS[qpgx]",Name), class := "AluS[qpgx]"]
repeatsAlu[grep("AluSc",Name), class := "AluSc"]
repeatsAlu[grep("AluY",Name), class := "AluY"]

ggplot(repeatsAlu, aes(Score, color=class)) + geom_density()

ggplot(repeatsAlu, aes(CG_count)) + geom_histogram(binwidth = 1) + facet_wrap(~class, scales='free_y')
```

# Aligned Alu CpG

multi-align AluJo sequences with each other (as variable poly A in center messes things up otherwise - see histogram above). The compare aligned positions for diff in frequency of CpG

## Using ensembl kalign

```{r}

overlap_idx <- which(repeatsAlu$Name=="AluJo" & repeatsAlu$overlap==T)[1:1000]
nonoverlap_idx <- which(repeatsAlu$Name=="AluJo" & repeatsAlu$overlap==F)[1:1000]
tmp <- AluRepeats[c(overlap_idx, nonoverlap_idx)]
names(tmp) <- paste0(seq_along(tmp),"_", names(tmp))
export_FASTA(tmp, "AluJo.fasta")

AluAligned <- seqinr::read.fasta("../data/kalign-I20190715-002019-0175-34958881-p2m.fasta", forceDNAtolower = FALSE, as.string = TRUE)
AluAligned <- unlist(AluAligned)

AluAlignedM <- matrix(ncol=nchar(AluAligned[1]), unlist(strsplit(AluAligned,"")), byrow=T)

nongaps <- apply(AluAlignedM, 2, function(x) sum(x=="-"))<200
AluAlignedM <- AluAlignedM[,nongaps]

# c_o <- apply(AluAlignedM[1:1000,], 2, function(x) mean(x=="c"))
# c_no <- apply(AluAlignedM[1001:2000,], 2, function(x) mean(x=="c"))
# 
# plot(log(c_o/c_no))

CG_locations_aligned <- sapply(AluAligned,function(x) stringr::str_locate_all(x,"c(-+)?g")[[1]][,1])

hist(unlist(CG_locations_aligned[1:1000]), breaks=5000, xlim=c(0,max(unlist(CG_locations_aligned))), ylim=c(0,175))
hist(unlist(CG_locations_aligned[1001:2000]), breaks=5000, xlim=c(0,max(unlist(CG_locations_aligned))), ylim=c(0,175))

o_v_no <- rbind(data.table(table(unlist(CG_locations_aligned[1:1000])), type="overlaping"),
                data.table(table(unlist(CG_locations_aligned[1001:2000])), type="nonOverlaping"))

o_v_no_cast <- dcast(o_v_no, V1 ~ type, value.var = "N", fill = 0)
o_v_no_cast[, OvsNOratio := (overlaping+30) / (nonOverlaping+30)]

ggplot(o_v_no_cast, aes(as.numeric(V1), OvsNOratio)) + geom_point() + labs(x="Aligned Position in Alu repeat", y="Ratio of CpG counts in Zcw overlaping Alus vs non-overlaping")

o_v_no_cast[order(-OvsNOratio)][1:20]
o_v_no_cast[order(OvsNOratio)][1:20]
```

## Using R msa

```{r}

tmp <- toupper(AluRepeats[c(overlap_idx[1:1000], nonoverlap_idx[1:1000])])
names(tmp) <- paste0(seq_along(tmp),"_", names(tmp))

AluOverlapAlign <- msa(tmp, type="dna", order="input")
AluOverlapAlignChar <- as.character(AluOverlapAlign)

CG_locations_aligned2 <- sapply(AluOverlapAlignChar,function(x) stringr::str_locate_all(x,"C(-+)?G")[[1]][,1])

hist(unlist(CG_locations_aligned2[1:1000]), breaks=5000, xlim=c(0,max(unlist(CG_locations_aligned2))), ylim=c(0,225))
hist(unlist(CG_locations_aligned2[1:2000]), breaks=5000, xlim=c(0,max(unlist(CG_locations_aligned2))), ylim=c(0,225))

o_v_no <- rbind(data.table(table(unlist(CG_locations_aligned2[1:1000])), type="overlaping"),
                data.table(table(unlist(CG_locations_aligned2[1001:2000])), type="nonOverlaping"))

o_v_no_cast <- dcast(o_v_no, V1 ~ type, value.var = "N", fill = 0)
o_v_no_cast[, OvsNOratio := (overlaping+30) / (nonOverlaping+30)]

ggplot(o_v_no_cast, aes(as.numeric(V1), OvsNOratio)) + geom_point() + labs(x="Aligned Position in Alu repeat", y="Ratio of CpG counts in Zcw overlaping Alus vs non-overlaping")

o_v_no_cast[order(-OvsNOratio)][1:20]
o_v_no_cast[order(OvsNOratio)][1:20]
```

## Try to align everything to reference.

```{r}
library(Biostrings)

AluJo <- "GGCCGGGCGCGGTGGCTCACGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGCGGGAGGATCGCTTGAGCCCAGGAGTTCGAGACCAGCCTGGGCAACATAGCGAGACCCCGTCTCTACAAAAAATACAAAAATTAGCCGGGCGTGGTGGCGCGCGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCGGGAGGATCGCTTGAGCCCAGGAGNTCGAGGCTGCAGTGAGCTATGATCGCGCCACTGCACTCCAGCCTGGGCGACAGAGCGAGACCCTGTCTCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"

sapply( seq2, function(x) pairwiseAlignment(toupper(c2s(x)), seq1string)) )

tmp <- pairwiseAlignment(toupper(AluRepeats[overlap_idx][1:3]), AluJo, type="overlap")
toString(subject(tmp))
toString(pattern(tmp))
nchar(strsplit(toString(pattern(tmp)), ", ")[[1]])
nchar(strsplit(toString(subject(tmp)), ", ")[[1]])
```

# CpG in general

```{r}

# # all CpG in Chr1
# head -1 ../single-cell/sequencing/metadata/hg38_sizes.bed |
# bedtools getfasta -fi motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa -bed stdin |
# ./seqkit locate --ignore-case --non-greedy --only-positive-strand -p CG --bed |
# sed s/:0-248956422//g > CpG_chr1.bed

# All CpG positions in genome
./seqkit locate --ignore-case --non-greedy --only-positive-strand -p CG --bed motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa > CpG_hg38.bed

cut -f 1-3 CpG_hg38.bed > CpG_hg38_cut.bed

# count CpG at each peak (Chr1)
tail -n +2 peaks/SingleBasePeaks.WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL.bed |
bedtools slop -i stdin -g ../single-cell/sequencing/metadata/hg38_sizes.chrom -b 150 | grep -P "chr1\t" |
bedtools intersect -a stdin -b CpG_chr1.bed -c > peaks/SingleBasePeaks.WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL_wCpGcount_chr1.bed

# count CpG at each peak Whole genome
tail -n +2 peaks/SingleBasePeaks.WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL.bed |
bedtools slop -i stdin -g ../single-cell/sequencing/metadata/hg38_sizes.chrom -b 150 |
bedtools intersect -a stdin -b CpG_hg38.bed -c > peaks/SingleBasePeaks.WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL_wCpGcount.bed


# 115320 peaks, each 300bp
# 2316765 CpG positions
# chr1 248956422
# 2316765 / (248956422/300) = 2.79 CpG per 300bp
# but clustering
# try randomisation instead
head -1 ../single-cell/sequencing/metadata/hg38_sizes.chrom > chr1_size.bed
bedtools random -n 115320 -l 300 -g chr1_size.bed -seed 71346 | sort -k1,1 -k2,2n > chr1_random.bed

bedtools intersect -a chr1_random.bed -b CpG_chr1.bed -c > chr1_random_CpGcount.bed

# whole genome
head -23 ../single-cell/sequencing/metadata/hg38_sizes.chrom > hg38_sizes23.bed
bedtools random -n 1264519 -l 300 -g hg38_sizes23.bed -seed 71346 | sort -k1,1 -k2,2n > peaks/Zcw_random.bed

bedtools intersect -a peaks/Zcw_random.bed -b CpG_hg38.bed -c > Zcw_random_CpGcount.bed



chr1_rand_cpg <- fread("../data/peaks/Zcw_random_CpGcount.bed") #"~/Downloads/chr1_random_CpGcount.bed")
# mean(chr1_rand_cpg$V7) = 2.817 - close!
setnames(chr1_rand_cpg, c("chr","CI_start_ext","CI_stop_ext","x","width","strand","CpGcount"))


zcw_peaks_cin_cpg <- fread("../data/peaks/SingleBasePeaks.WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL_wCpGcount.bed")
names(zcw_peaks_cin_cpg)[1:11] <- c("chr", "center_start", "center_stop", "CI_start", "CI_stop", "cov_r1", "cov_r2", "cov_input", "enrichment", "likelihood", "pvalue")
names(zcw_peaks_cin_cpg)[12] <- "CpGcount"
zcw_peaks_cin_cpg <- prejoin(zcw_peaks_cin_cpg, width=0)
cor(zcw_peaks_cin_cpg$enrichment, zcw_peaks_cin_cpg$CpGcount)


zcw_peaks_cin_cpg_alu <- foverlaps(zcw_peaks_cin_cpg, repeatsAlu, mult = "first")
zcw_peaks_cin_cpg_repeats <- foverlaps(zcw_peaks_cin_cpg, repeats, mult = "first")
chr1_rand_cpg_alu <- foverlaps(chr1_rand_cpg, repeatsAlu, mult = "first")

# non Alus
cpg_table <- rbind(data.table(table(chr1_rand_cpg_alu[is.na(Name)]$CpG), type="Randomised Peaks"),
      data.table(table(zcw_peaks_cin_cpg_alu[is.na(Name)]$CpGcount), type="Observed Peaks"))

ggplot(cpg_table, aes(as.numeric(V1), N, fill=type)) +
  geom_col(position="dodge") +
  xlim(-1,20) +
  scale_fill_brewer(palette = "Set1") +
  labs(x="Count of CpGs under peak +/-150", y="Frequency")

# Alus
cpg_table <- rbind(data.table(table(chr1_rand_cpg_alu[!is.na(Name)]$CpG), type="Randomised Peaks"),
      data.table(table(zcw_peaks_cin_cpg_alu[!is.na(Name)]$CpGcount), type="Observed Peaks"))

ggplot(cpg_table, aes(as.numeric(V1), N, fill=type)) +
  geom_col(position="dodge") +
  xlim(-1,20) +
  scale_fill_brewer(palette = "Set1") +
  labs(x="Count of CpGs under peak +/-150", y="Frequency")

###

zcw_peaks_cin_cpg[CpGcount==0][order(-enrichment)][cov_input>5]
cor.test(zcw_peaks_cin_cpg[cov_input>5]$enrichment, zcw_peaks_cin_cpg[cov_input>5]$CpGcount)

plot(zcw_peaks_cin_cpg[cov_input>5]$CpGcount, zcw_peaks_cin_cpg[cov_input>5]$enrichment)

# overall
tmp <- zcw_peaks_cin_cpg[cov_input>5][,.(MeanEnrich=mean(enrichment),.N),by=CpGcount][order(CpGcount)]
qplot(tmp$CpGcount, tmp$MeanEnrich)

#zcw_peaks_cin_cpg_repeats[Repeat == "L1M", .N, by=Name][order(-N)]

# all repeats
zcw_peaks_cin_cpg_repeats[is.na(Name), Repeat := "Non Repeat"]
zcw_peaks_cin_cpg_repeats[!is.na(Name), Repeat := "Other Repeat"]
zcw_peaks_cin_cpg_repeats[grep("^Alu",Name), Repeat := "Alu"]
#zcw_peaks_cin_cpg_repeats[grep("^L1",Name), Repeat := "L1"]

#zcw_peaks_cin_cpg_repeats[grep("^L1M",Name), Repeat := "L1M"], remove for simplicity

zcw_peaks_cin_cpg_repeats[grep("^L1M[1-3]|^L1MA",Name), Repeat := "L1M1-3 or L1MA"]
# zcw_peaks_cin_cpg_repeats[grep("^L1M4",Name), Repeat := "L1M4"]
# zcw_peaks_cin_cpg_repeats[grep("^L1M5",Name), Repeat := "L1M5"]
# zcw_peaks_cin_cpg_repeats[grep("^L1M6",Name), Repeat := "L1M6"]

#zcw_peaks_cin_cpg_repeats[grep("^L1MA",Name), Repeat := "L1MA"] # now included in above or

#zcw_peaks_cin_cpg_repeats[grep("^L1MD",Name), Repeat := "L1MD"]
# zcw_peaks_cin_cpg_repeats[grep("^L1MB",Name), Repeat := "L1MB"]
# zcw_peaks_cin_cpg_repeats[grep("^L1MC",Name), Repeat := "L1MC"]
# zcw_peaks_cin_cpg_repeats[grep("^L1ME",Name), Repeat := "L1ME"]

zcw_peaks_cin_cpg_repeats[grep("^L1P",Name), Repeat := "L1P"]
#zcw_peaks_cin_cpg_repeats[grep("^L1PR",Name), Repeat := "L1PR"] # big change, but don't show to simplify
# zcw_peaks_cin_cpg_repeats[grep("^L1PA",Name), Repeat := "L1PA"]
# zcw_peaks_cin_cpg_repeats[grep("^L1PB",Name), Repeat := "L1PB"]


# zcw_peaks_cin_cpg_repeats[grep("L2",Name), Repeat := "L2"]
# zcw_peaks_cin_cpg_repeats[grep("MIR",Name), Repeat := "MIR"]
# zcw_peaks_cin_cpg_repeats[grep("MER",Name), Repeat := "MER"]
# zcw_peaks_cin_cpg_repeats[grep("LTR",Name), Repeat := "LTR"]
# zcw_peaks_cin_cpg_repeats[grep("THE",Name), Repeat := "THE"]
# zcw_peaks_cin_cpg_repeats[grep("HER",Name), Repeat := "HER"]
# zcw_peaks_cin_cpg_repeats[grep("MST",Name), Repeat := "MST"]




tmp <- zcw_peaks_cin_cpg_repeats[cov_input>5][,.(MeanEnrich=mean(i.enrichment),.N, sd=sd(i.enrichment)),by=c("CpGcount","Repeat")][order(CpGcount)]

p1 <- ggplot(tmp, aes(CpGcount, MeanEnrich, colour=Repeat, size=N)) +
  geom_point() +
  scale_color_brewer(palette = "Set1") +
  xlim(0,20) +
  ylim(0.8, 2.5) +
  geom_line(size=0.5, linetype="dashed") +
  theme_minimal() +
  geom_pointrange(aes(ymin=MeanEnrich - 2*(sd/sqrt(N)), ymax=MeanEnrich + 2*(sd/sqrt(N))), size=0.5) +
  ylab("Mean Enrichment of Zcwpw1 peaks (without Prdm9)") +
  xlab("Count of CpGs under peak +/-150")

pdf("../results/CpG_general.pdf", width=10)
p1
dev.off()

tmp <- cbind(data.table(table(chr1_rand_cpg_alu[!is.na(Name)]$CpG), type="Randomised Peaks"),
      data.table(table(zcw_peaks_cin_cpg_alu[!is.na(Name)]$CpGcount), type="Observed Peaks"))
tmp <- tmp[1:25]
tmp2 <- cbind(data.table(table(chr1_rand_cpg_alu[is.na(Name)]$CpG), type="Randomised Peaks"),
      data.table(table(zcw_peaks_cin_cpg_alu[is.na(Name)]$CpGcount), type="Observed Peaks"))
tmp2 <- tmp2[1:25]
tmp$isAlu <- T
tmp2$isAlu <- F
tmp <- rbind(tmp, tmp2)
setnames(tmp, c("CpGCount","NRand","t1","v1","NObs","t2","isAlu"))

tmp$CpGCount <- as.numeric(tmp$CpGCount)
ggplot(tmp, aes(CpGCount, NObs/NRand, colour=isAlu)) + geom_point()

p2 <- ggplot(tmp, aes(CpGCount, NObs/NRand, colour=isAlu)) + geom_point() + scale_y_log10() + ylab("Observed Peaks : Randomised Peaks") + scale_color_brewer(palette = "Set1") + theme(legend.position = "bottom")

pdf("../results/Cpg_general.pdf")
p1
p2
dev.off()

```


```{r}
# format for IGV
tail -n +2 SingleBasePeaks.WTCHG_538916_223180_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL.bed | cut -f 1-3 > SingleBasePeaks.WTCHG_538916_223180_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL_headless.bed

tail -n +2 ../results/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627146_AND_SRA_Altemose2015_SRR5627147_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL.bed | cut -f 1-3 > ../results/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627146_AND_SRA_Altemose2015_SRR5627147_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL_headless.bed

tail -n +2 ../results/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627148_vs_SRA_Altemose2015_SRR5627142.p0.000001.sep250.ALL.bed | cut -f 1-3 > ../results/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627148_vs_SRA_Altemose2015_SRR5627142.p0.000001.sep250.ALL_headless.bed

tail -n +2 ../results/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627152_AND_SRR5627153_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL.bed | cut -f 1-3 > ../results/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627152_AND_SRR5627153_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL_headless.bed

#http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&hgta_group=hub_213889&hgta_track=hub_213889_Umap_24_quantitative&hgta_table=hub_213889_Umap.24.quantitative&hgta_doSchema=describe+table+schema


plot(density(repeats[Name!="Alu"][grep("Alu",Name)][,CI_stop-CI_start]))
abline(v=250, col="red")
abline(v=350, col="red")



awk '$3-$2 == 299' Repeat_MaskerSorted_Alu.bed  > Repeat_MaskerSorted_Alu_299.bed

bwtool aggregate 300:300:300 Repeat_Masker_Alu_299.bed bedgraphs/depth_WTCHG_538916_221156.bigWig Alu_meta299.txt -fill=0 
bwtool aggregate 300:300:300 Repeat_Masker_Alu_299.bed bedgraphs/depth_WTCHG_538916_217108.bigWig Alu_meta_In299.txt -fill=0 


tmp <- fread("grep -v '#' Alu_meta299.txt")
tmpIn <- fread("grep -v '#' Alu_meta_In299.txt")

a <- fread("MAPeakCaller/Fragment_Position_WTCHG_538916_221156.total")
b <- fread("MAPeakCaller/Fragment_Position_WTCHG_538916_217108.total")

plot(tmp$V2 / (a$V1/1e6), type="l")
lines(tmpIn$V2 / (b$V1/1e6), col="blue")
abline(v=300, col="red")
abline(v=300+150, col="red", lty=2)
abline(v=600, col="red")
dev.off()

plot((tmp$V2 / tmpIn$V2) * (b$V1/a$V1), type="l")
abline(v=300, col="red")
abline(v=300+150, col="red", lty=2)
abline(v=600, col="red")
dev.off()


```




# Remove more alignments from BWA MEM tags
```{r}
sambamba view -t 12 -h -f bam -F "mapping_quality >= 1 and not (unmapped or secondary_alignment) and not ([XA] != null or [SA] != null)" filtered/WTCHG_538916_221156.bam -o WTCHG_538916_221156_SBBF.bam

samtools flagstat -@ 5 WTCHG_538916_221156_SBBF.bam
# 80705408 + 0 in total (QC-passed reads + QC-failed reads)
# 0 + 0 secondary
# 0 + 0 supplementary
# 0 + 0 duplicates
# 80705408 + 0 mapped (100.00% : N/A)
# 80705408 + 0 paired in sequencing
# 40448867 + 0 read1
# 40256541 + 0 read2
# 80705408 + 0 properly paired (100.00% : N/A)
# 80705408 + 0 with itself and mate mapped
# 0 + 0 singletons (0.00% : N/A)
# 0 + 0 with mate mapped to a different chr
# 0 + 0 with mate mapped to a different chr (mapQ>=5)


samtools flagstat -@ 5 filtered/WTCHG_538916_221156.bam
# 88007214 + 0 in total (QC-passed reads + QC-failed reads)
# 0 + 0 secondary
# 0 + 0 supplementary
# 0 + 0 duplicates
# 88007214 + 0 mapped (100.00% : N/A)
# 88007214 + 0 paired in sequencing
# 44003607 + 0 read1
# 44003607 + 0 read2
# 88007214 + 0 properly paired (100.00% : N/A)
# 88007214 + 0 with itself and mate mapped
# 0 + 0 singletons (0.00% : N/A)
# 0 + 0 with mate mapped to a different chr
# 0 + 0 with mate mapped to a different chr (mapQ>=5)
```

# Create mappability track with GEM

(only exists publicly on hg19)

```{r}
# Mappability track from genome browser (>7GB)
https://epgg-test.wustl.edu/d/hg38/hg38.mappability.75.bigwig
    
# convert Hoffman lab mappability track to bed
wget https://www.pmgenomics.ca/hoffmanlab/proj/bismap/trackhub/hg38/k24.Umap.MultiTrackMappability.bw
./bigWigToBedGraph k24.Umap.MultiTrackMappability.bw k24.Umap.MultiTrackMappability.bedGraph
awk '$4 > 0.75' k24.Umap.MultiTrackMappability.bedGraph | LC_ALL=C sort -k1,1 -k2,2n -S5G --parallel=5 | bedtools merge -i stdin > 24.Umap.MultiTrackMappability_Keep.bed

bwtool extract bed ch6_sample.bed k24.Umap.MultiTrackMappability.bw k24.Umap.MultiTrackMappability_bwtool.bed

bedtools merge -d 10 -i 24.Umap.MultiTrackMappability.bed > 24.Umap.MultiTrackMappability_d10.bed


# example - not sure it's correct due to large streaches
wget https://github.com/xuefzhao/Reference.Mappability/raw/master/hg38.50mer/hg38.50mer.Excludable.bed


sed 's/ /\t/g' hg38.50mer.Excludable.bed |
bedtools intersect -v -a  peaks/SingleBasePeaks.WTCHG_538916_223180_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL.bed \
                          -b stdin > SingleBasePeaks.WTCHG_538916_223180_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL_QCMappability50.bed


# Use GEM

wget https://downloads.sourceforge.net/project/gemlibrary/gem-library/Binary%20pre-release%203/GEM-binaries-Linux-x86_64-core_i3-20130406-045632.tbz2

tar -jxvf GEM-binaries-Linux-x86_64-core_i3-20130406-045632.tbz2

export PATH=/homes/wells/saxony/zcwpw1/GEM-binaries-Linux-x86_64-core_i3-20130406-045632/bin/:$PATH

wget https://github.com/xuefzhao/Reference.Mappability/raw/master/Scripts/bedGraphTobed
chmod 777 bedGraphTobed
bedGraphTobed

# must change from py2 to py3 add paren to print statements, and file arg instead of >>


# unzip fasta reference
zcat /homes/wells/saxony/single-cell/sequencing/metadata/GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.fna.gz > /homes/wells/saxony/single-cell/sequencing/metadata/GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.fna

# create index
gem-indexer -i /homes/wells/saxony/single-cell/sequencing/metadata/GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.fna  -o /homes/wells/saxony/single-cell/sequencing/metadata/GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_setGEM

# generate mappability scores
gem-mappability -I /homes/wells/saxony/single-cell/sequencing/metadata/GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_setGEM.gem -l 75 -o hg38.75mer -T 12
gem-2-wig -I /homes/wells/saxony/single-cell/sequencing/metadata/GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_setGEM.gem -i hg38.75mer.mappability -o hg38.75mer

# remove erreneous names in chromosome names
sed "s/ AC//" hg38.75mer.wig > hg38.75mer2.wig

# convert to bed
./wigToBigWig hg38.75mer2.wig /homes/wells/saxony/single-cell/sequencing/metadata/hg38_sizes.chrom hg38.75mer.bw
./bigWigToBedGraph hg38.75mer.bw hg38.75mer.bedGraph
./bedGraphTobed hg38.75mer.bedGraph hg38.75mer.Includable.bed 0.33
../bedGraphTobed hg38.75mer.bedGraph hg38.75mer.Includable.t0.5.bed 0.5

##### NOT EXCLUDABLE BUT INCLUDABLE ######
#find -type f -name '*' -execdir rename -v 's:Excludable:Includable:g' {} \;

# subset smaller files
grep -P 'chr6 ' hg38.75mer.Includable.bed | sed -e 's/ /\t/g' > hg38.75mer.Includable.chr6.bed
grep -P 'chr1 ' hg38.75mer.Includable.bed | sed -e 's/ /\t/g' > hg38.75mer.Includable.chr1.bed
grep -P 'chr1 ' hg38.75mer.Includable.t0.5.bed | sed -e 's/ /\t/g' > hg38.75mer.Includable.t0.5.chr1.bed

head -1 ../single-cell/sequencing/metadata/hg38_sizes.chrom > hg38_chr1.chrom

bedtools complement -i hg38.75mer.Includable.chr1.bed -g ../hg38_chr1.chrom > hg38.75mer.Excludable.chr1.bed
bedtools complement -i hg38.75mer.Includable.t0.5.chr1.bed -g ../hg38_chr1.chrom > hg38.75mer.Excludable.t0.5.chr1.bed

bedtools merge -i hg38.75mer.Includable.chr1.bed -d 100 > hg38.75mer.Includable.chr1.merge100.bed
bedtools merge -i hg38.75mer.Excludable.chr1.bed -d 500 > hg38.75mer.Excludable.chr1.merge500.bed
bedtools merge -i hg38.75mer.Excludable.t0.5.chr1.bed -d 500 > hg38.75mer.Excludable.t0.5.chr1.merge500.bed

rg 'chr6\t' hg38.75mer.bedGraph > hg38.75mer.chr6.bedGraph
rg 'chr1\t' hg38.75mer.bedGraph > hg38.75mer.chr1.bedGraph


```
