---
title: "DeNovo Motif Finding"
output: html_notebook
---


# Find ZCVC motif

```{r}

zcw_CVC_NonPromoters_300w <- load_sequences("motifs/zcw_wP9_vs_chip_Z_NonPromoters_2000w.fasta", flank=250)

motif_found_ZcwCVC <- findamotif(head(zcw_CVC_NonPromoters_300w,2000), len=7, motif_rank=1, seed=138624)




# Where is Zcw motif located?
zcw_CVC_NonPromoters_2000w <- load_sequences("motifs/zcw_wP9_vs_chip_Z_NonPromoters_2000w.fasta", flank=1000, sm.rm=F)
zcw_CVC_NonPromoters_2000w_positions <- getmotifs(motif_found_ZcwCVC$scoremat, nrow(motif_found_ZcwCVC$scoremat), 
                                                  head(zcw_CVC_NonPromoters_2000w,1000), maxwidth = max(nchar(zcw_CVC_NonPromoters_2000w)),
                                                  alpha = 0.6, maxits = 10, updatemot = 0, ourprior = rep(0.1, 10))

# Discovery in broader window

zcw_CVC_NonPromoters_2000w <- load_sequences("motifs/zcw_wP9_vs_chip_Z_NonPromoters_2000w.fasta", flank=500)
motif_found_ZcwCVC_NonPromoters_2000w <- findamotif(head(zcw_CVC_NonPromoters_2000w,1000), len=7, seed=286438, nits=50)

pdf(width=10, height=5, file="motifs/Zcw_denovo.pdf")
ggseqlogo::ggseqlogo(get_PWM(motif_found_ZcwCVC))
ggseqlogo::ggseqlogo(get_PWM(motif_found_ZcwCVC, complement=T))

plot_motif_location(motif_found_ZcwCVC)
plot(motif_found_ZcwCVC$prior)

plot_motif_location(zcw_CVC_NonPromoters_2000w_positions)
plot(zcw_CVC_NonPromoters_2000w_positions$prior)

ggseqlogo::ggseqlogo(get_PWM(motif_found_ZcwCVC_NonPromoters_2000w))
ggseqlogo::ggseqlogo(get_PWM(motif_found_ZcwCVC_NonPromoters_2000w, complement=T))
plot_motif_location(motif_found_ZcwCVC_NonPromoters_2000w)
plot(motif_found_ZcwCVC_NonPromoters_2000w$prior)

dev.off()

export_PWM(motif_found_ZcwCVC_NonPromoters_2000w, file="motif_found_ZcwCVC_NonPromoters_2000w.pwm", name="motif_found_ZcwCVC_NonPromoters_2000w")
```

# Find Zcw &/ wP9 motif

```{r}
zcw_wP9_NonPromoters_300w <- load_sequences("motifs/zcw_wP9_peaks_cin_NonPromoters_300w.fasta")
zcw_NonPromoters_300w <- load_sequences("motifs/zcw_peaks_NonPromoters_300w.fasta")

motif_found_Zcw_wP9 <- findamotif(head(zcw_wP9_NonPromoters_300w,1000), len=6)
motif_found_Zcw <- findamotif(head(zcw_NonPromoters_300w,1000), len=6)

pdf(width=10, height=5, file="motifs/denovo_Zcw_wP9.pdf")
ggseqlogo::ggseqlogo(get_PWM(motif_found_Zcw_wP9))
ggseqlogo::ggseqlogo(get_PWM(motif_found_Zcw_wP9, complement=T))
plot_motif_location(motif_found_Zcw_wP9)
plot(motif_found_Zcw_wP9$prior)

ggseqlogo::ggseqlogo(get_PWM(motif_found_Zcw))
ggseqlogo::ggseqlogo(get_PWM(motif_found_Zcw, complement=T))
plot_motif_location(motif_found_Zcw)
plot(motif_found_Zcw$prior)

dev.off()



```


```{r}
zcw_CVC_NonPromoters_300w <- load_sequences("motifs/zcw_wP9_vs_chip_Z_NonPromoters_2000w.fasta", flank=500)

zcw_CVC_NonPromoters_300w_Prdm9_locations <- getmotifs(znew$mot, znew$sizes, 
                                        head(zcw_CVC_NonPromoters_300w,1000), maxwidth = max(nchar(zcw_CVC_NonPromoters_300w)),
                                        alpha = 0.2, maxits = 10, updatemot = 0, ourprior = znew$prior, updateprior=1)

HP9N_peaks_NonPromoters_300w_ZcwM_locations <- getmotifs(motif_found_ZcwCVC$scoremat, motif_found_ZcwCVC$scorematdim, 
                                        head(HP9N_peaks_NonPromoters_300w,5000), maxwidth = max(nchar(HP9N_peaks_NonPromoters_300w)),
                                        alpha = 0.2, maxits = 50, updatemot = 0, ourprior = motif_found_ZcwCVC$prior, updateprior=1)

pdf(file="motifs/ZCW_at_P9.pdf")
plot_motif_location(zcw_CVC_NonPromoters_300w_Prdm9_locations)
plot(zcw_CVC_NonPromoters_300w_Prdm9_locations$prior)

plot_motif_location(HP9N_peaks_NonPromoters_300w_ZcwM_locations)
plot(HP9N_peaks_NonPromoters_300w_ZcwM_locations$prior)
dev.off()


tmp <- copy(HP9N_peaks_NonPromoters_300w_positions$dt)

tmp <- merge(HP9N_peaks_NonPromoters_300w_ZcwM_locations$dt, tmp, by='sequence', suffixes = c(".Zcw",".P9"), all=T)[!(is.na(whichpos.Zcw) & is.na(whichpos.P9))]

tmp <- data.table(whichmotif.P9=1:7, length=znew$sizes)[tmp,on="whichmotif.P9"]
tmp[,P9.end := whichpos.P9 + length]
tmp[,Zcw.end := whichpos.Zcw + motif_found_ZcwCVC$scorematdim]

motifcenters$whichmotif.P9 = motifcenters$whichmotif

tmp <- motifcenters[tmp, on="whichmotif.P9"]

tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)]
# 3049 P9 motif containing sequences, also have Zcw motif, out of 5k

plot(tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)]$whichpos.P9,
     tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)]$whichpos.Zcw)

cor(tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][whichstrand.P9==whichstrand.Zcw]$whichpos.P9,
     tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][whichstrand.P9==whichstrand.Zcw]$whichpos.Zcw, method="spearman")
# 27%

# calculate relative positioning
tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9) & whichstrand.P9==1 & whichstrand.Zcw==1 ,relative_position := whichpos.P9 + motifcenterZF4T - whichpos.Zcw]
tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9) & whichstrand.P9==0 & whichstrand.Zcw==0 ,relative_position := -1 * (whichpos.P9 + length - motifcenterZF4T - Zcw.end)]

# which is most common relative position
sort(table(tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][whichstrand.P9==1 & whichstrand.Zcw==1]$relative_position))
sort(table(tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][whichstrand.P9==0 & whichstrand.Zcw==0]$relative_position))

sort(table(tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][whichstrand.P9==whichstrand.Zcw]$relative_position))
#-4   22    4   -5   -1    8
#71   72   74   85  216  238

tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][whichstrand.P9==whichstrand.Zcw, mean(relative_position), by=whichmotif.P9]

tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][,.N, by=whichmotif.P9]


pdf(file="motifs/ZCW_at_P9_2.pdf")
ggplot(head(tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][order(-regprob.Zcw-regprob.P9)],250)) +
  geom_segment(aes(x=whichpos.P9, xend=P9.end, y=sequence, yend=sequence), size=1, alpha=0.5, colour="red") +
  geom_segment(aes(x=whichpos.Zcw, xend=Zcw.end, y=sequence, yend=sequence), size=1, alpha=0.5, colour="blue") +
  facet_wrap(~whichstrand.P9 + whichstrand.Zcw) + ggtitle("red = P9, blue = Zcw motif, 0/1 = strand, most confident")

ggplot(head(tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9) & relative_position==8][order(-regprob.Zcw-regprob.P9)],250)) +
  geom_segment(aes(x=whichpos.P9, xend=P9.end, y=sequence, yend=sequence), size=1, alpha=0.5, colour="red") +
  geom_segment(aes(x=whichpos.Zcw, xend=Zcw.end, y=sequence, yend=sequence), size=1, alpha=0.5, colour="blue") +
  facet_wrap(~whichstrand.P9 + whichstrand.Zcw) + ggtitle("red = P9, blue = Zcw motif, 0/1 = strand, relative_position=8")

plot(table(tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][whichstrand.P9==whichstrand.Zcw]$relative_position), xlim=c(-20,50))
plot(table(tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][whichstrand.P9==whichstrand.Zcw & whichmotif.P9!=1]$relative_position), xlim=c(-20,50))
plot(table(tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][whichstrand.P9==whichstrand.Zcw & whichmotif.P9==1]$relative_position), xlim=c(-20,50))
dev.off()

tmp[!is.na(whichpos.Zcw) & !is.na(whichpos.P9)][,cor(whichstrand.P9,whichstrand.Zcw), by=whichmotif.P9]
# up to 87.5% +ve strand correlation

```




# Infer Motifs at Zcw ST sites

```{r}
zcw_ST_ALL_500w <- load_sequences("motifs/zcw_peaks_NonPromoters_300w.fasta", flank=250, sm.rm=F)

zcw_ST_ALL_500w_M1 <- findamotif(head(zcw_ST_ALL_500w,1000), len=7, seed=42, nits=200)

plot_moitf(zcw_ST_ALL_500w_M1, "ZcwST_M1")

zcw_ST_ALL_500w_M2 <- findamotif(sample(zcw_ST_ALL_500w,1000), len=7, seed=42, nits=200)
```

## Remove repeat sequences

```{r}
zcw_ST_ALL_500w <- load_sequences("motifs/zcw_peaks_NonPromoters_300w.fasta", flank=250, sm.rm=T)
zcw_ST_ALL_500w_M3 <- findamotif(head(zcw_ST_ALL_500w,1000), len=7, seed=42, nits=200)

plot_moitf(zcw_ST_ALL_500w_M3, "ZcwST_sm")

zcw_ST_ALL_500w_MB1 <- findamotif(sample(zcw_ST_ALL_500w,1000), len=7, seed=111, nits=100, motif_seed="random")
plot_moitf(zcw_ST_ALL_500w_MB1, "tmp")
```


# Infer Motif in unexplained ZcwST peaks

```{r}
system("bedtools getfasta -s -fi motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa -bed motifs/Zcw_alone_unexplained_500w.bed -name > motifs/Zcw_alone_unexplained_500w.fasta")

zcw_ST_unexplained <- load_sequences("motifs/Zcw_alone_unexplained_500w.fasta", flank=250, sm.rm=F)
zcw_ST_unexplained_M1 <- findamotif(head(zcw_ST_unexplained,1000), len=7, seed=42, nits=200)

plot_moitf(zcw_ST_unexplained_M1, "zcw_ST_unexplained_M1")


# export_PWM(zcw_ST_unexplained_M1, "zcw_ST_unexplained_M1", "motifs/zcw_ST_unexplained_M1.meme")
# 
# ~/meme/bin/tomtom \
#   -no-ssc -verbosity 1 -min-overlap 5 -dist pearson -evalue -thresh 10.0 \
#   motifs/zcw_ST_unexplained_M1.meme \
#   ../single-cell/motif_databases/MOUSE/HOCOMOCOv11_full_MOUSE_mono_meme_format.meme \
#   ../single-cell/motif_databases/HUMAN/HOCOMOCOv11_full_HUMAN_mono_meme_format.meme \
#   -o motifs/unexpZcw/
#   -text > zcw_ST_unexplained_M1.tsv


zcw_ST_unexplained_M2 <- findamotif(head(zcw_ST_unexplained,1000), len=7, nits=200, motif_seed = "random")

plot_moitf(zcw_ST_unexplained_M2, "zcw_ST_unexplained_M2")
```

## Without repeats

```{r}
zcw_ST_unexplained_sm <- load_sequences("motifs/Zcw_alone_unexplained_500w.fasta", flank=250, sm.rm=T)

# modified to get 4-5k
zcw_ST_unexplained_M3 <- findamotif(tail(head(zcw_ST_unexplained_sm,5000),1000), len=7, nits=200, seed = 42)

plot_moitf(zcw_ST_unexplained_M3, "tmp")
# not found in 4k-5k tranche
plot_moitf(zcw_ST_unexplained_M3, "zcw_ST_unexplained_M3")


wget https://github.com/shenwei356/seqkit/releases/download/v0.10.1/seqkit_linux_amd64.tar.gz
tar xf seqkit_linux_amd64.tar.gz

head -1 ../single-cell/sequencing/metadata/hg38_sizes.bed |
bedtools getfasta -fi motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa -bed stdin |
./seqkit locate --ignore-case --non-greedy --degenerate -p CNNCNNCNNCNNCNNCNNC --bed |
sed s/:0-248956422//g > motif_locs_CNN7.bed


# With stranded prior
zcw_ST_unexplained_M4 <- findamotif(head(zcw_ST_unexplained_sm,1000), len=7, nits=200, seed = 42, stranded_prior = T)

plot_moitf(zcw_ST_unexplained_M4, "zcw_ST_unexplained_M4")

export_PWM(zcw_ST_unexplained_M4, "zcw_ST_unexplained_M4", "motifs/zcw_ST_unexplained_M4.meme")

~/meme/bin/tomtom \
  -no-ssc -verbosity 1 -min-overlap 5 -dist pearson -evalue -thresh 10.0 \
  motifs/zcw_ST_unexplained_M4.meme \
  ../single-cell/motif_databases/MOUSE/HOCOMOCOv11_full_MOUSE_mono_meme_format.meme \
  ../single-cell/motif_databases/HUMAN/HOCOMOCOv11_full_HUMAN_mono_meme_format.meme \
  motifs/Altemose_P9_1.meme \
  motifs/Altemose_P9_2.meme \
  motifs/Altemose_P9_3.meme \
  motifs/Altemose_P9_4.meme \
  motifs/Altemose_P9_5.meme \
  motifs/Altemose_P9_6.meme \
  motifs/Altemose_P9_7.meme \
  -oc motifs/unexpZcw/
  -text > zcw_ST_unexplained_M4.tsv

zcw_ST_unexplained_M5 <- findamotif(head(zcw_ST_unexplained_sm,1000), len=7, nits=200, seed = 42, stranded_prior = T, plen=0.99)

plot_moitf(zcw_ST_unexplained_M5, "zcw_ST_unexplained_M5")

```

## 1k width

```{r}
system("bedtools getfasta -s -fi motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa -bed motifs/Zcw_alone_unexplained_1000w.bed -name > motifs/Zcw_alone_unexplained_1000w.fasta")
zcw_ST_unexplained_sm1k <- load_sequences("motifs/Zcw_alone_unexplained_1000w.fasta", flank=250, sm.rm=T)

zcw_ST_unexplained_M6 <- findamotif(head(zcw_ST_unexplained_sm1k,1000), len=7, nits=200, seed = 42)

plot_moitf(zcw_ST_unexplained_M6, "zcw_ST_unexplained_M6")

zcw_ST_unexplained_M7 <- findamotif(head(zcw_ST_unexplained_sm1k,1000), len=7, nits=200, seed = 42, stranded_prior = T)

plot_moitf(zcw_ST_unexplained_M7, "zcw_ST_unexplained_M7")



zcw_ST_unexplained_M8 <- findamotif(sample(zcw_ST_unexplained_sm1k,1000), len=7, nits=200, seed = 42, stranded_prior = T)


plot_moitf <- function(motif, name){
  pdf(width=10, height=5, file=paste0("motifs/motif_at_",name,".pdf"))
  print(ggseqlogo::ggseqlogo(get_PWM(motif)))
  print(ggseqlogo::ggseqlogo(get_PWM(motif, complement = T)))
  plot(motif$prior)
  plot(sort(motif$dt$regprob))
  print(ggplot(melt(data.table(motif$alphas, seq=1:nrow(motif$alphas)), id="seq"), aes(seq, value, colour=variable)) + geom_line())
  plot_motif_location(motif, top_n=1000)
  dev.off()
}

plot_moitf(zcw_ST_unexplained_M8, "zcw_ST_unexplained_M8")

```

## Enriched in Zcw?

Motif7


```{r}
system("bedtools getfasta -s -fi motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa -bed chr1_sample2.bed -name > chr1_sample2.fasta")

echo -e "chr1\t42315466\t42479181" > chr1_sample2.bed

cat chr1_sample2.fasta |
./seqkit locate --ignore-case --non-greedy --degenerate -p CNNNNNCNNCNNCNNCNNC --bed > chr1_sample_M7.bed


~/meme/bin/fimo --text --max-stored-scores 1000000 motifs/zcw_ST_unexplained_M4.meme motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa | tail -n +2 | cut -f 2-10 | sed 's/^\t//g' > fimo_M4_WG.tsv


M4 <- MotifFinder::read_meme("motifs/zcw_ST_unexplained_M4.meme")
tmp <- t(M4$pwm[19:1,4:1])
rownames(tmp) <- c("A", "C", "G", "T")
ggseqlogo::ggseqlogo(tmp)
ggseqlogo::ggseqlogo(tmp, method="prob")
dev.off()

ggseqlogo::ggseqlogo(M4)
#cut -f 4,5 fimo_out/fimo.tsv | awk -v OFS='\t'  '{print "chr1", $1+42315466, $2+42315466}' | tail -n +2 > fimo_out/fimo2.bed
```

```{r}
fimo <- fread("~/Downloads/fimo_M4_WG.tsv")
fimo$V7 <- NULL
names(fimo) <- c("chr","CI_start","CI_stop","strand","score","p","seq")

fimo <- prejoin(fimo, ext = 0)

fimo[, rep := grepl("a|c|g|t", seq)]

fraction_overlapN2(zcw_peaks[cov_input>=5], fimo, x1="Zcw UT", y1="M4", shift=1e8)
fraction_overlapN2(zcw_peaks[cov_input>=5], fimo[score>11], x1="Zcw UT", y1="M4", shift=1e8)
fraction_overlapN2(zcw_peaks[cov_input>=5], fimo[score>13], x1="Zcw UT", y1="M4", shift=1e8)
fraction_overlapN2(zcw_peaks[cov_input>=5], fimo[score>16], x1="Zcw UT", y1="M4", shift=1e8)

plot(fimo$score, type=)

fraction_overlapN2(zcw_peaks_cin[cov_input>=5], fimo, x1="Zcw UT", y1="M4", shift=1e8)
fraction_overlapN2(zcw_wP9_peaks[cov_input>=5], fimo, x1="Zcw UT", y1="M4", shift=1e8)

```


```{r}
head -1 ../single-cell/sequencing/metadata/hg38_sizes.bed |
bedtools getfasta -fi motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa -bed stdin |
./seqkit locate --ignore-case --non-greedy --degenerate -p CNNNNNCNNCNNCNNCNNC --bed |
sed s/:0-248956422//g > motif_locs_M7.bed


head -1 ../single-cell/sequencing/metadata/hg38_sizes.bed |
bedtools getfasta -fi motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa -bed stdin |
./seqkit locate --ignore-case --non-greedy --degenerate -p CHBCHBCHBCHBCHBCHBC --bed |
sed s/:0-248956422//g > motif_locs_CHB6.bed



CHB6_locs <- fread("~/Downloads/motif_locs_CHB6.bed")
names(CHB6_locs)[1:3] <- c("chr","CI_start","CI_stop")
CHB6_locs <- prejoin(CHB6_locs, ext = 0)
fraction_overlapN2(zcw_peaks[cov_input>=5 & chr=="chr1"], CHB6_locs, x1="Zcw UT", y1="CHB6", shift=1e8)

M4_locs <- fread("~/Downloads/motif_locs_M7.bed")
names(M4_locs)[1:3] <- c("chr","CI_start","CI_stop")
M4_locs <- prejoin(M4_locs)

fraction_overlapN2(zcw_peaks[cov_input>=5 & chr=="chr1"], M4_locs, x1="Zcw UT", y1="M7", shift=1e8)
fraction_overlapN2(HP9combo_peaks[cov_input>=5 & chr=="chr1"], M4_locs, x1="HP9combo", y1="M7", shift=1e8)
fraction_overlapN2(HP9N_peaks[cov_input>=5 & chr=="chr1"], M4_locs, x1="HP9N", y1="M7", shift=1e8)
fraction_overlapN2(CP9combo_peaks[cov_input>=5 & chr=="chr1"], M4_locs, x1="CP9combo", y1="M7", shift=1e8)



tmp <- zcw_peaks_cin[cov_input>5][chr!="chrX"][order(-enrichment)][set_matrix_zcw_peaks3[,which(Zcw_Alone==T & hP9_Cterm==F & H3K4me3_UT==F & Alu==F & H3K36me3_P9==F)]][chr=="chr1"]
fraction_overlapN2(tmp, M4_locs, x1="Unexplained Zcw UT", y1="M7", shift=1e8)
```

Motif (CNN)7

```{r}
M4_locs <- fread("~/Downloads/motif_locs_CNN7.bed")
names(M4_locs)[1:3] <- c("chr","CI_start","CI_stop")
M4_locs <- prejoin(M4_locs)

fraction_overlapN2(zcw_peaks[cov_input>=5 & chr=="chr1"], M4_locs, x1="Zcw UT", y1="CNN7", shift=1e8)
fraction_overlapN2(HP9combo_peaks[cov_input>=5 & chr=="chr1"], M4_locs, x1="HP9combo", y1="CNN7", shift=1e8)
fraction_overlapN2(HP9N_peaks[cov_input>=5 & chr=="chr1"], M4_locs, x1="HP9N", y1="CNN7", shift=1e8)
fraction_overlapN2(CP9combo_peaks[cov_input>=5 & chr=="chr1"], M4_locs, x1="CP9combo", y1="CNN7", shift=1e8)

# check at unexplained peaks
tmp <- zcw_peaks_cin[cov_input>5][chr!="chrX"][order(-enrichment)][set_matrix_zcw_peaks3[,which(Zcw_Alone==T & hP9_Cterm==F & H3K4me3_UT==F & Alu==F & H3K36me3_P9==F)]][chr=="chr1"]
fraction_overlapN2(tmp, M4_locs, x1="Unexplained Zcw UT", y1="CNN7", shift=1e8)
```


