---
title: "DMC1 Chipseq"
output: html_notebook
---

# Calculate Average Profiles

```{r}
# combine seperate chromosome files into one
cat ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chr*.5prime.bedgraph.gz > ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chrALL.5prime.bedgraph.gz

cat ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chr*.3prime.bedgraph.gz > ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chrALL.3prime.bedgraph.gz

# remove odd chromosomes
# reomve Mitochondria
# convert chr1 to 1

zgrep -v '_' ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chrALL.5prime.bedgraph.gz | grep -v 'M' | sed 's/chr//' > ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chrALLclean.5prime.bedgraph
zgrep -v '_' ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chrALL.3prime.bedgraph.gz | grep -v 'M' | sed 's/chr//' > ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph


# create bed file from composite file form Anjali
#1 = chr (20 = X?)
#4 = allele (B6, KO, or UNK)
#5 = hshared
#10 = motif center pos
awk -v OFS='\t' '$10=sprintf("%.0f",$10) {if($1 != "20" && $5=="0" && $4=="B6") print $1,$10,$10,"0","0","+";}'  B6_composite.txt > B6.bed

awk -v OFS='\t' '{if($1 != "20") print $1,$2,$2,"0","0","+";}'  KO.txt | tail -n +2 > KO.bed

# convert bedgraph to Bigwig
bedGraphToBigWig ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph ../../../single-cell/sequencing/metadata/mm10_sizes.chrom hom3p.bigWig
bedGraphToBigWig ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chrALLclean.5prime.bedgraph ../../../single-cell/sequencing/metadata/mm10_sizes.chrom hom5p.bigWig

# bigWig to profile
bwtool aggregate 2000:2000 B6.bed hom3p.bigWig B6_3p.tsv -fill=0
bwtool aggregate 2000:2000 B6.bed hom5p.bigWig B6_5p.tsv -fill=0

bwtool aggregate 2000:2000 KO.bed hom3p.bigWig KO_3p.tsv -fill=0
bwtool aggregate 2000:2000 KO.bed hom5p.bigWig KO_5p.tsv -fill=0

```

# plot profile

```{r}
library(data.table)
B6_3p <- fread("dmc1/new_data/B6_3p.tsv")
B6_5p <- fread("dmc1/new_data/B6_5p.tsv")

KO_3p <- fread("dmc1/new_data/KO_3p.tsv")
KO_5p <- fread("dmc1/new_data/KO_5p.tsv")


KO_5p$Strand <- "5 prime"
KO_5p$Genotype <- "B6 Prdm9 KO DMC1 regions"
KO_3p$Strand <- "3 prime"
KO_3p$Genotype <- "B6 Prdm9 KO DMC1 regions"

B6_3p$Strand <- "3 prime"
B6_3p$Genotype <- "B6 Wild Type DMC1 regions"
B6_5p$Strand <- "5 prime"
B6_5p$Genotype <- "B6 Wild Type DMC1 regions"

tmp <- rbind(KO_5p, KO_3p, B6_5p, B6_3p)
names(tmp)[1:2] <- c("Position","MeanCoverage")
tmp[, Genotype := factor(Genotype, levels=c("B6 Wild Type DMC1 regions","B6 Prdm9 KO DMC1 regions"))]

pdf("DMC1_SSDS.pdf", height = 5, width = 8)
ggplot(tmp, aes(Position, MeanCoverage,colour=Strand, group=Strand)) +
  geom_line() +
  facet_wrap(~Genotype, nrow=1) +
  scale_color_brewer(palette = "Set1") +
  xlab("Position from center (bp)") +
  theme_minimal() +
  theme(legend.position = "bottom",
        panel.border = element_rect(color = "grey40", fill = NA, size = 0.5),
        strip.background = element_rect(color = "grey40", size = 0.5))
dev.off()

```



# DMC1 Vs Spo11 in WT and KO

```{r, fig.width=6}
B6 <- fread("../data/dmc1/B6_composite.txt")
# cut -f 1 -d " " B6.txt | uniq
hom3p <- fread("../data/dmc1/ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chrALL.3prime.bedgraph")
hom5p <- fread("../data/dmc1/ssDNA_ZCWPW1_HOM_260619_rep1_type1_filtered_only_rmdup.chrALL.5prime.bedgraph")

hom5p[,end:=5]
hom3p[,end:=3]

hom <- rbind(hom3p,hom5p)

hom[,.N,by=end]

hist(hom[,.(V3-V2)]$V1, breaks=100)

names(hom) <- c("chr", "start", "stop", "d", "end")
hom[,chr := gsub("chr","",chr)]

# shift 3' and 5' to overlap
hom[end==3, start := start-400]
hom[end==3, stop := stop-400]
hom[end==5, start := start+400]
hom[end==5, stop := stop+400]

hom[,start := (start+stop)/2]
hom[,stop := start+1]

setkey(hom, chr, start, stop)
B6[,chr := as.character(chr)]
B6[chr==20, chr:="X"]
B6[,start := pos-1000]
B6[,stop := pos+1000]
setkey(B6, chr, start, stop)

str(hom[chr %in% as.character(c(1:19))])

hom[,.N,by=chr][order(-N)]
B6[,.N,by=chr][order(-N)] # no X or Y ....

joined <- foverlaps(B6, hom)

# str(foverlaps(hom, B6, type="within", nomatch = 0)) ## matches iff 'hom' is within 'B6'

joined[,.N,by=end]

#ggplot(joined, aes(stop-start, colour=is.na(d))) + geom_density(bw=1)

# hist(joined[is.na(pos),.(i.stop-i.start)]$V1, breaks=100)
# hist(joined[!is.na(pos),.(i.stop-i.start)]$V1, breaks=100)

#joined[,relative_pos := pos-i.start]
#ggplot(joined[!is.na(pos),.("sum"=sum(d)),by=c("relative_pos","end")], aes(relative_pos, sum, colour=as.character(end))) + geom_point(bw=50)

ggplot(joined[!is.na(d)], aes(pos-start, colour=as.character(end))) + geom_density(bw=20)

str(joined[!is.na(pos)])

plot(sort(joined[!is.na(pos),.(N=sum(d, na.rm = T)),by=c("pos","DMC1")]$N))

joined[is.na(d),d:=0]
```

```{r, fig.width=6}
perhotspot <- joined[!is.na(pos),.(N=sum(d, na.rm = T)),by=c("chr","pos","DMC1","SPO11","spo11_orig","dmc1_orig_heat","hshared")]
perhotspot$isX <- perhotspot$chr=="X"
perhotspot[chr!="X", Chromosome := "Autosome"]
perhotspot[chr=="X", Chromosome := "X"]

# Dmc1
setorder(perhotspot, dmc1_orig_heat)
perhotspot[chr!="X", bin:=cut(1:nrow(perhotspot[chr!="X"]), 50)]# <- 
perhotspot[chr=="X"]$bin <- cut(1:nrow(perhotspot[chr=="X"]), 20)

a <- ggplot(perhotspot[hshared==0,.(Zcw=mean(N),dmc1=mean(dmc1_orig_heat)),by=c("bin","isX")], aes(dmc1, Zcw, colour=isX)) + geom_point()

# Spo11
setorder(perhotspot, spo11_orig)
perhotspot[chr!="X", bin:=cut(1:nrow(perhotspot[chr!="X"]), 50)]# <- 
perhotspot[chr=="X"]$bin <- cut(1:nrow(perhotspot[chr=="X"]), 20)

b <- ggplot(perhotspot[hshared==0,.(Dmc1_WT=mean(dmc1_orig_heat),Spo11_WT=mean(SPO11)),by=c("bin","isX")], aes(Spo11_WT, Dmc1_WT, colour=isX)) +
 # geom_smooth(method="lm", colour="black") +
  geom_point() #+
#  facet_wrap(~isX, scales = "free")

c <- ggplot(perhotspot[hshared==0,.(Dmc1_KO=mean(N),Spo11_WT=mean(SPO11)),by=c("bin","isX")], aes(Spo11_WT, Dmc1_KO, colour=isX)) +
  #geom_smooth(method="lm", colour="black") +
  geom_point() #+
  #facet_wrap(~isX, scales = "free")
  
load("../data/dmc1/b6full.out")

cowplot::plot_grid(a,b,c)



ggplot(perhotspot[hshared==0,.(Dmc1_WT=mean(dmc1_orig_heat),Spo11_WT=mean(SPO11)),by=c("bin","isX")], aes(bin, Spo11_WT/Dmc1_WT, colour=isX)) +
 # geom_smooth(method="lm", colour="black") +
  geom_point() #+

 ggplot(perhotspot[hshared==0,.(Dmc1_KO=mean(N),Spo11_WT=mean(SPO11)),by=c("bin","isX")], aes(bin, Spo11_WT/Dmc1_KO, colour=isX)) +
  #geom_smooth(method="lm", colour="black") +
  geom_point() #+

# compare
plot(perhotspot[,.(Zcw=mean(N),dmc1=mean(dmc1_orig_heat)),by=c("bin","isX")][isX==T]$dmc1, mm[,3])
plot(perhotspot[,.(Zcw=mean(N),dmc1=mean(dmc1_orig_heat)),by=c("bin","isX")][isX==T]$Zcw, mm[1:20,4])


plot(perhotspot[,.(Zcw=mean(N),dmc1=mean(dmc1_orig_heat)),by=c("bin","isX")][isX==F]$dmc1, mm[,1])
plot(perhotspot[,.(Zcw=mean(N),dmc1=mean(dmc1_orig_heat)),by=c("bin","isX")][isX==F]$Zcw, mm[,2])

perhotspot[chr!="X",.(Zcw=mean(N),dmc1=mean(DMC1)),by=bin]
perhotspot[chr=="X",.(Zcw=mean(N),dmc1=mean(DMC1)),by=binX]

```

