---
title: "DMC1 Chipseq"
output: html_notebook
---

```{r}


# WTCHG_688471_1.fastq.gz
# GATCGGAAGAGCACACGTCTGAACTCCAGTCACATTCCTTTATCTCGTATG #TruSeq Adapter, Index 27 (97% over 44bp)
# GATCGGAAGAGCACACGTCTGAACTCCAGTCACATTCCTTTATCTCGTTTT #Index TruSeq adapter up to the bar code (5'-GATCGGAAGAGCACACGTCTGAACTCCAGTCA*C)

# WTCHG_688471_2.fastq.gz
# GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGT # Illumina Single End PCR Primer 1
# GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGT

# WTCHG_688472_1.fastq.gz
# GATCGGAAGAGCACACGTCTGAACTCCAGTCACCGTACGTAATCTCGTATG # TruSeq Adapter, Index 22
# GATCGGAAGAGCACACGTCTGAACTCCAGTCACCGTACGTAATTTCGTATT

# WTCHG_688472_2.fastq.gz
# GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGT # Illumina Single End PCR Primer 1
# GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGT



fastqc raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688471_1.fastq.gz -o qc/fastqc
fastqc raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688471_2.fastq.gz -o qc/fastqc

fastqc raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688472_1.fastq.gz -o qc/fastqc
fastqc raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688472_2.fastq.gz -o qc/fastqc

fastqc raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688472_0_1.fastq.gz -o qc/fastqc
fastqc raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688472_0_2.fastq.gz -o qc/fastqc


ASCIIGenome-1.15.0/ASCIIGenome raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688472.bam

ASCIIGenome-1.15.0/ASCIIGenome raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688471.bam raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688472.bam -fa ../single-cell/sequencing/metadata/Mus_musculus.GRCm38.dna_sm.toplevel.fasta

chr10:23398243-23398480



java -jar /homes/wells/saxony/single-cell/sequencing/software/picard.jar MarkDuplicates \
    INPUT=raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688471.bam \
    OUTPUT=undup/WTCHG_688471.bam \
    METRICS_FILE=undup/WTCHG_688471_metrics.txt \
    REMOVE_DUPLICATES=true

bedtools genomecov -bg -ibam undup/WTCHG_688471.bam > bedgraphs/WTCHG_688471.bedgraph


java -jar /homes/wells/saxony/single-cell/sequencing/software/picard.jar MarkDuplicates \
    INPUT=raw_data/bsg-ftp.well.ox.ac.uk/190524_D00529_0716_AH2G53BCX3/WTCHG_688472.bam \
    OUTPUT=undup/WTCHG_688472.bam \
    METRICS_FILE=undup/WTCHG_688471_metrics.txt \
    REMOVE_DUPLICATES=true

bedtools genomecov -bg -ibam undup/WTCHG_688472.bam > bedgraphs/WTCHG_688472.bedgraph

#samtools view undup/WTCHG_688471.bam | less

samtools view undup/WTCHG_688471.bam -L chr19_sample.bed -b -o sample_regions/WTCHG_688471_chr19.bam
samtools index sample_regions/WTCHG_688471_chr19.bam

chr6:103,647,017-103,651,428

b6 <- fread("~/Downloads/B6.bedgraph")
fwrite(b6[,.(paste0("chr",chr), pos, pos+1, heat)], "~/Downloads/B6_IGV.bedgraph", sep = "\t", col.names = F, quote = F)
```

```{r}
wget -qO- http://ftp.ensembl.org/pub/release-96/gtf/mus_musculus/Mus_musculus.GRCm38.96.gtf.gz \
    | gunzip --stdout - \
    | grep "protein_coding" - \
    | awk '$3 == "gene"' - \
    | awk '{ if ($0 ~ "transcript_id") print $0; else print $0" transcript_id \"\";"; }' - \
    | convert2bed -i gtf - \
    > deeptools/beds/Mus_Musculus_genes_prot_coding_ens.bed

```

# Plot SSDS Dmc1 reads at B6 and KO hotspots

```{r}

b6 <- fread("~/Downloads/B6.bedgraph")
fwrite(b6[,.(chr, pos, pos+1, heat)], "~/Downloads/B6_hotspots.bedgraph", sep = "\t", col.names = F, quote = F)
fwrite(b6[hshared==0,.(chr, pos, pos+1, heat)], "~/Downloads/B6_hotspots_nothshared.bedgraph", sep = "\t", col.names = F, quote = F)
fwrite(b6[hshared==1,.(chr, pos, pos+1, heat)], "~/Downloads/B6_hotspots_hshared.bedgraph", sep = "\t", col.names = F, quote = F)


b6 <- fread("~/Downloads/KO.txt")
fwrite(b6[,.(chr, pos, pos+1, heat)], "~/Downloads/B6_KO_hotspots.bedgraph", sep = "\t", col.names = F, quote = F)


computeMatrix reference-point \
    -S deeptools/bigwigs/WTCHG_688472_CPM.bw \
        deeptools/bigwigs/WTCHG_688471_CPM.bw \
        deeptools/bigwigs/WTCHG_688471_CPM_max50.bw \
        deeptools/bigwigs/WTCHG_688472_0_CPM.bw \
    -R deeptools/beds/B6_hotspots.bedgraph \
      deeptools/beds/B6_hotspots_nothshared.bedgraph \
      deeptools/beds/B6_hotspots_hshared.bedgraph \
      deeptools/beds/B6_KO_hotspots.bedgraph \
    -a 5000 \
    -b 5000 \
    -p 15 \
    --skipZeros \
    -o deeptools/matrices/Dmc1_r1.gz

plotProfile \
    -m deeptools/matrices/Dmc1_r1.gz \
    -out deeptools/plots/Dmc1_r1_AT_B6.pdf \
    --refPointLabel "Centre" \
    --samplesLabel Het Hom Hom50max Het0 \
    --numPlotsPerRow 2 \
    --perGroup \
    --regionsLabel B6 B6_notshared B6_shared B6_KO



cat 

#awk '{print "chr" $0;}' ../single-cell/sequencing/metadata/mm10_sizes.chrom > ../single-cell/sequencing/metadata/mm10_sizes_chr.chrom


#cut -f 1 ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph | uniq

# bedGraphToBigWig ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALLclean.5prime.bedgraph ../single-cell/sequencing/metadata/mm10_sizes.chrom ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALL.5prime.bw
# bedGraphToBigWig ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph ../single-cell/sequencing/metadata/mm10_sizes.chrom ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALL.3prime.bw
# bedGraphToBigWig ssDNA_ZCWPW1_HET_290519_rep1_type1_filtered_only_rmdup.chrALLclean.5prime.bedgraph ../single-cell/sequencing/metadata/mm10_sizes.chrom ssDNA_ZCWPW1_HET_290519_rep1_type1_filtered_only_rmdup.chrALL.5prime.bw
# bedGraphToBigWig ssDNA_ZCWPW1_HET_290519_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph ../single-cell/sequencing/metadata/mm10_sizes.chrom ssDNA_ZCWPW1_HET_290519_rep1_type1_filtered_only_rmdup.chrALL.3prime.bw

#wget http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigWigToBedGraph
#chmod 777 bigWigToBedGraph
#./bigWigToBedGraph deeptools/bigwigs/WTCHG_688472_CPM.bw tmp.bedgraph
#wc -l tmp.bedgraph
#./bigWigToBedGraph ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALL.3prime.bw tmp.bedgraph
#wc -l tmp.bedgraph

#bedtools makewindows -g ../single-cell/sequencing/metadata/mm10_sizes.chrom -w 100 > 100bpwindows.bed
#bedtools map -a 100bpwindows.bed -b ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph -c 4 -o mean > tmp.bed
##sed 's/\./0/' tmp.bed > tmp.bed
#grep -v '\.' tmp.bed > tmp2.bed
#bedGraphToBigWig tmp2.bed ../single-cell/sequencing/metadata/mm10_sizes.chrom tmp.bw




# remove odd chromosomes
# reomve Mitochondria
# convert chr1 to 1

grep -v '_' ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALL.5prime.bedgraph | grep -v 'M' | sed 's/chr//' > ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALLclean.5prime.bedgraph
grep -v '_' ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALL.3prime.bedgraph | grep -v 'M' | sed 's/chr//' > ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph

grep -v '_' ssDNA_ZCWPW1_HET_290519_rep1_type1_filtered_only_rmdup.chrALL.5prime.bedgraph | grep -v 'M' | sed 's/chr//' > ssDNA_ZCWPW1_HET_290519_rep1_type1_filtered_only_rmdup.chrALLclean.5prime.bedgraph
grep -v '_' ssDNA_ZCWPW1_HET_290519_rep1_type1_filtered_only_rmdup.chrALL.3prime.bedgraph | grep -v 'M' | sed 's/chr//' > ssDNA_ZCWPW1_HET_290519_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph

# convert bedgraph to bam for input into deeptools

bedToBam -i ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALLclean.5prime.bedgraph -g ../../single-cell/sequencing/metadata/mm10_sizes.chrom > dmc1_Hom_5p.bam
bedToBam -i ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph -g ../../single-cell/sequencing/metadata/mm10_sizes.chrom > dmc1_Hom_3p.bam

bedToBam -i ssDNA_ZCWPW1_HET_290519_rep1_type1_filtered_only_rmdup.chrALLclean.5prime.bedgraph -g ../../single-cell/sequencing/metadata/mm10_sizes.chrom > dmc1_Het_5p.bam
bedToBam -i ssDNA_ZCWPW1_HET_290519_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph -g ../../single-cell/sequencing/metadata/mm10_sizes.chrom > dmc1_Het_3p.bam

samtools index dmc1_Hom_5p.bam
samtools index dmc1_Hom_3p.bam
samtools index dmc1_Het_5p.bam
samtools index dmc1_Het_3p.bam

# calculate coverage using deeptools

bamCoverage \
      --bam dmc1_Hom_5p.bam \
      -o dmc1_Hom_5p.bw \
      --normalizeUsing CPM \
      -p 15

bamCoverage \
      --bam dmc1_Hom_3p.bam \
      -o dmc1_Hom_3p.bw \
      --normalizeUsing CPM \
      -p 15

bamCoverage \
      --bam dmc1_Het_5p.bam \
      -o dmc1_Het_5p.bw \
      --normalizeUsing CPM \
      -p 15

bamCoverage \
      --bam dmc1_Het_3p.bam \
      -o dmc1_Het_3p.bw \
      --normalizeUsing CPM \
      -p 15

# create profile plot

computeMatrix reference-point \
    -S  dmc1_Hom_3p.bw \
        dmc1_Hom_5p.bw \
    -R deeptools/beds/B6_hotspots.bedgraph \
      deeptools/beds/B6_KO_hotspots.bedgraph \
    -a 5000 \
    -b 5000 \
    -p 15 \
    --skipZeros \
    -o deeptools/matrices/Dmc1_r1_type1.gz

plotProfile \
    -m deeptools/matrices/Dmc1_r1_type1.gz \
    -out deeptools/plots/Dmc1_r1_type1_AT_B6.pdf \
    --refPointLabel "Centre" \
    --samplesLabel Dmc1_3prime Dmc1_5prime \
    --perGroup \
    --regionsLabel B6_WT_Hotspot_Locations B6_Prdm9_KO_Hotspot_Locations

  
```


# DMC1 reads averaged over genes (TSS Exons TES)

```{r}
wget http://ftp.ensembl.org/pub/release-96/gtf/mus_musculus/Mus_musculus.GRCm38.96.gtf.gz
gunzip Mus_musculus.GRCm38.96.gtf.gz

computeMatrix scale-regions \
    --metagene \
    -S dmc1_Hom_3p.bw \
        dmc1_Hom_5p.bw \
        dmc1_Het_3p.bw \
        dmc1_Het_5p.bw \
      -R Mus_musculus.GRCm38.96.gtf \
      -m 8000 \
      -a 4000 \
      -b 4000 \
      -bs 50 \
      -p 15 \
      --skipZeros \
      --missingDataAsZero \
      -o gtf_exon_dmc1.gz

plotProfile \
    -m gtf_exon_dmc1.gz \
    --numPlotsPerRow 2 \
    --samplesLabel Hom3p Hom5p Het3p Het5p \
    -out gtf_exon_dmc1.pdf


# type 1 and type 2 reads unfiltered

computeMatrix scale-regions \
    --metagene \
    -S ../deeptools/bigwigs/WTCHG_688472_CPM.bw \
        ../deeptools/bigwigs/WTCHG_688471_CPM.bw \
      -R Mus_musculus.GRCm38.96.gtf \
      -m 8000 \
      -a 4000 \
      -b 4000 \
      -bs 50 \
      -p 15 \
      --skipZeros \
      --missingDataAsZero \
      -o gtf_exon_dmc1_unfiltered.gz

plotProfile \
    -m gtf_exon_dmc1_unfiltered.gz \
    --numPlotsPerRow 2 \
    --samplesLabel Het Hom \
    -out gtf_exon_dmc1_unfiltered.pdf



```

# What fraction of QCreads are Type1 reads?

```{r}
bedtools intersect -v -a ../filtered/WTCHG_688471.bam \
  -b ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALLclean.5prime.bedgraph
      ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALLclean.3prime.bedgraph > WTCHG_688471.notT1.bam

samtools flagstat -@ 10 ../filtered/WTCHG_688471.bam
# 159986 + 0 in total (QC-passed reads + QC-failed reads)
# 79993 + 0 read1
# 79993 + 0 read2


samtools flagstat -@ 10 WTCHG_688471.notT1.bam
# 78871 + 0 in total (QC-passed reads + QC-failed reads)
# 39428 + 0 read1
# 39443 + 0 read2

# 159986-78871 = 81k
```

# DMC1 Vs Spo11 in WT and KO

```{r, fig.width=6}
B6 <- fread("../data/dmc1/B6_composite.txt")
# cut -f 1 -d " " B6.txt | uniq
hom3p <- fread("../data/dmc1/ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALL.3prime.bedgraph")
hom5p <- fread("../data/dmc1/ssDNA_ZCWPW1_HOM_290519_rep1_type1_filtered_only_rmdup.chrALL.5prime.bedgraph")

hom5p[,end:=5]
hom3p[,end:=3]

hom <- rbind(hom3p,hom5p)

hom[,.N,by=end]

hist(hom[,.(V3-V2)]$V1, breaks=100)

names(hom) <- c("chr", "start", "stop", "d", "end")
hom[,chr := gsub("chr","",chr)]

# shift 3' and 5' to overlap
hom[end==3, start := start-400]
hom[end==3, stop := stop-400]
hom[end==5, start := start+400]
hom[end==5, stop := stop+400]

hom[,start := (start+stop)/2]
hom[,stop := start+1]

setkey(hom, chr, start, stop)
B6[,chr := as.character(chr)]
B6[chr==20, chr:="X"]
B6[,start := pos-1000]
B6[,stop := pos+1000]
setkey(B6, chr, start, stop)

str(hom[chr %in% as.character(c(1:19))])

hom[,.N,by=chr][order(-N)]
B6[,.N,by=chr][order(-N)] # no X or Y ....

joined <- foverlaps(B6, hom)

# str(foverlaps(hom, B6, type="within", nomatch = 0)) ## matches iff 'hom' is within 'B6'

joined[,.N,by=end]

#ggplot(joined, aes(stop-start, colour=is.na(d))) + geom_density(bw=1)

# hist(joined[is.na(pos),.(i.stop-i.start)]$V1, breaks=100)
# hist(joined[!is.na(pos),.(i.stop-i.start)]$V1, breaks=100)

#joined[,relative_pos := pos-i.start]
#ggplot(joined[!is.na(pos),.("sum"=sum(d)),by=c("relative_pos","end")], aes(relative_pos, sum, colour=as.character(end))) + geom_point(bw=50)

ggplot(joined[!is.na(d)], aes(pos-start, colour=as.character(end))) + geom_density(bw=20)

str(joined[!is.na(pos)])

plot(sort(joined[!is.na(pos),.(N=sum(d, na.rm = T)),by=c("pos","DMC1")]$N))

joined[is.na(d),d:=0]
```

```{r, fig.width=6}
perhotspot <- joined[!is.na(pos),.(N=sum(d, na.rm = T)),by=c("chr","pos","DMC1","SPO11","spo11_orig","dmc1_orig_heat","hshared")]
perhotspot$isX <- perhotspot$chr=="X"
perhotspot[chr!="X", Chromosome := "Autosome"]
perhotspot[chr=="X", Chromosome := "X"]

# Dmc1
setorder(perhotspot, dmc1_orig_heat)
perhotspot[chr!="X", bin:=cut(1:nrow(perhotspot[chr!="X"]), 50)]# <- 
perhotspot[chr=="X"]$bin <- cut(1:nrow(perhotspot[chr=="X"]), 20)

a <- ggplot(perhotspot[hshared==0,.(Zcw=mean(N),dmc1=mean(dmc1_orig_heat)),by=c("bin","isX")], aes(dmc1, Zcw, colour=isX)) + geom_point()

# Spo11
setorder(perhotspot, spo11_orig)
perhotspot[chr!="X", bin:=cut(1:nrow(perhotspot[chr!="X"]), 50)]# <- 
perhotspot[chr=="X"]$bin <- cut(1:nrow(perhotspot[chr=="X"]), 20)

b <- ggplot(perhotspot[hshared==0,.(Dmc1_WT=mean(dmc1_orig_heat),Spo11_WT=mean(SPO11)),by=c("bin","isX")], aes(Spo11_WT, Dmc1_WT, colour=isX)) +
 # geom_smooth(method="lm", colour="black") +
  geom_point() #+
#  facet_wrap(~isX, scales = "free")

c <- ggplot(perhotspot[hshared==0,.(Dmc1_KO=mean(N),Spo11_WT=mean(SPO11)),by=c("bin","isX")], aes(Spo11_WT, Dmc1_KO, colour=isX)) +
  #geom_smooth(method="lm", colour="black") +
  geom_point() #+
  #facet_wrap(~isX, scales = "free")
  


cowplot::plot_grid(a,b,c)



ggplot(perhotspot[hshared==0,.(Dmc1_WT=mean(dmc1_orig_heat),Spo11_WT=mean(SPO11)),by=c("bin","isX")], aes(bin, Spo11_WT/Dmc1_WT, colour=isX)) +
 # geom_smooth(method="lm", colour="black") +
  geom_point() #+

 ggplot(perhotspot[hshared==0,.(Dmc1_KO=mean(N),Spo11_WT=mean(SPO11)),by=c("bin","isX")], aes(bin, Spo11_WT/Dmc1_KO, colour=isX)) +
  #geom_smooth(method="lm", colour="black") +
  geom_point() #+

# compare
plot(perhotspot[,.(Zcw=mean(N),dmc1=mean(dmc1_orig_heat)),by=c("bin","isX")][isX==T]$dmc1, mm[,3])
plot(perhotspot[,.(Zcw=mean(N),dmc1=mean(dmc1_orig_heat)),by=c("bin","isX")][isX==T]$Zcw, mm[1:20,4])


plot(perhotspot[,.(Zcw=mean(N),dmc1=mean(dmc1_orig_heat)),by=c("bin","isX")][isX==F]$dmc1, mm[,1])
plot(perhotspot[,.(Zcw=mean(N),dmc1=mean(dmc1_orig_heat)),by=c("bin","isX")][isX==F]$Zcw, mm[,2])

perhotspot[chr!="X",.(Zcw=mean(N),dmc1=mean(DMC1)),by=bin]
perhotspot[chr=="X",.(Zcw=mean(N),dmc1=mean(DMC1)),by=binX]

```



```{r}
bamCoverage \
      --bam filtered/WTCHG_688471.bam \
      -o deeptools/bigwigs/WTCHG_688471_CPM_min50.bw \
      --normalizeUsing CPM \
      --centerReads \
      --extendReads \
      -p 5 \
      --minFragmentLength 50

bamCoverage \
      --bam filtered/WTCHG_688471.bam \
      -o deeptools/bigwigs/WTCHG_688471_CPM_max50.bw \
      --normalizeUsing CPM \
      --centerReads \
      --extendReads \
      -p 5 \
      --maxFragmentLength 50

```

