---
title: "Domain Architechture"
output: html_notebook
---


```{r, fig.width=10}

pkgload::load_all("~/Dropbox/Github/drawProteins/")

rel_json <- drawProteins::get_features("Q9H0M4 Q6IR42") # H2RBY7 E2RFJ2 F7DK93 A0A3S5ZP38 G1PE65
rel_data <- drawProteins::feature_to_dataframe(rel_json)

rel_data[which(rel_data$type=="COILED"),]$description <- "Coiled Coil"

rel_data <- rbind(rel_data,
                  data.frame(type="DOMAIN2", description="SCP-1 ?", begin=76, end=228, length=228-76+1, accession="Q6IR42", entryName="ZCPW1_MOUSE", taxid=10090, order=1),
                  data.frame(type="DOMAIN2", description="MBD ?", begin=462, end=541, length=541-462+1, accession="Q6IR42", entryName="ZCPW1_MOUSE", taxid=10090, order=1),
                  data.frame(type="DOMAIN2", description="MBD ?", begin=471, end=573, length=573-471+1, accession="Q9H0M4", entryName="ZCPW1_HUMAN", taxid=9606, order=2))
rownames(rel_data[nrow(rel_data),]) <- "featuresTemp.23"
draw_domains2 <- function(p,
                        data = data,
                        label_domains = TRUE,
                        label_size = 4,
                        show.legend = TRUE,
                        type = "DOMAIN2"){
    begin=end=description=NULL
    p <- p + ggplot2::geom_rect(data= data[data$type == type,],
            mapping=ggplot2::aes(xmin=begin,
                        xmax=end,
                        ymin=order-0.25,
                        ymax=order+0.25,
                        fill=description),
                        show.legend = show.legend, alpha=0.5)

    if(label_domains == TRUE){
        p <- p + ggplot2::geom_label(data = data[data$type == type, ],
                        ggplot2::aes(x = begin + (end-begin)/2,
                            y = order,
                            label = description),
                            size = label_size)
    }

    return(p)
}

rel_data[rel_data$order==1,]$order <- 4
rel_data[rel_data$order==2,]$order <- 3

rel_data[rel_data$order==4,]$order <- 2
rel_data[rel_data$order==3,]$order <- 1

p <- draw_canvas(rel_data)
p <- draw_chains(p, rel_data, label_chains = F)
p <- draw_domains(p, rel_data)
p <- draw_domains2(p, rel_data)
p <- draw_domains(p, rel_data, type="ZN_FING")
p <- draw_domains(p, rel_data, type="COILED")
p <- p + scale_y_continuous(breaks=seq_along(unique(rel_data$entryName)), labels = c("Human","Mouse")) + xlim(0,NA)
p <- p + theme_bw(base_size = 20) + # white background
    theme(panel.grid.minor=element_blank(), 
          panel.grid.major=element_blank(),
          axis.ticks = element_blank(),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          panel.border = element_blank()) +
  scale_fill_brewer(palette = "Set1")

p

# add exact AA positions
p2 <- p +
  geom_text(aes(x = begin, y = order+0.30, label = begin),
               data=data.table(rel_data)[type %in% c("DOMAIN","ZN_FING","COILED","DOMAIN2")]) +
  geom_text(aes(x = end, y = order-0.30, label = end),
               data=data.table(rel_data)[type %in% c("CHAIN","DOMAIN","ZN_FING","COILED","DOMAIN2")]) +
  theme(legend.position = "none")

pdf("../results/Zcwpw1_domain_architechture.pdf", width = 10, height = 5)
p2
dev.off()
```


# PRDM9
```{r, fig.width=10}
rel_json <- drawProteins::get_features("Q9NQV7 Q96EQ9")
rel_data <- drawProteins::feature_to_dataframe(rel_json)

rel_data$description <- gsub(";.+$", "", rel_data$description, perl = T)
rel_data$description <- gsub(" [0-9]+$", "", rel_data$description, perl = T)

p <- draw_canvas(rel_data)
p <- draw_chains(p, rel_data, label_chains = F)
p <- draw_domains(p, rel_data)
p <- draw_domains(p, rel_data, type="ZN_FING", label_domains = F)
p <- p + scale_y_continuous(breaks=seq_along(unique(rel_data$entryName)), labels = unique(rel_data$entryName)) + xlim(0,NA)
p <- p + theme_bw(base_size = 20) + # white background
    theme(panel.grid.minor=element_blank(), 
        panel.grid.major=element_blank()) +
    theme(axis.ticks = element_blank(), 
        ) + #axis.text.y = element_blank()
    theme(panel.border = element_blank()) +
  scale_fill_brewer(palette = "Set1")

pdf("../results/Prdm9_domain_architechture.pdf", width = 10, height = 5)
p + theme(legend.position = "bottom") + labs(fill = "Feature")
dev.off()

# 
# draw_protein <- function(ac="P07683"){
#   rel_json <- drawProteins::get_features(ac)
#     
#   rel_data <- drawProteins::feature_to_dataframe(rel_json)
#   p <- draw_canvas(rel_data)
#   p <- draw_chains(p, rel_data)
#   return(p)
# 
# }
# 
# draw_protein()

```

# Conservation

```{r}
# Old Alignment

# cons <- seqinr::read.fasta("~/Dropbox/Downloads/Human_ZCWPW1_orthologues.fa", as.string=T)
# ids <- names(cons)
# cons <- as.character(cons)
# 
# mat <- sapply(cons, function(x) strsplit(x, "")[[1]])
# dimnames(mat) <- NULL
# colnames(mat) <- ids
# 
# consensus <- sapply(1:ncol(t(mat)), function(x) tail(names(sort(table(mat[x,]))), 1))
# mat2 <- cbind(mat, consensus)
# ident <- apply(mat2, 1, function(x) mean(x[-ncol(mat2)]==x[ncol(mat2)]))
# 
# human_aa <- mat[,"ENSP00000381109_Hsap/1-648"] != "-"
# 
# jalview <- data.table(t(fread("~/Dropbox/Downloads/Zcwpw1_Jalview.txt")[,-1]))
# names(jalview) <- c("Conservation","Quality","Occupancy","Consensus","Logo","SMERFS")
# jalview$Conservation <- as.numeric(jalview$Conservation)
# jalview$Quality <- as.numeric(jalview$Quality)
# jalview$Occupancy <- as.numeric(jalview$Occupancy)
# jalview$Consensus <- as.numeric(jalview$Consensus)
# jalview$SMERFS <- as.numeric(jalview$SMERFS)
# jalview <- jalview[!is.na(Conservation)]
# jalview$JSD <- fread("~/Dropbox/Downloads/JSD.txt")$V2

#consurf <- fread("~/Dropbox/Downloads/consurf.grades", skip = 12)
```


Using Simon's alignment

```{r}
# load("../../ZCWPW1/fordanielalignment.out")
# saveRDS(ouralignmat,"data/Simons_Zcwpw1_alignment.rds")

ouralignmat <- readRDS("../data/Simons_Zcwpw1_alignment.rds")
ouralignmat[is.na(ouralignmat)] <- "-"
seqs <- apply(ouralignmat, 1, function(x) paste0(x, collapse=""))
MotifFinder::export_FASTA(seqs, "data/simons_alignment.fa")

# from http://compbio.cs.princeton.edu/conservation/
JSD <- fread("../data/JSD_simons_alignment.txt")
setnames(JSD, c("id", "JSD", "seq"))

#jalview[human_aa]
cons <- ggplot(JSD, aes(1:sum(human_aa),scale(JSD))) + geom_col(width=1, fill="black") + theme_minimal() + xlab("Amino acid number") + ylab("Jensen-Shannon\n divergence (normalised)")

pdf("../results/Zcwpw1_domain_architechture_cons.pdf", width = 10, height = 8)
cowplot::plot_grid(p2, cons, ncol=1, align = "v",axis = "lrtb", rel_heights = c(2,1), labels = "AUTO")
dev.off()

```
