---
title: "FC 100bp Windows"
output: html_notebook
---

```{r}
knitr::opts_knit$set(base.dir = "results")
knitr::opts_chunk$set(fig.path = "100bp_windows/")
knitr::opts_chunk$set(dev="pdf")
knitr::opts_chunk$set(fig.show="hold")

library(data.table)
library(ggplot2)
library(gghighlight)
library(cowplot)
library(mgcv)

source("../functions.R")
```


# Load data


```{r}

CpG_100bp <- fread("../data/CpG/CpG_hg38_100bpwindows.bed")

M7 <- fread("../data/motifs/motif_locs_M7_100bpwindows.bed")


h3k4me3_UT_100bp <- fread("../data/peaks/ForceCalledPeaks_NA15-SRR5627150_vs_NA15-SRR5627142_AT_genome.windows.100wide.100slide.bed.bed")

h3k36me3_UT_100bp <- fread("../data/peaks/ForceCalledPeaks_NA15-SRR5627148_vs_NA15-SRR5627142_AT_genome.windows.100wide.100slide.bed.bed")

h3k4me3_hP9_100bp <- fread("../data/peaks/ForceCalledPeaks_NA15-SRR5627152_AND_NA15-SRR5627153_vs_NA15-SRR5627143_AT_genome.windows.100wide.100slide.bed.bed")

h3k36me3_hP9_100bp <- fread("../data/peaks/ForceCalledPeaks_NA15-SRR5627149_vs_NA15-SRR5627143_AT_genome.windows.100wide.100slide.bed.bed")


zcw_100bp <- fread("../data/peaks/ForceCalledPeaks_WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144_AT_genome.windows.100wide.100slide.bed.bed")

zcw_hP9_100bp <- fread("../data/peaks/ForceCalledPeaks_WTCHG_538916_223180_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144_AT_genome.windows.100wide.100slide.bed.bed")

zcw_cP9_100bp <- fread("../data/peaks/ForceCalledPeaks_WTCHG_538916_224192_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144_AT_genome.windows.100wide.100slide.bed.bed")

zcw_hCVC <- fread("../data/peaks/ForceCalledPeaks_WTCHG_538916_223180_vs_WTCHG_538916_221156_AT_genome.windows.100wide.100slide.bed.bed")

zcw_cCVC <- fread("../data/peaks/ForceCalledPeaks_WTCHG_538916_224192_vs_WTCHG_538916_221156_AT_genome.windows.100wide.100slide.bed.bed")


hP9combo <- fread("../data/peaks/ForceCalledPeaks_NA15-SRR5627146_AND_NA15-SRR5627147_vs_NA15-SRR5627143_AT_genome.windows.100wide.100slide.bed.bed")

cP9combo <- fread("../data/peaks/ForceCalledPeaks_NA15-SRR5627145_AND_NA15-SRR5627144_vs_NA15-SRR5627143_AT_genome.windows.100wide.100slide.bed.bed")

hP9GFP <- fread("../data/peaks/ForceCalledPeaks_NA15-SRR5627138_AND_NA15-SRR5627139_vs_NA15-SRR5627140_AT_genome.windows.100wide.100slide.bed.bed")


# CAREFULL WITH ORDERING WHEN ADDING NEW COLUMNS!

forcecalled <- h3k4me3_UT_100bp

stopifnot(sum(diff(forcecalled[chr=="chr1"]$center_start)<100)==0)
stopifnot(sum(diff(forcecalled[chr=="chr11"]$center_start)<100)==0)
stopifnot(sum(diff(forcecalled[chr=="chr18"]$center_start)<100)==0)



forcecalled$h3k4me3_UT <- forcecalled$enrichment
forcecalled$h3k4me3_UT_enrichment <- forcecalled$h3k4me3_UT
forcecalled$h3k4me3_UT_reads <- forcecalled[,cov_r1 + cov_r2]
forcecalled$h3k4me3_UT_cov_input <- forcecalled$cov_input
forcecalled$h3k4me3_UT_pvalue <- forcecalled$pvalue


forcecalled$h3k36me3_UT_enrichment <- h3k36me3_UT_100bp$enrichment
forcecalled$h3k36me3_UT_pvalue <- h3k36me3_UT_100bp$pvalue

forcecalled$CpG <- CpG_100bp[V1 %in% paste0("chr",c(1:22,"X"))]$V4
forcecalled$M7 <- M7[V1 %in% paste0("chr",c(1:22,"X"))]$V4

forcecalled$h3k4me3_hP9_enrichment <- h3k4me3_hP9_100bp$enrichment
forcecalled$h3k4me3_hP9_reads <- h3k4me3_hP9_100bp[,cov_r1 + cov_r2]
forcecalled$h3k4me3_hP9_cov_input <- h3k4me3_hP9_100bp$cov_input
forcecalled$h3k4me3_hP9_pvalue <- h3k4me3_hP9_100bp$pvalue

forcecalled$hP9GFP_enrichment <- hP9GFP$enrichment

forcecalled$zcw_enrichment <- zcw_100bp$enrichment
forcecalled$zcw_pvalue <- zcw_100bp$pvalue
forcecalled$zcw_cov_input <- zcw_100bp$cov_input

forcecalled$zcw_hP9_enrichment <- zcw_hP9_100bp$enrichment
forcecalled$zcw_hP9_reads <- zcw_hP9_100bp[,cov_r1 + cov_r2]
forcecalled$zcw_hP9_pvalue <- zcw_hP9_100bp$pvalue
#forcecalled$zcw_P9_cov_input <- zcw_hP9_100bp$cov_input # identical to zcw

forcecalled$zcw_cP9_enrichment <- zcw_cP9_100bp$enrichment
forcecalled$zcw_cP9_pvalue <- zcw_cP9_100bp$pvalue

forcecalled$zcw_hCVC_enrichment <- zcw_hCVC$enrichment
forcecalled$zcw_cCVC_enrichment <- zcw_cCVC$enrichment

forcecalled$hP9combo_enrichment <- hP9combo$enrichment
forcecalled$hP9combo_cov_input <- hP9combo$cov_input
forcecalled$hP9combo_pvalue <- hP9combo$pvalue


forcecalled$cP9combo_enrichment <- cP9combo$enrichment
forcecalled$cP9combo_cov_input <- cP9combo$cov_input
forcecalled$cP9combo_pvalue <- cP9combo$pvalue


forcecalled$h3k36me3_hP9_enrichment <- h3k36me3_hP9_100bp$enrichment
forcecalled$h3k36me3_hP9_cov_input <- h3k36me3_hP9_100bp$cov_input
forcecalled$h3k36me3_hP9_pvalue <- h3k36me3_hP9_100bp$pvalue

sum(forcecalled$center_start!=h3k36me3_hP9_100bp$center_start)
#stopifnot(sum(diff(forcecalled$center_start)<100)==0)

# Islands

CpG_Island_100bp <- fread("../data/CpG/cpgIslandExtUnmasked_100bpwindows.bed")
setnames(CpG_Island_100bp, c("chr","center_start","center_stop","CpG_Island"))
forcecalled <- merge(forcecalled, CpG_Island_100bp)

CpG_Island_100bp <- fread("../data/CpG/cpgIslandExtUnmasked_100bpwindows_f1.bed")
setnames(CpG_Island_100bp, c("chr","center_start","center_stop","CpG_Island_f1"))
forcecalled <- merge(forcecalled, CpG_Island_100bp)


## Meth

cpg_Meth_100bp <- fread("../data/CpG/cpg_Meth_100bpwindows.bed")
setnames(cpg_Meth_100bp, c("chr","center_start","center_stop","meth","unmeth"))

setkey(forcecalled, chr, center_start, center_stop)
setkey(cpg_Meth_100bp, chr, center_start, center_stop)

forcecalled <- merge(forcecalled, cpg_Meth_100bp)
forcecalled[, prop_meth := (meth+3) / (meth + unmeth + 6)]

```

```{r}

Alu100bp <- fread("../data/repeats/Repeat_MaskerSorted_Alu_QC_100bp.bed")
setnames(Alu100bp, c("chr","center_start","center_stop","Alu"))
forcecalled <- merge(forcecalled, Alu100bp)

Alu100bp_f0.5 <- fread("../data/repeats/Repeat_MaskerSorted_Alu_QC_100bp_f0.5.bed") # 0.5 actually 1.0, meaning Alu must overlap full 100bp
setnames(Alu100bp_f0.5, c("chr","center_start","center_stop","Aluf1"))
forcecalled <- merge(forcecalled, Alu100bp_f0.5)

# Non Alu Repeats
NotAlu100bp <- fread("../data/repeats/Repeat_MaskerSorted_NotAlu_100bp.bed")
setnames(NotAlu100bp, c("chr","center_start","center_stop","NotAlu"))
forcecalled <- merge(forcecalled, NotAlu100bp)

###### Zcw peaks
zcw_peak <- fread("../data/peaks/SingleBasePeaks.WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL.100bpWindows.bed")
setnames(zcw_peak, c("chr","center_start","center_stop","ZcwPeak"))
forcecalled <- merge(forcecalled, zcw_peak)

zcw_peak_rand <- fread("../data/peaks/Zcw_random_1bp.100bpWindows.bed")
setnames(zcw_peak_rand, c("chr","center_start","center_stop","ZcwPeakRand"))
forcecalled <- merge(forcecalled, zcw_peak_rand)

#just using Chromosome 1, as many data points
#grep -P 'chr1\\t|chr\\t' 
#forcecalled <- forcecalled[chr=="chr1"]
forcecalled <- forcecalled[chr!="chrX"]
```

# CpG Island Methylation

```{r CGI_meth}
cpgImeth <- fread("../data/CpG/cpgIsland_Meth.bed")
setnames(cpgImeth, c("chr","center_start","center_stop","meth","unmeth"))
cpgImeth[, len := center_stop-center_start]
cpgImeth[, meth_prop := (meth+3) / (meth + unmeth + 6)]

#pdf("Meth_by_len.pdf")
# longer islands more unmethylated
plot(cpgImeth[meth+unmeth > 10]$len, cpgImeth[meth+unmeth > 10]$meth_prop, log="x", pch=".")
hist(cpgImeth[meth+unmeth > 10][len<350]$meth_prop, breaks=100)
hist(cpgImeth[meth+unmeth > 10][len>500]$meth_prop, breaks=100)
#dev.off()


H3K4UT_CGI <- fread("../data/peaks/ForceCalledPeaks_NA15-SRR5627150_AND_NA15-SRR5627151_vs_NA15-SRR5627142_AT_cpgIslandExtUnmasked.bed.bed")
Zcw_CGI <- fread("../data/peaks/ForceCalledPeaks_WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144_AT_cpgIslandExtUnmasked.bed.bed")

setkey(cpgImeth, chr,center_start,center_stop)
setkey(H3K4UT_CGI, chr,center_start,center_stop)
setkey(Zcw_CGI, chr,center_start,center_stop)

cgi <- H3K4UT_CGI[Zcw_CGI[cpgImeth]]
# add prefixes post merge, i=zcw
tmp <- grep("i\\.", names(cgi), value=T)
setnames(cgi, tmp, gsub("i\\.","zcw.",tmp))

# chrY and alts arent forcecalled
cgi[is.na(enrichment), .N,by=chr][order(-N)]

cgi[, isMethylated := meth_prop > 0.5 & (meth+unmeth > 10)]
cgi[, isH3K4 := enrichment > 0.1 & cov_input>=5]


# large diff in H3K4 between unmeth & meth
cgi[!is.na(enrichment), mean(enrichment), by=meth_prop > 0.5]
#   meth_prop         V1
# 1:      TRUE  0.5512826
# 2:     FALSE 14.9644733

# smaller diff for Zcwpw1 enrichment
cgi[!is.na(enrichment), mean(zcw.enrichment), by=meth_prop > 0.5]
#   meth_prop       V1
# 1:      TRUE 1.465204
# 2:     FALSE 1.129344

cgi[zcw.cov_input>=5, mean(zcw.enrichment), by=c("isMethylated","isH3K4")]
#    isMethylated isH3K4       V1
# 1:         TRUE  FALSE 1.482543
# 2:        FALSE   TRUE 1.179341
# 3:        FALSE  FALSE 1.049465
# 4:         TRUE   TRUE 1.187082



#pdf("CpG/CGI_meth.pdf")
# promoters mostly unmethylated
ggplot(cgi[meth+unmeth > 10 & cov_input>=5 & zcw.cov_input>=5], aes(enrichment+0.01, meth_prop, colour=sqrt(zcw.enrichment))) +
  geom_point(stroke=0, size=0.5) + scale_x_log10() +
  scale_colour_viridis_c()

ggplot(cgi[meth+unmeth > 10 & cov_input>=5 & zcw.cov_input>=5], aes(enrichment+0.01, meth_prop)) +
  geom_hex(aes(fill = stat(log(count+0.1))), bins=200) + theme(legend.position = "bottom") +
  scale_fill_viridis_c(direction = -1) + scale_x_log10()

ggplot(cgi[meth+unmeth > 10 & cov_input>=5 & zcw.cov_input>=5], aes(enrichment+0.01, meth_prop, z=sqrt(zcw.enrichment))) +
  stat_summary_hex(bins=200) + theme(legend.position = "bottom") +
  scale_fill_viridis_c(name = "sqrt(Zcw)", direction = -1) + scale_x_log10()

ggplot(cgi[meth+unmeth > 10 & cov_input>=5 & enrichment==0 & zcw.cov_input>=5], aes(as.character(round(meth_prop,1)), zcw.enrichment)) +
  geom_boxplot() +
  ggtitle("CGI with no H3K4")

ggplot(forcecalled[zcw_cov_input>=5], aes(CpG, zcw_enrichment+0.01)) +
  geom_hex(aes(fill = stat(log(count+0.1))), bins=200) + theme(legend.position = "bottom") +
  scale_fill_viridis_c(direction = -1) + scale_y_log10()

ggplot(forcecalled[meth+unmeth > 10 & zcw_cov_input>=5], aes(prop_meth, zcw_enrichment+0.01)) +
  geom_hex(aes(fill = stat(log(count+0.1))), bins=200) + theme(legend.position = "bottom") +
  scale_fill_viridis_c(direction = -1) + scale_y_log10()

```

# Prob Peak

```{r prob_peak}

#By meth Status of CpgIslands - not much diff

forcecalled[, methStatus := "Unk"]
forcecalled[prop_meth>0.75, methStatus := "Meth"]
forcecalled[prop_meth<0.25, methStatus := "UnMeth"]

pPeak_CpG <- rbind(
forcecalled[CpG_Island>0,.(PeakProb=mean(ZcwPeak),.N, Type="Observed"), by=c("CpG","methStatus")][order(CpG)],
forcecalled[CpG_Island>0,.(PeakProb=mean(ZcwPeakRand>0),.N, Type="Random"), by=c("CpG","methStatus")][order(CpG)])

pPeak_CpG[,sep := sqrt((PeakProb * (1- PeakProb))/N), by=c("CpG","methStatus")]

ggplot(pPeak_CpG[CpG<16], aes(CpG, PeakProb, colour=Type)) +
  geom_pointrange(aes(ymax = PeakProb + 2*sep, ymin = PeakProb - 2*sep)) +
  scale_color_brewer(palette = "Set1") +
  facet_wrap(~methStatus)+
  ggtitle("By methStatus")

# By CpG Island - ok

pPeak_CpG <- rbind(
forcecalled[,.(PeakProb=mean(ZcwPeak),.N, Type="Observed"), by=c("CpG","CpG_Island")][order(CpG)],
forcecalled[,.(PeakProb=mean(ZcwPeakRand>0),.N, Type="Random"), by=c("CpG","CpG_Island")][order(CpG)])

pPeak_CpG[,sep := sqrt((PeakProb * (1- PeakProb))/N), by=c("CpG","CpG_Island")]

ggplot(pPeak_CpG[CpG_Island<2 & CpG<16], aes(CpG, PeakProb, colour=Type)) +
  geom_pointrange(aes(ymax = PeakProb + 2*sep, ymin = PeakProb - 2*sep)) +
  scale_color_brewer(palette = "Set1") +
  facet_wrap(~CpG_Island) +
  ggtitle("By CpG Island")


# Repeats

pPeak_CpG <- rbind(
forcecalled[CpG_Island==0,.(PeakProb=mean(ZcwPeak),.N, Type="Observed"), by=c("CpG","Alu")][order(CpG)],
forcecalled[CpG_Island==0,.(PeakProb=mean(ZcwPeakRand>0),.N, Type="Random"), by=c("CpG","Alu")][order(CpG)])

pPeak_CpG[,sep := sqrt((PeakProb * (1- PeakProb))/N), by=c("CpG","Alu")]

ggplot(pPeak_CpG[ CpG<16], aes(CpG, PeakProb, colour=Type)) +
  geom_pointrange(aes(ymax = PeakProb + 2*sep, ymin = PeakProb - 2*sep)) +
  scale_color_brewer(palette = "Set1") +
  facet_wrap(~Alu) +
  ggtitle("By Alu (non CpGIsland)")


# Not alu

pPeak_CpG <- rbind(
forcecalled[CpG_Island==0 & NotAlu<3,.(PeakProb=mean(ZcwPeak),.N, Type="Observed"), by=c("CpG","NotAlu")][order(CpG)],
forcecalled[CpG_Island==0 & NotAlu<3,.(PeakProb=mean(ZcwPeakRand>0),.N, Type="Random"), by=c("CpG","NotAlu")][order(CpG)])

pPeak_CpG[,sep := sqrt((PeakProb * (1- PeakProb))/N), by=c("CpG","NotAlu")]

ggplot(pPeak_CpG[ CpG<16], aes(CpG, PeakProb, colour=Type)) +
  geom_pointrange(aes(ymax = PeakProb + 2*sep, ymin = PeakProb - 2*sep)) +
  scale_color_brewer(palette = "Set1") +
  facet_wrap(~NotAlu) +
  ggtitle("By Not Alu (non CpGIsland)")


#dev.off()

# Normal peaks
pPeak_CpG <- rbind(
forcecalled[CpG_Island==0 & NotAlu==0 & Alu==0,.(PeakProb=mean(ZcwPeak),.N, Type="Observed"), by=c("CpG")][order(CpG)],
forcecalled[CpG_Island==0 & NotAlu==0 & Alu==0,.(PeakProb=mean(ZcwPeakRand>0),.N, Type="Random"), by=c("CpG")][order(CpG)])

pPeak_CpG[,sep := sqrt((PeakProb * (1- PeakProb))/N), by=c("CpG")]

ggplot(pPeak_CpG[CpG<16], aes(CpG, PeakProb, colour=Type)) +
  geom_pointrange(aes(ymax = PeakProb + 2*sep, ymin = PeakProb - 2*sep)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("Non Repeat, Non Alu, Non CpG Island Peaks")
#dev.off()


# Summarise for every 3 rows to get 300bp windows (~approx, ignoring changes between chrs)

forcecalled300 <- forcecalled[,lapply(.SD, sum), 
                    by = gl(ceiling(nrow(forcecalled)/3), 3, nrow(forcecalled)),
                    .SDcols=c("CpG","ZcwPeak","ZcwPeakRand","CpG_Island","CpG_Island_f1","Alu","NotAlu","meth","unmeth","Aluf1")]

forcecalled300[, prop_meth := (meth+3) / (meth + unmeth + 6)]

forcecalled300[, methStatus := "Unk"]
forcecalled300[prop_meth>0.75, methStatus := "Meth"]
forcecalled300[prop_meth<0.25, methStatus := "UnMeth"]

forcecalled300[CpG==0, methStatus := "NA"]

# Normal peaks 300bp
pPeak_CpG <- rbind(
forcecalled300[CpG_Island==0 & NotAlu==0 & Alu==0,.(PeakProb=mean(ZcwPeak>0),.N, Type="Observed"), by=c("CpG","methStatus")][order(CpG)],
forcecalled300[CpG_Island==0 & NotAlu==0 & Alu==0,.(PeakProb=mean(ZcwPeakRand>0),.N, Type="Random"), by=c("CpG","methStatus")][order(CpG)])

pPeak_CpG[,sep := sqrt((PeakProb * (1- PeakProb))/N), by=c("CpG")]
pPeak_CpG[methStatus=="NA", methStatus := "Meth"]

p <- ggplot(pPeak_CpG[CpG<20 & methStatus!="Unk"], aes(CpG, PeakProb, colour=Type, alpha=methStatus)) +
  geom_pointrange(aes(ymax = PeakProb + 2*sep, ymin = PeakProb - 2*sep)) +
  scale_color_brewer(palette = "Set1", name="Peak Type") +
  ylab("Probability of window overlapping a ZCWPW1 peak") +
  xlab("CpG count in 300bp window") +
  #ggtitle("300bp, Non Repeat, Non Alu, Non CpG Island Peaks") +
  scale_alpha_manual(values=c("Meth"=1, "UnMeth"=0.5), name="Methylation\nStatus") +
  theme_minimal()
p
#dev.off()

saveRDS(p,"../data/CpG/NormalPeaks300bp.rds")



# Alu peaks 300bp
pPeak_CpG <- rbind(
forcecalled300[CpG_Island==0 & NotAlu==0 & Aluf1==1,.(PeakProb=mean(ZcwPeak),.N, Type="Observed"), by=c("CpG","methStatus")][order(CpG)],
forcecalled300[CpG_Island==0 & NotAlu==0 & Aluf1==1,.(PeakProb=mean(ZcwPeakRand>0),.N, Type="Random"), by=c("CpG","methStatus")][order(CpG)])

pPeak_CpG[,sep := sqrt((PeakProb * (1- PeakProb))/N), by=c("CpG")]

ggplot(pPeak_CpG[CpG<20 & methStatus!="Unk"], aes(CpG, PeakProb, colour=Type, alpha=methStatus)) +
  geom_pointrange(aes(ymax = PeakProb + 2*sep, ymin = PeakProb - 2*sep)) +
  scale_color_brewer(palette = "Set1", name="Peak Type") +
  ylab("Probability of overlapping a Zcwpw1 peak") +
  ggtitle("300bp,Alu Peaks") +
  #coord_cartesian(ylim=c(0,0.8)) +
  scale_alpha_manual(values=c("Meth"=1, "NA"=1, "UnMeth"=0.5), name="Methylation\nStatus") +
  theme_minimal()
#dev.off()

forcecalled300[CpG_Island_f1>0, CGI:=1]

# CpG_Islands 300bp
pPeak_CpG <- rbind(
forcecalled300[NotAlu==0 & Alu==0,.(PeakProb=mean(ZcwPeak),.N, Type="Observed"), by=c("CpG","methStatus","CGI")][order(CpG)],
forcecalled300[NotAlu==0 & Alu==0,.(PeakProb=mean(ZcwPeakRand>0),.N, Type="Random"), by=c("CpG","methStatus","CGI")][order(CpG)])

pPeak_CpG[,sep := sqrt((PeakProb * (1- PeakProb))/N), by=c("CpG")]

ggplot(pPeak_CpG[CpG<25 & methStatus!="Unk"], aes(CpG, PeakProb, colour=Type, alpha=methStatus)) +
  geom_pointrange(aes(ymax = PeakProb + 2*sep, ymin = PeakProb - 2*sep)) +
  scale_color_brewer(palette = "Set1", name="Peak Type") +
  ylab("Probability of overlapping a Zcwpw1 peak") +
  ggtitle("300bp,CpGI") +
  facet_wrap(~CGI) +
  scale_alpha_manual(values=c("Meth"=1, "NA"=1, "UnMeth"=0.5), name="Methylation\nStatus") +
  theme_minimal()
#dev.off()


```

# Regressions

```{r regressions}
zcw_lm <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment + h3k36me3_UT_enrichment, forcecalled[zcw_cov_input>5])
zcw_lm2 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment + h3k36me3_UT_enrichment + CpG, forcecalled[zcw_cov_input>5])
zcw_lm3 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment + h3k36me3_UT_enrichment + CpG + CpG_Island, forcecalled[zcw_cov_input>5])
zcw_lm2.5 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment + h3k36me3_UT_enrichment + CpG, forcecalled[zcw_cov_input>5 & chr=="chr1"]) 
zcw_lm4 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment + h3k36me3_UT_enrichment + CpG + M7, forcecalled[zcw_cov_input>5 & chr=="chr1"])
zcw_lm4.5 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment + h3k36me3_UT_enrichment + CpG + M7, forcecalled[zcw_cov_input>5])
zcw_lm5 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment + h3k36me3_UT_enrichment + CpG * CpG_Island, forcecalled[zcw_cov_input>5])
zcw_lm6 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment + h3k36me3_UT_enrichment + CpG * CpG_Island + M7, forcecalled[zcw_cov_input>5])

forcecalled[,mean(zcw_enrichment),by=CpG_Island]
forcecalled[chr=="chr1",mean(zcw_enrichment),by=M7]
forcecalled[,mean(zcw_enrichment),by=CpG][order(-CpG)]


zcw_lm7 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment + h3k36me3_UT_enrichment + CpG + CpG_Island + prop_meth, forcecalled[zcw_cov_input>5])

zcw_lm8 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment + h3k36me3_UT_enrichment + CpG * prop_meth  + M7 + CpG_Island, forcecalled[zcw_cov_input>5])

zcw_lm9 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment * h3k36me3_UT_enrichment + CpG * prop_meth  + M7 + CpG_Island, forcecalled[zcw_cov_input>5])

zcw_lm10 <- lm(zcw_enrichment ~ h3k4me3_UT_enrichment * h3k36me3_UT_enrichment , forcecalled[zcw_cov_input>5])

```


```{r CpG_Proportions_windows, fig.width = 7, fig.height = 5}
forcecalled2 <- copy(forcecalled)
setorder(forcecalled2, zcw_enrichment)

forcecalled2$bin <- cut(1:nrow(forcecalled2), 35)

binned_cpg_prop <- forcecalled2[zcw_cov_input>5,.(enrichment=mean(zcw_enrichment),
                                        CpG_0=mean(CpG==0),
                     CpG_1=mean(CpG==1),
                     CpG_2_3=mean(CpG>=2 & CpG<=3),
                     CpG_4_6=mean(CpG>=4 & CpG<=6),
                     CpG_7_9=mean(CpG>=7 & CpG<=9),
                     CpG_10_12=mean(CpG>=10 & CpG<=12),
                     CpG_13gt=mean(CpG>=13)),
                  by=bin]

binned_cpg_prop <- melt(binned_cpg_prop, id.vars=c("bin","enrichment"))

cpg_prop_rand <- forcecalled2[zcw_cov_input>5,.(enrichment=mean(zcw_enrichment),
                                        CpG_0=mean(CpG==0),
                     CpG_1=mean(CpG==1),
                     CpG_2_3=mean(CpG>=2 & CpG<=3),
                     CpG_4_6=mean(CpG>=4 & CpG<=6),
                     CpG_7_9=mean(CpG>=7 & CpG<=9),
                     CpG_10_12=mean(CpG>=10 & CpG<=12),
                     CpG_13gt=mean(CpG>=13))]

cpg_prop_rand <- melt(cpg_prop_rand, id.vars = "enrichment")
cpg_prop_rand$enrichment=-0.5
cpg_prop_rand$bin="Average"

tmp <- rbind(binned_cpg_prop, cpg_prop_rand)

tmp$CpG <- gsub("CpG_","",tmp$variable)
tmp$CpG <- gsub("_"," to ",tmp$CpG)
tmp$CpG <- gsub("gt"," >=",tmp$CpG)
tmp$CpG <- factor(tmp$variable, levels = tmp$variable, labels=tmp$CpG)
tmp$CpG <- forcats::fct_rev(tmp$CpG)

# many bins have mean 0 enrichment, so average them so they don't pile up
tmp <- tmp[,.(value=mean(value)),by=c("enrichment","CpG")]

# if the width is too big, the right column overlaps the left
# meaning the center of the col is no longer the mean enrichment
tmp[,diffnext := c(diff(enrichment),0.5), by=CpG]
tmp[,diffprev := c(0.5,diff(enrichment)), by=CpG]
tmp[diffnext<0, diffnext := 1]
tmp[diffprev<0, diffprev := 1]
tmp[,maxwidth := min(c(diffnext,diffprev)), by=c("enrichment")]


ggplot(tmp, aes(enrichment, value, fill=CpG)) +
  geom_col(aes(width=maxwidth)) +
  scale_x_continuous(labels=c("Random \n Windows",0:5), breaks=c(-0.5,0:5)) +
  scale_fill_brewer(direction = -1, palette = "RdBu") +
  ylab("Proportion of 100bp Windows") +
  xlab("Enrichment of Zcwpw1") +
  labs(fill="Number\nof CpGs")

#ggsave("CpG/CpG_Proportions_windows.pdf", width = 7, height = 5)

ggplot(binned_cpg_prop, aes(bin, -value, fill=variable)) +
  geom_col(width=1) +
  scale_fill_brewer()
```

```{r CpG_Zcw}
#pdf("CpG_Zcw.pdf")
ggplot(forcecalled[zcw_cov_input>5,.(mean_enrich=mean(zcw_enrichment),stdv=sd(zcw_enrichment),.N), by=CpG],
       aes(CpG, mean_enrich)) +
  geom_pointrange(aes(ymin = mean_enrich - 2*(stdv/sqrt(N)), ymax = mean_enrich + 2*(stdv/sqrt(N))))
#dev.off()
```


# qPRC control regions

```{r qPCR_control, eval=FALSE}

quantile(forcecalled$h3k36me3_hP9_enrichment, c(0.8,0.9,0.95,0.975,0.98))
quantile(forcecalled$h3k4me3_hP9_enrichment, c(0.8,0.9,0.95,0.975,0.98))
quantile(forcecalled$h3k4me3_UT_pvalue, 1-c(0.8,0.9,0.95,0.975,0.98))
quantile(forcecalled$cP9combo_enrichment, c(0.8,0.9,0.95,0.975,0.98))
quantile(forcecalled$hP9combo_enrichment, c(0.8,0.9,0.95,0.975,0.98))
quantile(forcecalled$zcw_hP9_enrichment, c(0.8,0.9,0.95,0.975,0.98))
quantile(forcecalled$zcw_cP9_enrichment, c(0.8,0.9,0.95,0.975,0.98))
quantile(forcecalled$hP9GFP_enrichment, c(0.8,0.9,0.95,0.975,0.98))

# positive for hPrdm9 Zcw peak
pos <- forcecalled[cov_input>5 & 
              hP9combo_enrichment>1.5 & 
              hP9GFP_enrichment>1 &
              zcw_hP9_enrichment>1.5 & 
              #zcw_cP9_enrichment<0.1 & # NB human Zcw sites, are also enriched when cP9 c/t, only 1 region found when using this
              h3k36me3_hP9_enrichment>1.5 & 
              h3k4me3_hP9_enrichment>2.4 & 
              h3k4me3_UT_pvalue>0.5 &
              h3k36me3_UT_pvalue>0.1 &
              cP9combo_enrichment<0.1]

# positive for cPrdm9 Zcw peak
pos <- forcecalled[cov_input>5 & 
              hP9combo_enrichment<0.1 & 
              hP9GFP_enrichment<0.1 &
              #zcw_hP9_enrichment>1.5 & 
              zcw_cP9_enrichment>1.7 &
              #h3k36me3_hP9_enrichment>1.5 & 
              #h3k4me3_hP9_enrichment>2.4 & 
              h3k4me3_UT_pvalue>0.5 &
              h3k36me3_UT_pvalue>0.1 &
              cP9combo_enrichment>3]

# positive for Zcw without Prdm9
pos <- forcecalled[cov_input>5 & 
              hP9combo_enrichment<0.1 & 
              hP9GFP_enrichment<0.1 &
              zcw_hP9_enrichment>1.5 & 
              #zcw_cP9_enrichment<0.1 & # NB human Zcw sites, are also enriched when cP9 c/t, only 1 region found when using this
              h3k4me3_UT_pvalue>0.5 &
              h3k36me3_UT_pvalue>0.1 &
              cP9combo_enrichment<0.1 &
                zcw_enrichment>2]

pos[sample(1:nrow(pos),5)]

# hP9 +ve Zcw
a <- forcecalled[(chr=="chr1" & center_start==19060500) |
              (chr=="chr17" & center_start==50584700) |
              (chr=="chr3" & center_start==127563900) |
              (chr=="chr19" & center_start==11423300) |
              (chr=="chr4" & center_start==38733300) |
              (chr=="chr15" & center_start==25767900)]

# cP9 +vs Zcw
b <- forcecalled[(chr=="chr5" & center_start==126310600) |
              (chr=="chr1" & center_start==60858700) |
              (chr=="chr19" & center_start==29128700) |
              (chr=="chr7" & center_start==154372200) |
              (chr=="chr8" & center_start==52201700)]
    
# P9 -ve Zcw
c <- forcecalled[(chr=="chr2" & center_start==204098500) |
            (chr=="chr3" & center_start==154940200) |
            (chr=="chr8" & center_start==129397800) |
            (chr=="chr2" & center_start==223205300) |
            (chr=="chr4" & center_start==18418700)]
 
a$class <- "humanPrdm9+ve_&_ZCWPW1"
b$class <- "chimpPrdm9+ve_&_ZCWPW1"
c$class <- "Prdm9-ve_&_ZCWPW1"

fwrite(rbind(a,b,c), "qPRC_regions_12.09.2019.csv")

# forcecalled[chr=="chr5"][abs(center_start-96003240)<1e2]
# pos[chr=="chr2"][abs(center_start-100016200)<1e3]
 

# collate force called enrichment values from actual qPCR coordinates into matrix
read_plus <- function(flnm) {
    fread(flnm) %>% 
    mutate(filename = flnm)
}

tbl_with_sources <-
    list.files(path="peaks",
               pattern = "AT_qPCRregions_18-09-19.bed.bed", 
               full.names = T) %>% 
    map_df(~read_plus(.))

tbl_with_sources <- data.table(tbl_with_sources)[center_start!=1]
tbl_with_sources[, filename := gsub("peaks/ForceCalledPeaks_","",filename)]
tbl_with_sources[, filename := gsub("_AT_qPCRregions_18-09-19.bed.bed","",filename)]

regions <- fread("../data/forcepeaks/qPCRregions_18-09-19.bed")
regions <- regions[V2!=1]

regions$class <- rep(c("humanPrdm9+ve_&_ZCWPW1","chimpPrdm9+ve_&_ZCWPW1","Prdm9-ve_&_ZCWPW1"),each=5)
setnames(regions, c("chr","center_start","center_stop","class"))

tbl_with_sources <- regions[tbl_with_sources, on=c("chr","center_start","center_stop")]

casted <- dcast(tbl_with_sources, class + chr + center_start + center_stop ~ filename, value.var="enrichment")
fwrite(casted, "qPCRregions_forcecalled_enrichments_18-09-19.csv")

```


# Dependence of transfected Zcw on UT

```{r transfected_UT_dependence}
# plot(forcecalled$zcw_enrichment,forcecalled$zcw_hP9_enrichment, pch=".", log="xy")
# plot(forcecalled$zcw_enrichment,forcecalled$zcw_hP9_enrichment, pch=".")

# nb intercept doesn't move
fclm <- lm(zcw_hP9_enrichment ~ 1 + zcw_enrichment + hP9combo_enrichment, forcecalled[zcw_cov_input>5])
summary(fclm)

fclm <- lm(zcw_hP9_enrichment ~ 0 + zcw_enrichment + zcw_enrichment:hP9combo_enrichment, forcecalled[zcw_cov_input>5])
summary(fclm)

fclm <- lm(log(zcw_hP9_enrichment+0.1) ~ log(zcw_enrichment+0.1), forcecalled[zcw_cov_input>5])
summary(fclm)

####
# Counts
####

# count hP9
cHP9 <- ggplot(forcecalled[zcw_cov_input>5], (aes(zcw_enrichment, zcw_hP9_enrichment))) +
  geom_hex(aes(fill = stat(log(count+0.1))), bins=200) +
  scale_fill_viridis_c(direction = -1) +
  theme(legend.position = "bottom")

  
# count cP9
cCP9 <- ggplot(forcecalled[zcw_cov_input>5], (aes(zcw_enrichment, zcw_cP9_enrichment))) +
  geom_hex(aes(fill = stat(log(count+0.1))), bins=200) +
  scale_fill_viridis_c(direction = -1) +
  theme(legend.position = "bottom")

####
# Color by P9 enrichment
####

# hP9 colour
p1 <- ggplot(forcecalled[zcw_cov_input>5], (aes(zcw_enrichment, zcw_hP9_enrichment, z=log(hP9combo_enrichment+0.1)))) +
  stat_summary_hex(bins=200) +
  scale_fill_viridis_c(name = "log(hP9+0.1)", direction = -1) +
  theme(legend.position = "bottom")

# cP9 colour
p2 <- ggplot(forcecalled[zcw_cov_input>5], (aes(zcw_enrichment, zcw_hP9_enrichment, z=log(cP9combo_enrichment+0.1)))) +
  stat_summary_hex(bins=200) +
  scale_fill_viridis_c(name = "log(cP9+0.1)", direction = -1) +
  theme(legend.position = "bottom")

# hP9 colour
p3 <- ggplot(forcecalled[zcw_cov_input>5], (aes(zcw_enrichment, zcw_cP9_enrichment, z=log(hP9combo_enrichment+0.1)))) +
  stat_summary_hex(bins=200) +
  scale_fill_viridis_c(name = "log(hP9+0.1)", direction = -1) +
  theme(legend.position = "bottom")

# cP9 colour
p4 <- ggplot(forcecalled[zcw_cov_input>5], (aes(zcw_enrichment, zcw_cP9_enrichment, z=log(cP9combo_enrichment+0.1)))) +
  stat_summary_hex(bins=200) +
  scale_fill_viridis_c(name = "log(cP9+0.1)", direction = -1) +
  theme(legend.position = "bottom")


####
# Color by CVC enrichment
####

p1CVC <- ggplot(forcecalled[zcw_cov_input>5], (aes(zcw_enrichment, zcw_hP9_enrichment, z=log(zcw_hCVC_enrichment+0.1)))) +
  stat_summary_hex(bins=200) +
  scale_fill_viridis_c(name = "log(hCVC+0.1)", direction = -1) +
  theme(legend.position = "bottom")

# cP9 colour
p2CVC <- ggplot(forcecalled[zcw_cov_input>5], (aes(zcw_enrichment, zcw_hP9_enrichment, z=log(zcw_cCVC_enrichment+0.1)))) +
  stat_summary_hex(bins=200) +
  scale_fill_viridis_c(name = "log(cCVC+0.1)", direction = -1) +
  theme(legend.position = "bottom")

# hP9 colour
p3CVC <- ggplot(forcecalled[zcw_cov_input>5], (aes(zcw_enrichment, zcw_cP9_enrichment, z=log(zcw_hCVC_enrichment+0.1)))) +
  stat_summary_hex(bins=200) +
  scale_fill_viridis_c(name = "log(hCVC+0.1)", direction = -1) +
  theme(legend.position = "bottom")

# cP9 colour
p4CVC <- ggplot(forcecalled[zcw_cov_input>5], (aes(zcw_enrichment, zcw_cP9_enrichment, z=log(zcw_cCVC_enrichment+0.1)))) +
  stat_summary_hex(bins=200) +
  scale_fill_viridis_c(name = "log(cCVC+0.1)", direction = -1) +
  theme(legend.position = "bottom")

```

# Arrange & Save plots

```{r Zcw_vs_CoTransfection, fig.width=14, fig.height=14}
#pdf(width = 14, height = 14, file = "Zcw_vs_CoTransfection.pdf")
plot_grid(cHP9 + ylim(0,8),
          cHP9 + scale_x_log10(limits=c(0.1,NA)) + scale_y_log10(limits=c(0.1,NA)),
          cCP9 + ylim(0,8),
          cCP9 + scale_x_log10(limits=c(0.1,NA)) + scale_y_log10(limits=c(0.1,NA))
          )

plot_grid(p1 + ylim(0,8), p2 + ylim(0,8), p3 + ylim(0,8), p4 + ylim(0,8))

plot_grid(p1 + scale_x_log10(limits=c(0.1,NA)) + scale_y_log10(limits=c(0.1,NA)),
          p2 + scale_x_log10(limits=c(0.1,NA)) + scale_y_log10(limits=c(0.1,NA)),
          p3 + scale_x_log10(limits=c(0.1,NA)) + scale_y_log10(limits=c(0.1,NA)),
          p4 + scale_x_log10(limits=c(0.1,NA)) + scale_y_log10(limits=c(0.1,NA)))

plot_grid(p1CVC + ylim(0,8), p2CVC + ylim(0,8), p3CVC + ylim(0,8), p4CVC + ylim(0,8))
#dev.off()
```

# Split into definately bound and definately not, for visual clarity

```{r Zcw_vs_CoTransfection2}
#pdf("Zcw_vs_CoTransfection2.pdf")
p1 + ylim(0,8) + gghighlight(hP9combo_pvalue<1e-9, unhighlighted_colour="darkgrey")
p1 + ylim(0,8) + gghighlight(hP9combo_pvalue==1, unhighlighted_colour="darkgrey")

cHP9 + ylim(0,8) + gghighlight(hP9combo_pvalue<1e-9, unhighlighted_colour="darkgrey")
cHP9 + ylim(0,8) + gghighlight(hP9combo_pvalue==1, unhighlighted_colour="darkgrey")

forcecalled[,hP9Combo.bound := NA]
forcecalled[hP9combo_pvalue==1, hP9Combo.bound := FALSE]
forcecalled[hP9combo_pvalue<1e-9, hP9Combo.bound := TRUE]
```

```{r Zcw_vs_CoTransfection_summary_plot_allchr, fig.width=12}
p1 <- ggplot(forcecalled[zcw_cov_input>5 & hP9combo_cov_input>5][!zcw_enrichment<1e-2 & !zcw_hP9_enrichment<1e-2], (aes(zcw_enrichment, zcw_hP9_enrichment, z=hP9Combo.bound))) +
  #stat_summary_hex(bins=200, alpha=0.5, fun= function(x) as.character(mean(x))) + # na.rm=T
  stat_summary_hex(bins=200, alpha=0.5, fun= function(x)if(length(x)>=3){as.character(mean(x))}else{NA}) + # na.rm=T
  geom_density_2d(aes(colour=hP9Combo.bound),bins=15) +
  labs(x="Enrichment of ZCWPW1", y="Enrichment of ZCWPW1 when cotransfected with human PRDM9") +
  scale_fill_manual(labels = c("p = 1", "p < 10^(-9)"), values = c("#E41A1C", "#377EB8")) +
  scale_colour_manual(values = c("#E41A1C", "#377EB8"), guide=FALSE) +
  labs(fill="Evidence of human PRDM9 binding") +
  theme(legend.position = "bottom") +
  ylim(0,8) + xlim(0,8)

# log scale
p1l <- ggplot(forcecalled[zcw_cov_input>5 & hP9combo_cov_input>5], (aes(log(zcw_enrichment+0.1), log(zcw_hP9_enrichment+0.1), z=hP9Combo.bound))) +
  stat_summary_hex(bins=200, alpha=0.5, fun= function(x) as.character(mean(x))) + # na.rm=T
  #scale_fill_brewer(palette="Set1") +
  scale_fill_manual(labels = c("p = 1", "p < 10^(-9)"), values = c("#E41A1C", "#377EB8")) +
  labs(fill="Evidence of hPrdm9cterm binding") +
  theme(legend.position = "bottom") + 
  xlim(-0.5,NA) + ylim(-0.5, NA)


# CHIMP

#1e-4
forcecalled[,cP9Combo.bound := NA]
forcecalled[cP9combo_pvalue==1, cP9Combo.bound := FALSE]
forcecalled[cP9combo_pvalue<1e-3, cP9Combo.bound := TRUE]

p2 <- ggplot(forcecalled[zcw_cov_input>5 & cP9combo_cov_input>5][!zcw_enrichment<1e-2 & !zcw_cP9_enrichment<1e-2], (aes(zcw_enrichment, zcw_cP9_enrichment, z=cP9Combo.bound))) +
  #stat_summary_hex(bins=200, alpha=0.5, fun= function(x) as.character(mean(x))) + # na.rm=T
  stat_summary_hex(bins=200, alpha=0.5, fun= function(x)if(length(x)>=3){as.character(mean(x))}else{NA}) + # na.rm=T
  geom_density_2d(aes(colour=cP9Combo.bound),bins=15) +
  labs(x="Enrichment of ZCWPW1", y="Enrichment of ZCWPW1 when cotransfected with chimp PRDM9") +
  scale_fill_manual(labels = c("p = 1", "p < 10^(-3)"), values = c("#E41A1C", "#377EB8")) +
  scale_colour_manual(values = c("#E41A1C", "#377EB8"), guide=FALSE) +
  labs(fill="Evidence of chimp PRDM9 binding") +
  theme(legend.position = "bottom") +
  ylim(0,6) + xlim(0,6)


#pdf("Zcw_vs_CoTransfection_summary_plot_allchr.pdf", width=12)
plot_grid(p1, p2)
#dev.off()


```


# Relative Strength/Attraction of H3K4me3

```{r H3K4strength}

#pdf("H3K4strength.pdf")

# K4UT vs K4wP9

p <- ggplot(forcecalled[h3k4me3_UT_cov_input>5 & h3k4me3_hP9_cov_input>5], aes(h3k4me3_UT, h3k4me3_hP9_enrichment)) +
  geom_hex(aes(fill = stat(log(count))), bins=200) +
  scale_fill_viridis_c()

# All & Zoom
p
p + xlim(0,30) + ylim(0,30)

# Colour by Zcw
ggplot(forcecalled[h3k4me3_UT_cov_input>5 & h3k4me3_hP9_cov_input>5], aes(h3k4me3_UT, h3k4me3_hP9_enrichment, z=log(zcw_hP9_enrichment+0.1))) +
  stat_summary_hex(bins=200) + theme(legend.position = "bottom") +
  scale_fill_viridis_c(name = "log(Zcw_w/hP9+0.1)", direction = -1) + xlim(0,30) + ylim(0,30)

# Log scale count
ggplot(forcecalled[h3k4me3_UT_cov_input>5 & h3k4me3_hP9_cov_input>5], aes(log(h3k4me3_UT+0.1), log(h3k4me3_hP9_enrichment+0.1))) +
  geom_hex(aes(fill = stat(log(count))), bins=200) +
  scale_fill_viridis_c() + geom_abline(intercept = 0, slope = 1, col='red')

# Log scale colour by Zcw
ggplot(forcecalled[h3k4me3_UT_cov_input>5 & h3k4me3_hP9_cov_input>5], aes(log(h3k4me3_UT+0.1), log(h3k4me3_hP9_enrichment+0.1), z=log(zcw_hP9_enrichment+0.1))) +
  #stat_summary_hex(bins=200) +
  stat_summary_hex(bins=200, fun= function(x)if(length(x)>=3){mean(x)}else{NA}) +
  theme(legend.position = "bottom") +
  scale_fill_viridis_c(name = "log(Zcw_w/hP9+0.1)", direction = -1)


# Zcw colour, as P9 increases, for a given H3K4
ggplot(forcecalled[zcw_cov_input>5 & hP9combo_cov_input>5 & h3k4me3_hP9_cov_input>5], aes(hP9combo_enrichment, h3k4me3_hP9_enrichment, z=log(zcw_hP9_enrichment+0.1))) +
  stat_summary_hex(bins=200) + theme(legend.position = "bottom") +
  scale_fill_viridis_c(name = "log(Zcw_w/hP9+0.1)", direction = -1) + xlim(0,20) + ylim(0,50) +
  ggtitle("zcw cov input")

# min 3
ggplot(forcecalled[zcw_cov_input>5 & hP9combo_cov_input>5 & h3k4me3_hP9_cov_input>5], aes(log(hP9combo_enrichment+0.1), log(h3k4me3_hP9_enrichment+0.1), z=log(zcw_hP9_enrichment+0.1))) +
  stat_summary_hex(bins=200, fun= function(x)if(length(x)>=3){mean(x)}else{NA}) +
  theme(legend.position = "bottom") +
  scale_fill_viridis_c(name = "log(Zcw_w/hP9+0.1)", direction = -1)

#dev.off()
```


# K4 vs Zcw (CT) split by hP9 status

For a given amount of H3K4, Zcw is higher enriched when P9 bound at non promoters than promoters


```{r H3K4strength_allchr, fig.width = 14, fig.height = 14}

forcecalled[,UTP9 := NULL]
forcecalled[hP9combo_cov_input>5 & hP9combo_pvalue==1, UTP9 := "0"]
forcecalled[hP9combo_cov_input>5 & hP9combo_pvalue<1e-7, UTP9 := "1"]
forcecalled[h3k4me3_UT_cov_input>5 & h3k4me3_UT_pvalue<1e-5, UTP9 := "2"] # this means the above also has condition h3k4me3_UT_pvalue>1e-5

#forcecalled[,mean(zcw_hP9_enrichment),by=UTP9]
#      UTP9        V1
# 1:    0 0.1162629
# 2: <NA> 0.3742038
# 3:    1 2.1501786
# 4:    2 1.8304093


mingroup <- min(forcecalled[,.N,by=UTP9]$N)
stopifnot(mingroup>5e3)
#forcecalled[,.SD[sample(.N, min(1e4,.N))], by = UTP9][,.N,by=UTP9]

p1 <- ggplot(forcecalled[zcw_cov_input>5 & h3k4me3_hP9_cov_input>5][,.SD[sample(.N, min(mingroup,.N))], by = UTP9], (aes(log(h3k4me3_hP9_enrichment+0.1), log(zcw_hP9_enrichment+0.1), group=UTP9, z=as.numeric(UTP9))))

p2 <- ggplot(forcecalled[zcw_cov_input>5 & h3k4me3_UT_cov_input>5][,.SD[sample(.N, min(mingroup,.N))], by = UTP9], (aes(log(h3k4me3_UT_enrichment+0.1), log(zcw_enrichment+0.1), group=UTP9, z=as.numeric(UTP9))))

p3 <- ggplot(forcecalled[zcw_cov_input>5 & h3k4me3_hP9_cov_input>5][,.SD[sample(.N, min(mingroup,.N))], by = UTP9], (aes(log(h3k4me3_hP9_enrichment+0.1), log(zcw_enrichment+0.1), group=UTP9, z=as.numeric(UTP9))))

p4 <- ggplot(forcecalled[zcw_cov_input>5 & h3k4me3_UT_cov_input>5][,.SD[sample(.N, min(mingroup,.N))], by = UTP9], (aes(log(h3k4me3_UT_enrichment+0.1), log(zcw_hP9_enrichment+0.1), group=UTP9, z=as.numeric(UTP9)))) #+

commongrobs <- list(
  stat_summary_hex(bins=150, alpha=0.5, fun= function(x) if(length(x)>=5){as.character(mean(x))}else{NA}), # duplicate this line for multi-alpha density shading?
  geom_density_2d(aes(colour=UTP9)),
  theme(legend.position = "bottom"),
  scale_fill_manual(labels = c("Not humanPRDM9 bound (p=1)",
                               "humanPRDM9 bound (p<1e-7) & No Untransfected H3K4me3 (p>1e-5)",
                               "Untransfected H3K4me3 (p<1e-5)"),
                    values = RColorBrewer::brewer.pal(3, "Set1"), 
                    name = "Subset"),
  scale_colour_manual(values = RColorBrewer::brewer.pal(3, "Set1"), guide=FALSE),
  xlim(NA,4),
  ylim(NA,2.5)
)

# Count density
# 
# commongrobsCount <- list(
#   #stat_summary_hex(aes(group=1), bins=200, fun= function(x) if(length(x)>=1){log(length(x))}else{NA}),
#   geom_hex(aes(fill = stat(log(count))), bins=200),
#   scale_fill_viridis_c(direction = -1),
#   theme(legend.position = "bottom"),
#   xlim(NA,4),
#   ylim(NA,2.5)
# )
# 
# p1 + commongrobsCount
# dev.off()

pmain <- plot_grid(p1 + commongrobs + theme(legend.position = "none") + labs(title="Transfected vs Cotransfected",
                                                                             x="ln(0.1 + Enrichment of H3K4me3 in humanPRDM9 transfection)", 
                                                                             y="ln(0.1 + Enrichment of ZCWPW1 in Zcwpw1-humanPRDM9 cotransfection)"),
          p3 + commongrobs + theme(legend.position = "none") + labs(title="Transfected vs Singly-transfected",
                                                                             x="ln(0.1 + Enrichment of H3K4me3 in humanPRDM9 transfection)", 
                                                                             y="ln(0.1 + Enrichment of ZCWPW1 in ZCWPW1 single transfection)"),
          p4 + commongrobs + theme(legend.position = "none") + labs(title="Un-transfected vs Cotransfected",
                                                                             x="ln(0.1 + Enrichment of H3K4me3 in untransfected)", 
                                                                             y="ln(0.1 + Enrichment of ZCWPW1 in ZCWPW1-humanPRDM9 cotransfection)"),
          p2 + commongrobs + theme(legend.position = "none") + labs(title="Un-transfected vs Singly-transfected",
                                                                             x="ln(0.1 + Enrichment of H3K4me3 in untransfected)", 
                                                                             y="ln(0.1 + Enrichment of ZCWPW1 in ZCWPW1 single transfection)"))

#pdf("H3K4strength_allchr.pdf", width = 14, height = 14)
plot_grid(pmain, get_legend(p1 + commongrobs), ncol=1, rel_heights = c(1, 0.05))
#dev.off()

```


## Mean by bin

```{r}

fraction_overlapN <- function(dt,x,y, n=25, filters=c("All"), returndf=F, pseudoenrich=0.01){
  setorder(dt, enrichment)
  dt$enrichment_bin <- cut(1:nrow(dt), n)
  
  All=TRUE
  
  tmp <- dt[eval(parse(text=filters[1])),.(mean_enrichment = mean(enrichment + pseudoenrich),
                                      num = .N,
                                      Mean = mean(get(y)),
                                      Subset = filters[1],
                                      sem = sd(get(y))/sqrt(length(get(y))),
                                      max = max(get(y)),
                                      min = min(get(y)),
                                      uq = quantile(get(y), 0.75),
                                      lq = quantile(get(y), 0.25)
                                      #sep=sqrt((mean(get(y)) * (1- mean(get(y))))/length(get(y)))
                                      ),by=enrichment_bin]

    if(returndf){
    return(tmp)
  }else{
  return(
    ggplot(tmp, aes(mean_enrichment, Mean, colour=Subset, group=Subset)) + #mean_enrichment
      #geom_hline(yintercept = dt[,mean(get(y))], colour="blue", linetype="dashed") +
      geom_point(size=2, stroke=0) +
      geom_linerange(aes(ymax = Mean+2*sem, ymin = Mean-2*sem), colour='black') +
      theme_minimal() +
      scale_color_brewer(palette = 'Set1') +
      xlab(paste("Mean",x,"Enrichment")) +
      ylab(paste("Mean",y))
  )}

}
```


# Histone marks +ve correlate with Zcw enrich

```{r H3Kme3_vs_Zcw_wP9, fig.width=11}

forcecalled$enrichment <- forcecalled$h3k4me3_hP9_enrichment

p1 <- fraction_overlapN(forcecalled[sample(1:nrow(forcecalled))][h3k4me3_hP9_cov_input>15 & zcw_cov_input>5 & h3k4me3_UT_pvalue>1e-5],
                        "h3k4me3_hP9_enrichment","zcw_hP9_enrichment", n=100) +
  scale_y_log10() + scale_x_log10() +
  theme(legend.position = "none") +
  ylab("Mean ZCWPW1 Enrichment when co-transfected with hPRDM9") +
  xlab("Mean H3K4me3 Enrichment when co-transfected with hPRDM9+ 0.01")


forcecalled$enrichment <- forcecalled$h3k36me3_hP9_enrichment

p2 <- fraction_overlapN(forcecalled[sample(1:nrow(forcecalled))][h3k36me3_hP9_cov_input>15 & zcw_cov_input>5],
                        "h3k36me3_hP9_enrichment","zcw_hP9_enrichment", n=100) + 
  scale_y_log10() + scale_x_log10() +
  theme(legend.position = "none") +
  ylab("Mean ZCWPW1 Enrichment when co-transfected with hPRDM9") +
  xlab("Mean H3K36me3 Enrichment when co-transfected with hPRDM9 + 0.01")

#pdf("results/H3Kme3_vs_Zcw_wP9.pdf", width = 11)
plot_grid(p1,p2)
#dev.off()

```


# Binned H3K4 Strength

```{r H3K4strength_binned_allchr_A}

#pdf("H3K4strength_binned_allchr.pdf")
forcecalled$enrichment <- forcecalled$h3k4me3_hP9_enrichment

tmp <- rbind(
  fraction_overlapN(forcecalled[h3k4me3_hP9_cov_input>15 & zcw_cov_input>5][UTP9==1], 
                    "h3k4me3_hP9_enrichment","zcw_hP9_enrichment", n=25, returndf = T, filter="UTP9==1", pseudoenrich = 0.01),
  fraction_overlapN(forcecalled[h3k4me3_hP9_cov_input>15 & zcw_cov_input>5][UTP9==2], 
                    "h3k4me3_hP9_enrichment","zcw_hP9_enrichment", n=25, returndf = T, filter="UTP9==2", pseudoenrich = 0.01)
)


p <- ggplot(tmp, aes(mean_enrichment, Mean, colour=Subset)) +
  geom_point() +
  #scale_x_log10() +
  xlab("Mean H3K4me3 enrichment in human PRDM9 transfection") +
  ylab("Mean ZCWPW1 enrichment in ZCWPW1 human PRDM9 cotransfection") +
  geom_pointrange(aes(ymax = uq, ymin = lq), linetype="dotted") +
  geom_errorbar(aes(ymax = Mean + 2*sem, ymin = Mean - 2*sem), width = 0.05) +
  scale_colour_manual(labels = c(expression("Human-PRDM9 bound (p<10"^"-7"*") & No pre-existing H3K4me3 (p>10"^"-5"*")"),
                               expression("Pre-existing H3K4me3 (p<10"^"-5"*")")),
                      guide=guide_legend(nrow=2),
                    values = RColorBrewer::brewer.pal(3, "Set1")) +
  theme_minimal() +
  theme(legend.position = "bottom")
p
#dev.off()

```

## Gam version

```{r H3K4strength_binned_allchr_B}

labels <- c("humanPRDM9 bound (p<1e-7) & No Untransfected H3K4me3 (p>1e-5)","Untransfected H3K4me3 (p<1e-5)")

forcecalled[,UTP9F := NULL]
forcecalled[hP9combo_cov_input>5 & hP9combo_pvalue<1e-7, UTP9F := labels[1]]
forcecalled[h3k4me3_UT_cov_input>5 & h3k4me3_UT_pvalue<1e-5, UTP9F := labels[2]] # this means the above also has condition h3k4me3_UT_pvalue>1e-5

forcecalled[, UTP9F := factor(UTP9F)]

gam_y <- gam(zcw_hP9_enrichment ~ s(log(0.01+h3k4me3_hP9_enrichment), by=UTP9F) + UTP9F, method = "REML", data = forcecalled[h3k4me3_hP9_cov_input>15 & zcw_cov_input>5][!is.na(UTP9F)]) #[UTP9==1]

# gam_y <- gam(zcw_hP9_enrichment ~ s(h3k4me3_hP9_enrichment, UTP9F, bs='fs'), method = "REML", data = forcecalled[h3k4me3_hP9_cov_input>15 & zcw_cov_input>5][!is.na(UTP9F)]) #[UTP9==1]
summary(gam_y)

n=500
x_new <- seq(0.15, 30, length.out = n)
y_pred <- data.table(h3k4me3_hP9_enrichment=rep(x_new, 2),
                     UTP9F=factor(rep(labels, each=n)),
                     data.frame(predict(gam_y, data.frame(h3k4me3_hP9_enrichment = rep(x_new, 2), UTP9F = factor(rep(labels, each=n))), se.fit=TRUE)))

tmp2 <- y_pred
names(tmp2)[1:3] <- c("mean_enrichment","Subset","Mean")

#pdf("H3K4strength_binned_allchr.pdf")
p + geom_ribbon(aes(ymin = Mean - 2*se.fit, ymax = Mean + 2*se.fit, group=Subset),
                data=tmp2[se.fit<0.05], fill = "grey70", colour="grey70", alpha=0.5) +
  geom_hline(yintercept = 1.95, linetype="dashed") + scale_x_log10() + scale_y_log10()

```


# Prop meth

```{r prop_meth}
forcecalled$enrichment <- forcecalled$prop_meth

tmp <- rbind(
  fraction_overlapN(forcecalled[zcw_cov_input>5 & (meth + unmeth>10)][CpG_Island==0],
                    "prop_meth","zcw_enrichment", n=50, returndf = T, filter="CpG_Island==0", pseudoenrich = 0.01),
  fraction_overlapN(forcecalled[zcw_cov_input>5 & (meth + unmeth>10)][CpG_Island>0], 
                    "prop_meth","zcw_enrichment", n=50, returndf = T, filter="CpG_Island>0", pseudoenrich = 0.01)
)


p <- ggplot(tmp, aes(mean_enrichment, Mean, colour=Subset)) +
  geom_point() +
  #scale_x_log10() +
  xlab("Mean methylation") +
  ylab("Mean Zcwpw1 enrichment") +
  geom_pointrange(aes(ymax = uq, ymin = lq), linetype="dotted") +
  geom_errorbar(aes(ymax = Mean + 2*sem, ymin = Mean - 2*sem), width = 0.05) +
  scale_colour_manual(
                      guide=guide_legend(nrow=2),
                    values = RColorBrewer::brewer.pal(3, "Set1")) +
  theme_minimal() +
  theme(legend.position = "bottom")
p


forcecalled[zcw_cov_input>5 & (meth + unmeth>10)][CpG_Island==0][prop_meth<0.1][order(-zcw_enrichment)]
# 
#  chr1    195096200   195096300 
# 
#  # region for screenshot
#  chr1:42,317,006-42,480,721
#  chr1:53,882,156-53,949,798
# 
#   
```

```{r, eval=FALSE}
############## Create Enrichment BigWig for IGV
con1=gzfile("peaks/enrichments/SingleBaseEnrichment.SingleBasePeaks.WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL.bed.chr1.binary.r",open="rb")
	vec=readBin(con1,what=numeric(1),n=1e9)
	close(con1)
 
	
	starts <- !c(1,diff(vec))==0
	tmp <- data.table("chr1",as.integer((1:length(vec))[starts]), round((vec)[starts], 3))
	tmp$V4 = c(tail(tmp$V2,-1),tail(tmp$V2,1)+1L)
	
	setcolorder(tmp, c("V1","V2","V4","V3"))
	
	fwrite(tmp, "enrich_WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.bedGraph", sep="\t", col.names=F, quote = F)

	bedGraphToBigWig enrich_WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.bedGraph ../single-cell/sequencing/metadata/hg38_sizes.chrom enrich_WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.bigWig

	# result is quite spikey, (from low input)
```
	
######### CpG count stripes

```{r CpG_count_stripes}
forcecalled[zcw_cov_input>5 & (meth + unmeth>10)][, .(mean(CpG), quantile(CpG, 0.25), quantile(CpG, 0.75), quantile(CpG, 0.95)), by=CpG_Island]

forcecalled[zcw_cov_input>5][CpG==0][meth>0]

 

tmp <- rbind(
  fraction_overlapN(forcecalled[zcw_cov_input>5 & (meth + unmeth>10)][CpG_Island==0][CpG==1],
                    "prop_meth","zcw_enrichment", n=25, returndf = T, filter="CpG==1", pseudoenrich = 0.01),
  fraction_overlapN(forcecalled[zcw_cov_input>5 & (meth + unmeth>10)][CpG_Island==0][CpG==2],
                    "prop_meth","zcw_enrichment", n=25, returndf = T, filter="CpG==2", pseudoenrich = 0.01),
  fraction_overlapN(forcecalled[zcw_cov_input>5 & (meth + unmeth>10)][CpG_Island==0][CpG==4],
                    "prop_meth","zcw_enrichment", n=25, returndf = T, filter="CpG==4", pseudoenrich = 0.01),
  fraction_overlapN(forcecalled[zcw_cov_input>5 & (meth + unmeth>10)][CpG_Island==0][CpG==7],
                    "prop_meth","zcw_enrichment", n=25, returndf = T, filter="CpG==7", pseudoenrich = 0.01),
  fraction_overlapN(forcecalled[zcw_cov_input>5 & (meth + unmeth>10)][CpG_Island==1][CpG==7],
                    "prop_meth","zcw_enrichment", n=25, returndf = T, filter="CpG==7 & CpG_Island==1", pseudoenrich = 0.01)
)


p <- ggplot(tmp, aes(mean_enrichment, Mean, colour=Subset)) +
  geom_point() +
  xlab("Fraction methylated reads") +
  ylab("Mean Zcwpw1 enrichment") +
  geom_errorbar(aes(ymax = Mean + 2*sem, ymin = Mean - 2*sem), width = 0.05) +
  scale_colour_manual(guide=guide_legend(nrow=2), values = RColorBrewer::brewer.pal(6, "Set1")) +
  theme_minimal() +
  theme(legend.position = "bottom")
p


```


# Zcw ST vs CT binned

```{r Zcw_ST_vs_CT_binned}
forcecalled$enrichment <- forcecalled$zcw_enrichment

#pdf("Zcw_ST_vs_CT_binned.pdf")

fraction_overlapN(forcecalled[zcw_cov_input>5], 
                    "zcw_enrichment","zcw_hP9_enrichment", n=25)


tmp <- rbind(
  fraction_overlapN(forcecalled[hP9combo_cov_input>5 & zcw_cov_input>5][hP9combo_pvalue==1], 
                    "zcw_enrichment","zcw_hP9_enrichment", n=25, returndf = T, filter="hP9combo_pvalue==1"),
  fraction_overlapN(forcecalled[hP9combo_cov_input>5 & zcw_cov_input>5][hP9combo_pvalue<1e-9], 
                    "zcw_enrichment","zcw_hP9_enrichment", n=25, returndf = T, filter="hP9combo_pvalue<1e-9")
)


ggplot(tmp, aes(mean_enrichment, Mean, colour=Subset)) +
  geom_point() +
  xlab("Mean zcw_enrichment") +
  ylab("Mean zcw_hP9_enrichment") +
  geom_pointrange(aes(ymax = uq, ymin = lq)) +
  geom_errorbar(aes(ymax = Mean + 2*sem, ymin = Mean - 2*sem), width = 0.05) +
  scale_colour_manual(labels = c("hP9combo_pvalue==1",
                               "hP9combo_pvalue<1e-9"),
                    values = RColorBrewer::brewer.pal(3, "Set1")) +
  theme_minimal() +
  theme(legend.position = "bottom")
#dev.off()

```

# Dmc1

```{r}

dmc1 <- fread("../data/pratto_dmc1/Pratto_human_DSB_Aintersect_autosomal_hg38.bed")
setnames(dmc1, c("chr","CI_start","CI_stop","enrichment","enrichmentAB"))
dmc1 <- prejoin(dmc1)
  
```

# Find negative control regions

```{r}
forcecalled$enrichment <- forcecalled$CpG

#pdf("CpG.pdf")
ggplot(forcecalled[zcw_cov_input>5][,.(mean_enrich=mean(zcw_hP9_enrichment),stdv=sd(zcw_hP9_enrichment),.N),by=CpG][order(-CpG)], aes(CpG, mean_enrich)) + geom_pointrange(aes(ymin = mean_enrich - 2*(stdv/sqrt(N)), ymax = mean_enrich + 2*(stdv/sqrt(N))))
#dev.off()
```

```{r, eval=FALSE}
#curl -s "http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/cytoBand.txt.gz" | gunzip -c | grep acen | head

gaps <- fread("curl -s http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/gap.txt.gz | gunzip -c")
setnames(gaps, c("V2","V3","V4","V7","V8"),c("chr","center_start","center_stop","gap_length","gap_type"))

setkey(gaps, chr, center_start, center_stop)
setkey(forcecalled, chr, center_start, center_stop)
forcecalled <- foverlaps(forcecalled, gaps)

plot(forcecalled[is.na(gap_type) & CpG==0 & hP9combo_pvalue==1 & h3k36me3_hP9_pvalue==1 & h3k4me3_hP9_pvalue==1]$i.center_start)
hist(forcecalled[is.na(gap_type) & CpG==0 & hP9combo_pvalue==1 & h3k36me3_hP9_pvalue==1 & h3k4me3_hP9_pvalue==1]$zcw_enrichment, breaks=100)
dev.off()

echo -e "chr1\t134143400\t134143500" > tmp.bed
bedtools getfasta -s -fi motifs/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa -bed tmp.bed -name

forcecalled[is.na(gap_type) & CpG==0 & hP9combo_pvalue==1 & h3k36me3_hP9_pvalue==1 & h3k4me3_hP9_pvalue==1][zcw_enrichment>10]

mean(forcecalled[is.na(gap_type) & CpG==0 & hP9combo_pvalue==1 & h3k36me3_hP9_pvalue==1 & h3k4me3_hP9_pvalue==1]$zcw_enrichment>0.5)
0.04527395

forcecalled[is.na(gap_type) & CpG==0 & hP9combo_pvalue==1 & h3k36me3_hP9_pvalue==1 & h3k4me3_hP9_pvalue==1][sample(1:229000, 5)]



mappability <- fread("../data/mappability/hg38.75mer.Excludable.t0.5.chr1.merge500.bed")
setnames(mappability, c("chr","i.center_start","i.center_stop"))
setkey(mappability, chr, i.center_start, i.center_stop)
mappability$exclude <- T
forcecalled2 <- foverlaps(forcecalled, mappability)[is.na(exclude)]

mean(forcecalled2[is.na(gap_type) & CpG==0 & hP9combo_pvalue==1 & h3k36me3_hP9_pvalue==1 & h3k4me3_hP9_pvalue==1]$zcw_enrichment>0.5)

forcecalled2[is.na(gap_type) & CpG==0 & hP9combo_pvalue==1 & h3k36me3_hP9_pvalue==1 & h3k4me3_hP9_pvalue==1][sample(1:168000, 10)]

177853400       177853500  
```


# K36wP9 vs K4wP9

```{r k36_vs_k4, eval=FALSE}
# need zcw_P9_pvalue before changing eval to TRUE
p <- ggplot(forcecalled[h3k36me3_hP9_enrichment>5], aes(h3k36me3_hP9_enrichment, h3k4me3_hP9_enrichment)) +
  geom_hex(aes(fill = stat(log(count))), bins=200) +
  scale_fill_viridis_c()

p + facet_wrap(~ zcw_P9_pvalue<1e-6 ) + ggtitle("Split by zcw_P9_pvalue<1e-6")


# K4UT vs Zcw
ggplot(forcecalled[h3k4me3_UT>5], aes(h3k4me3_UT, zcw_enrichment)) +
  geom_hex(aes(fill = stat(log(count))), bins=200) +
  scale_fill_viridis_c() + 
  ylim(0,10) + xlim(0,100)

# K4UT vs Zcw w/ hP9
ggplot(forcecalled[h3k4me3_UT>5], aes(h3k4me3_UT, zcw_hP9_enrichment)) +
  geom_hex(aes(fill = stat(log(count))), bins=200) +
  scale_fill_viridis_c() + 
  ylim(0,10) + xlim(0,100)

# K4UT vs P9
p <- ggplot(forcecalled[h3k4me3_UT>5], aes(h3k4me3_UT, hP9combo_enrichment)) +
  geom_hex(aes(fill = stat(log(count))), bins=200) +
  scale_fill_viridis_c() + xlim(0,100)

p + facet_wrap(~ zcw_P9_pvalue<1e-6 ) + ggtitle("Split by zcw_P9_pvalue<1e-6") + 
  ylim(0,15)

p + facet_wrap(~ zcw_enrichment>1 ) + ggtitle("Split by zcw_enrichment>1") + 
  ylim(0,15)


```

