---
title: "Peak Calling"
output: 
  html_notebook: 
    code_folding: show
    fig_height: 6
    fig_width: 10.5
    results: hold
    fig_show: hold
    toc: yes
    toc_float: yes
    number_sections: yes
    theme: spacelab
---

# Download results

```{r}

# peak bed files
rsync --stats -avpn -e ssh stats2:/data/saxony/wells/zcwpw1/peaks/*Peaks* ~/Dropbox/DPhil/Zcwpw1/results/peaks
rsync --stats -avpn -e ssh stats2:/data/saxony/wells/zcwpw1/peaks/SingleBasePeaks* ~/Dropbox/DPhil/Zcwpw1/results/peaks

# plots
rsync --stats -avpn --delete -e ssh stats2:/data/saxony/wells/zcwpw1/deeptools/*.pdf ~/Dropbox/DPhil/Zcwpw1/results/deeptools

# bigwigs for IGV
rsync --stats -avpn -e ssh stats2:/data/saxony/wells/zcwpw1/bedgraphs/*.bigWig ~/Downloads/BigWigs/

# log2 bigwigs for IGV
rsync --stats -avpn -e ssh stats2:/data/saxony/wells/zcwpw1/deeptools/bigwigs/*log2.bw ~/Downloads/BigWigs/log2/
  
# single base pair resolution bigwigs for IGV
rsync --stats -avpn -e ssh stats2:/data/saxony/wells/zcwpw1/deeptools/bigwigs/SBPsample/*log2.bw ~/Downloads/BigWigs/log2/SBPsample/

# format for IGV
for file in SingleBasePeaks*
do
tail -n +2 $file | awk -v OFS='\t' '{print $1, $4, $5, -log($11)/log(10)}' > igvbeds/$file
done

```

```{r}
library(ggplot2)
library(data.table)
```


# Load Results

```{r}

# HA TAG UNLESS OTHERWISE SPECIFIED

zcw_peaks <- fread("../data/peaks/SingleBasePeaks.WTCHG_538916_221156_vs_WTCHG_538916_217108.p0.000001.sep250.ALL.bed")

zcw_peaks_cin <- fread("../data/peaks/SingleBasePeaks.WTCHG_538916_221156_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL.bed")


zcw_wP9_peaks <- fread("../data/peaks/SingleBasePeaks.WTCHG_538916_223180_vs_WTCHG_538916_220144.p0.000001.sep250.ALL.bed")
zcw_wP9_peaks_cin <- fread("../data/peaks/SingleBasePeaks.WTCHG_538916_223180_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL.bed")

zcw_wP9_peaks_cin_QCM50 <- fread("../data/peaks/SingleBasePeaks.WTCHG_538916_223180_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL_QCMappability50.bed")

zcw_wP9_vs_chip_Z <- fread("../data/peaks/SingleBasePeaks.WTCHG_538916_223180_vs_WTCHG_538916_221156.p0.000001.sep250.ALL.bed")

zcw_wcP9_peaks_cin <- fread("../data/peaks/SingleBasePeaks.WTCHG_538916_224192_vs_WTCHG_538916_217108_AND_WTCHG_538916_220144.p0.000001.sep250.ALL.bed")
zcw_wcP9_vs_chip_Z <- fread("../data/peaks/SingleBasePeaks.WTCHG_538916_224192_vs_WTCHG_538916_221156.p0.000001.sep250.ALL.bed")


HP9_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627146_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL.bed")
HP9V5_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627147_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL.bed")

HP9combo_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627146_AND_SRA_Altemose2015_SRR5627147_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL.bed")


CP9_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627144_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL.bed")

CP9V5_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627145_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL.bed")

CP9combo_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627145_AND_SRA_Altemose2015_SRR5627144_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL.bed")


HP9N_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627138_AND_SRR5627139_vs_SRA_Altemose2015_SRR5627140.bed")

h3k4me3_P9_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627152_AND_SRR5627153_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL.bed")
h3k36me3_P9_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627149_vs_SRA_Altemose2015_SRR5627143.p0.000001.sep250.ALL.bed")

h3k4me3_UT_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627150_vs_SRA_Altemose2015_SRR5627142.p0.000001.sep250.ALL.bed")
h3k36me3_UT_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627148_vs_SRA_Altemose2015_SRR5627142.p0.000001.sep250.ALL.bed")

h3k4me3_P9vsUT_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627152_AND_SRA_Altemose2015_SRR5627153_vs_SRA_Altemose2015_SRR5627150_AND_SRA_Altemose2015_SRR5627151.p0.000001.sep250.ALL.bed")
h3k36me3_P9vsUT_peaks <- fread("../data/peaks/SingleBasePeaks.SRA_Altemose2015_SRR5627149_vs_SRA_Altemose2015_SRR5627148.p0.000001.sep250.ALL.bed")
```


# Mark TSS

```{r}
load_tss <- function(file="../data/hg38.wgEncodeGencodeCompV28.bed", flank=400){
encode_tss <- fread(file)
names(encode_tss) <- gsub("hg38.wgEncodeGencodeCompV28.|hg38.wgEncodeGencodeAttrsV28.","",names(encode_tss))

encode_tss[strand=='+', CI_start_ext := txStart - flank]
encode_tss[strand=='+', CI_stop_ext := txStart + flank]
encode_tss[strand=='-', CI_start_ext := txEnd - flank]
encode_tss[strand=='-', CI_stop_ext := txEnd + flank]
setnames(encode_tss,'chrom','chr')
setnames(encode_tss,'#name','tss')

encode_tss <- encode_tss[,.(chr, CI_start_ext, CI_stop_ext,tss)]
setkey(encode_tss, chr, CI_start_ext, CI_stop_ext)
return(encode_tss)
}

encode_tss <- load_tss(flank=500)

HP9combo_peaks$tss <- foverlaps(HP9combo_peaks, encode_tss, mult = 'first')$tss
HP9N_peaks$tss <- foverlaps(HP9N_peaks, encode_tss, mult = 'first')$tss

CP9combo_peaks$tss <- foverlaps(CP9combo_peaks, encode_tss, mult = 'first')$tss


zcw_wP9_vs_chip_Z$tss <- foverlaps(zcw_wP9_vs_chip_Z, encode_tss, mult = 'first')$tss
zcw_wP9_peaks_cin$tss <- foverlaps(zcw_wP9_peaks_cin, encode_tss, mult = 'first')$tss
zcw_peaks$tss <- foverlaps(zcw_peaks, encode_tss, mult = 'first')$tss

```

# Export beds

```{r}
fwrite(zcw_wP9_vs_chip_Z[cov_input>5][order(enrichment)][,.(chr, center_start, center_stop, enrichment)], "../data/zcw_wP9_vs_chip_Z.bed", col.names = FALSE, sep = "\t")


fwrite(HP9combo_peaks[cov_input>5][!is.na(tss)][order(enrichment)][,.(chr, center_start, center_stop, enrichment)], "../data/HP9combo_peaks_Promoters.bed", col.names = FALSE, sep = "\t")
fwrite(HP9combo_peaks[cov_input>5][is.na(tss)][order(enrichment)][,.(chr, center_start, center_stop, enrichment)], "../data/HP9combo_peaks_NonPromoters.bed", col.names = FALSE, sep = "\t")

fwrite(CP9combo_peaks[cov_input>5][!is.na(tss)][order(enrichment)][,.(chr, center_start, center_stop, enrichment)], "../data/CP9combo_peaks_Promoters.bed", col.names = FALSE, sep = "\t")
fwrite(CP9combo_peaks[cov_input>5][is.na(tss)][order(enrichment)][,.(chr, center_start, center_stop, enrichment)], "../data/CP9combo_peaks_NonPromoters.bed", col.names = FALSE, sep = "\t")

```

## Number of peaks

```{r}

number_of_peaks=c(
nrow(zcw_peaks[cov_input>5]),
nrow(zcw_wP9_peaks[cov_input>5]),
nrow(zcw_wP9_peaks_cin[cov_input>5]),
nrow(zcw_wP9_vs_chip_Z[cov_input>5]),
nrow(HP9_peaks[cov_input>5]),
nrow(HP9V5_peaks[cov_input>5]),
nrow(HP9combo_peaks[cov_input>5]),
nrow(HP9N_peaks[cov_input>5]),
nrow(CP9_peaks[cov_input>5]),
nrow(h3k4me3_P9_peaks[cov_input>5]),
nrow(h3k36me3_P9_peaks[cov_input>5]),
nrow(h3k4me3_UT_peaks[cov_input>5]),
nrow(h3k36me3_UT_peaks[cov_input>5]))

experiment=c('Zcw',
            'Zcw_wP9',
            'Zcw_wP9_cinput',
            'zcw_wP9_vs_chip_Z',
            'HP9',
            'HP9V5',
            'HP9V5&HA',
            'HP9-Nterm',
            'CP9',
            'h3k4me3_P9',
            'h3k36me3_P9',
            'h3k4me3_UT',
            'h3k36me3_UT')

ggplot(data.table(number_of_peaks, experiment), aes(experiment, number_of_peaks)) + geom_col() + coord_flip()

number_of_peaks=c(
nrow(zcw_peaks),
nrow(zcw_wP9_peaks),
nrow(zcw_wP9_peaks_cin),
nrow(zcw_wP9_vs_chip_Z),
nrow(HP9_peaks),
nrow(HP9V5_peaks),
nrow(HP9combo_peaks),
nrow(HP9N_peaks),
nrow(CP9_peaks),
nrow(h3k4me3_P9_peaks),
nrow(h3k36me3_P9_peaks),
nrow(h3k4me3_UT_peaks),
nrow(h3k36me3_UT_peaks))

ggplot(data.table(number_of_peaks, experiment), aes(experiment, number_of_peaks)) + geom_col() + coord_flip()
```

```{r}
ggplot(cbind(zcw_peaks, exp='Zcw'), aes(enrichment))+
  geom_density() +
  scale_x_log10() +
  geom_density(data=cbind(zcw_wP9_peaks[cov_input>5], exp='zcw_wP9')) +
  geom_density(data=cbind(HP9_peaks[cov_input>5], exp='hPrdm9')) +
  geom_density(data=cbind(HP9V5_peaks[cov_input>5], exp='hPrdm9V5')) +
  geom_density(data=cbind(HP9combo_peaks[cov_input>5], exp='hPrdm9V5&HA')) +
  geom_density(data=cbind(HP9N_peaks[cov_input>5], exp='hPrdm9-Nterm')) +
  geom_density(data=cbind(h3k4me3_UT_peaks[cov_input>5], exp='h3k4me3_UT')) +
  geom_density(data=cbind(h3k4me3_P9_peaks[cov_input>5], exp='h3k4me3_P9')) +
  geom_density(data=cbind(h3k36me3_UT_peaks[cov_input>5], exp='h3k36me3_UT')) +
  geom_density(data=cbind(h3k36me3_P9_peaks[cov_input>5], exp='h3k36me3_P9')) + 
  scale_color_brewer(type="qual", palette = "Set1") + facet_wrap("~exp")
```

```{r}

ggplot(cbind(zcw_peaks, exp='Zcw'), aes(cov_input))+
    geom_histogram(bins=100) +
    xlim(1,25) +
    geom_histogram(bins=100, data=cbind(zcw_wP9_peaks[cov_input>=0], exp='zcw_wP9')) +
  geom_histogram(bins=100, data=cbind(zcw_wP9_peaks_cin[cov_input>=0], exp='zcw_wP9_cin')) +
    geom_histogram(bins=100, data=cbind(HP9_peaks[cov_input>=0], exp='hPrdm9')) +
    geom_histogram(bins=100, data=cbind(HP9V5_peaks[cov_input>=0], exp='hPrdm9V5')) +
    geom_histogram(bins=100, data=cbind(HP9combo_peaks[cov_input>=0], exp='hPrdm9V5&HA')) +
    geom_histogram(bins=100, data=cbind(HP9N_peaks[cov_input>=0], exp='hPrdm9-Nterm')) +
    geom_histogram(bins=100, data=cbind(h3k4me3_UT_peaks[cov_input>=0], exp='h3k4me3_UT')) +
    geom_histogram(bins=100, data=cbind(h3k4me3_P9_peaks[cov_input>=0], exp='h3k4me3_P9')) +
    geom_histogram(bins=100, data=cbind(h3k36me3_UT_peaks[cov_input>=0], exp='h3k36me3_UT')) +
    geom_histogram(bins=100, data=cbind(h3k36me3_P9_peaks[cov_input>=0], exp='h3k36me3_P9')) + 
    scale_color_brewer(type="qual", palette = "Set1") + facet_wrap("~exp", scale="free_y")
```

```{r}
ggplot(cbind(zcw_peaks, exp='Zcw'), aes(cov_r1))+
    geom_histogram(binwidth = 1) +
    xlim(1,70) +
    geom_histogram(binwidth = 1, data=cbind(zcw_wP9_peaks[cov_input>=0], exp='zcw_wP9')) +
    geom_histogram(binwidth = 1, data=cbind(HP9_peaks[cov_input>=0], exp='hPrdm9')) +
    geom_histogram(binwidth = 1, data=cbind(HP9V5_peaks[cov_input>=0], exp='hPrdm9V5')) +
    geom_histogram(binwidth = 1, data=cbind(HP9combo_peaks[cov_input>=0], exp='hPrdm9V5&HA')) +
    geom_histogram(binwidth = 1, data=cbind(HP9N_peaks[cov_input>=0], exp='hPrdm9-Nterm')) +
    geom_histogram(binwidth = 1, data=cbind(h3k4me3_UT_peaks[cov_input>=0], exp='h3k4me3_UT')) +
    geom_histogram(binwidth = 1, data=cbind(h3k4me3_P9_peaks[cov_input>=0], exp='h3k4me3_P9')) +
    geom_histogram(binwidth = 1, data=cbind(h3k36me3_UT_peaks[cov_input>=0], exp='h3k36me3_UT')) +
    geom_histogram(binwidth = 1, data=cbind(h3k36me3_P9_peaks[cov_input>=0], exp='h3k36me3_P9')) + 
    scale_color_brewer(type="qual", palette = "Set1") + facet_wrap("~exp", scale="free_y")

```



# Functions

```{r}

prejoin <- function(dt, ext=100, width=NULL){
  if(is.null(width)){ # extend CI ends by fixed amount
    dt[,CI_start_ext := CI_start -ext]
    dt[,CI_stop_ext := CI_stop +ext]
    
  }else{ # set total peak width as set value around center
    dt[,CI_start_ext := center_start - width/2]
    dt[,CI_stop_ext := center_stop + width/2]
  }
  setkey(dt, chr, CI_start_ext, CI_stop_ext)
  return(dt)
}

trim_peaks <- function(dt, max_len=1000){
  dt[,width := CI_stop-CI_start]
  dt[,CI_start_ext := as.integer(CI_start_ext)]
  dt[,CI_stop_ext := as.integer(CI_stop_ext)]
  dt[,CI_start_ext := as.integer(CI_start + max(0,(width/2)-(max_len/2))), by=c("chr", "CI_start", "CI_stop")]
  dt[,CI_stop_ext := as.integer(CI_stop - max(0,(width/2)-(max_len/2))), by=c("chr", "CI_start", "CI_stop")]
  setkey(dt, chr, CI_start_ext, CI_stop_ext)
  return(dt)
}


fraction_overlapN_old <- function(dt,x,y,n=100){
  setorder(dt, enrichment)
  binning_vector <- cut(1:nrow(dt), n)
  plot(1:n/n, tapply(!is.na(dt[,get(y)]), binning_vector, mean),
       xlab=paste("Fraction of",x,"Peaks  (ordered low to high enrichment)"),
       ylab=paste("Fraction of",x,"peaks that overlap",y,"peaks"))  
}



fraction_overlapN <- function(dt,x,y, n=25, filters=c("All"), returndf=F){
  setorder(dt, enrichment)
  dt$enrichment_bin <- cut(1:nrow(dt), n)
  
  All=TRUE
  
  tmp <- dt[eval(parse(text=filters[1])),.(mean_enrichment=mean(enrichment),
                                      num=.N,
                                      Overlap=mean(!is.na(get(y))),
                                      Subset=filters[1],
                                      #sem=sd(!is.na(get(y)))/sqrt(length(get(y))),
                                      sep=sqrt((mean(!is.na(get(y))) * (1- mean(!is.na(get(y)))))/length(get(y)))),by=enrichment_bin]

  #dt[,mean(!is.na(get(y)))]
  
  if(returndf){
    return(tmp)
  }else{
  return(
    ggplot(tmp, aes(mean_enrichment, Overlap, colour=Subset, group=Subset)) +
      #geom_point(size=2) +
      geom_hline(yintercept = dt[,mean(!is.na(get(y)))], colour="blue", linetype="dashed") +
      geom_pointrange(aes(ymax = Overlap + 2*sep, ymin = Overlap - 2*sep)) +
      theme_minimal() +
      scale_color_brewer(palette = 'Set1') +
      xlab(paste("Mean",x,"Enrichment")) +
      scale_x_log10() +
      ylab(paste("Fraction of ",x,"peaks\n that overlap",y,"peaks"))
  )}
}


randomise_positions <- function(base, shift){
  base_chr_lengths <- base[,.("chr_len"=max(CI_stop_ext)),by=chr]
  setkey(base_chr_lengths, chr)
  
  base_shifted <- base[base_chr_lengths, on="chr"]
  
  base_shifted[,max_right_shift := min(chr_len,CI_stop_ext+shift) - CI_stop_ext, by=c("chr", "CI_start_ext", "CI_stop_ext")]
  base_shifted[,max_left_shift := CI_start_ext - max(0,CI_start_ext-shift), by=c("chr", "CI_start_ext", "CI_stop_ext")]
  
  set.seed(42)
  #shifts <- sample(-shift:shift, nrow(base_shifted), replace = T)
  base_shifted[,shift := sample(-max_left_shift:max_right_shift,1), by=c("chr", "CI_start_ext", "CI_stop_ext")]
  
  base_shifted[,CI_start_ext := CI_start_ext + shift]
  base_shifted[,CI_stop_ext := CI_stop_ext + shift]
  setkey(base_shifted, chr, CI_start_ext, CI_stop_ext)
  
  return(base_shifted)
}

fraction_overlapN2 <- function(x,y, x1, y1, n=25, shift=1e5, filters=c("All"), returndf=F, corrected=F){
  
  base <- x[,.(chr, CI_start_ext, CI_stop_ext, enrichment)]
  test <- y[,.(chr, CI_start_ext, CI_stop_ext)] # enrichment

  setkey(base, chr, CI_start_ext, CI_stop_ext)
  setkey(test, chr, CI_start_ext, CI_stop_ext)
  
  base_shifted <- randomise_positions(base, shift)
  
  base$test_enrichment <- foverlaps(base, test, mult="first")$CI_start_ext # enrichment
  base_shifted$test_enrichment <- foverlaps(base_shifted, test, mult="first")$CI_start_ext # enrichment
  
  a <- fraction_overlapN(base, "base", "test_enrichment", n=n, returndf = T)
  b <- fraction_overlapN(base_shifted, "base_shifted", "test_enrichment",n=n, returndf = T)
  
  a$type <- "Normal"
  b$type <- "Randomised"
  
  tmp <- rbind(a,b)
  
  if(corrected){
    tmp <- dcast(tmp, enrichment_bin + mean_enrichment + num + Subset ~ type, value.var = "Overlap")
    tmp[,Overlap := (1 - ( (1-Normal) / (1-Randomised) ))]
    tmp[,type := "Corrected"]
    tmp[,sep := sqrt((Overlap * (1- Overlap))/num), by=enrichment_bin]
  }
  
  
  if(returndf){
    return(tmp)
  }else{
  
    p <- ggplot(tmp, aes(mean_enrichment, Overlap, colour=type, group=type))
      
    if(!corrected){
      p <- p + geom_hline(yintercept = base[,mean(!is.na(test_enrichment))], colour="red", linetype="dashed") +
      geom_hline(yintercept = base_shifted[,mean(!is.na(test_enrichment))], colour="blue", linetype="dashed")
    }
    
    return(p + 
             geom_pointrange(aes(ymax = Overlap + 2*sep, ymin = Overlap - 2*sep)) +
      theme_minimal() +
      scale_color_brewer(palette = 'Set1') +
      xlab(paste("Mean",x1,"Enrichment")) +
      scale_x_log10() +
      ylab(paste("Fraction of ",x1,"peaks\n that overlap",y1,"peaks"))
  )}
}



fraction_overlapP <- function(dt,x,y,n=100){
  setorder(dt, -pvalue)
  binning_vector <- cut(1:nrow(dt), n)
  plot(1:n/n, tapply(!is.na(dt[,get(y)]), binning_vector, mean),
       xlab=paste("Fraction of",x,"Peaks  (ordered large to small pvalue)"),
       ylab=paste("Fraction of",x,"peaks that overlap",y,"peaks"))  
}


HP9_peaks <- prejoin(HP9_peaks)
HP9N_peaks <- prejoin(HP9N_peaks)
HP9V5_peaks <- prejoin(HP9V5_peaks)
CP9V5_peaks <- prejoin(CP9V5_peaks)
HP9combo_peaks <- prejoin(HP9combo_peaks)
CP9combo_peaks <- prejoin(CP9combo_peaks)
zcw_peaks <- prejoin(zcw_peaks)
zcw_wP9_peaks <- prejoin(zcw_wP9_peaks)
zcw_wP9_peaks_cin <- prejoin(zcw_wP9_peaks_cin)
zcw_wP9_vs_chip_Z <- prejoin(zcw_wP9_vs_chip_Z)


h3k4me3_P9_peaks <- prejoin(h3k4me3_P9_peaks)
h3k36me3_P9_peaks <- prejoin(h3k36me3_P9_peaks)
h3k4me3_UT_peaks <- prejoin(h3k4me3_UT_peaks)
h3k36me3_UT_peaks <- prejoin(h3k36me3_UT_peaks)
```

# DMC1 peaks

```{r}
wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE59nnn/GSE59836/suppl/GSE59836%5FPeak%5Fdata%5FSupplementary%5FFile%5F1%2Etxt%2Egz
gunzip GSE59836_Peak_data_Supplementary_File_1.txt.gz

# subset to only AorB allele peaks observed in all samples & compute average strength
awk -v OFS='\t' '{ if ($16 == 1 && ($3-$2)<3000 && $1!="chrY" && $1!="chrX") { print $1, $2, $3, ($4+$5)/2, ($6+$7)/4 } }' GSE59836_Peak_data_Supplementary_File_1.txt \ > Pratto_human_DSB_Aintersect_autosomal_hg19.bed
# 18342
# 18326
```

```{r}
dmc1 <- fread("~/Downloads/Pratto_human_DSB_Aintersect_autosomal_hg38.bed")
setnames(dmc1, c("chr","CI_start","CI_stop","enrichment","enrichmentAB"))
dmc1 <- prejoin(dmc1)

# awk -v OFS='\t' '{ if ($11 == 1 && ($3-$2)<3000 && $1!="chrY" && $1!="chrX") { print $1, $2, $3, ($4+$5)/2, ($6+$7)/4 } }' GSE59836_Peak_data_Supplementary_File_1.txt \ > Pratto_human_DSB_AB_autosomal_hg19.bed
# wc -l Pratto_human_DSB_AB_autosomal_hg19.bed
# #20336
# 
# awk -v OFS='\t' '{ if ($9 == 1 && $10 == 1 && ($3-$2)<3000 && $1!="chrY" && $1!="chrX") { print $1, $2, $3, ($4+$5)/2, ($6+$7)/4 } }' GSE59836_Peak_data_Supplementary_File_1.txt \ > Pratto_human_DSB_AA_autosomal_hg19.bed
# #21699
# 
# dmc1_AB <- fread("~/Downloads/Pratto_human_DSB_AB_autosomal_hg38.bed")
# setnames(dmc1_AB, c("chr","CI_start_ext","CI_stop_ext","enrichment","enrichmentAB"))
# setkey(dmc1_AB, chr, CI_start_ext, CI_stop_ext)
# 
# dmc1_AA <- fread("~/Downloads/Pratto_human_DSB_AA_autosomal_hg38.bed")
# setnames(dmc1_AA, c("chr","CI_start_ext","CI_stop_ext","enrichment","enrichmentAB"))
# setkey(dmc1_AA, chr, CI_start_ext, CI_stop_ext)
```


```{r}
# library(GenomicRanges)
# 
# HP9_peaks_G <- makeGRangesFromDataFrame(HP9_peaks, start.field = "CI_start", end.field = "CI_stop", keep.extra.columns = T)
# zcw_wP9_vs_chip_Z_G <- makeGRangesFromDataFrame(zcw_wP9_vs_chip_Z, start.field = "CI_start", end.field = "CI_stop", keep.extra.columns = T)
# 
# queryHits(findOverlaps(HP9_peaks_G, zcw_wP9_vs_chip_Z_G, select = "all"))
```

```{r}

nrow(HP9combo_peaks)
sum(!is.na(HP9combo_peaks$DMC1_enrichment))
sum(!is.na(HP9combo_peaks$zcw_CVC_enrichment))
sum(!is.na(HP9combo_peaks$zcw_wP9_enrichment))
sum(!is.na(HP9combo_peaks$zcw_enrichment))

print("HP9N")
nrow(HP9N_peaks)
sum(!is.na(HP9N_peaks$DMC1_enrichment))
sum(!is.na(HP9N_peaks$zcw_CVC_enrichment))
sum(!is.na(HP9N_peaks$zcw_wP9_enrichment))
sum(!is.na(HP9N_peaks$zcw_enrichment))

print("zcw_CVC")
nrow(zcw_wP9_vs_chip_Z)
sum(!is.na(zcw_wP9_vs_chip_Z$DMC1_enrichment))
sum(!is.na(zcw_wP9_vs_chip_Z$HP9N_enrichment))
sum(!is.na(zcw_wP9_vs_chip_Z$HP9combo_enrichment))

print("DMC1")
nrow(dmc1)
sum(!is.na(dmc1$HP9combo_peaks_enrichment))
sum(!is.na(dmc1$HP9N_peaks_enrichment))
sum(!is.na(dmc1$zcw_enrichment))
sum(!is.na(dmc1$zcw_wP9_enrichment))
sum(!is.na(dmc1$Zcw_CVC_enrichment))

```


# HP9HA_C vs HP9V5_C

HA C is a subset of V5 C

```{r}
fraction_overlapN2(HP9V5_peaks[cov_input>6], HP9_peaks, "hPrdm9V5", "hPrdm9HA", shift=1e5) 
fraction_overlapN2(HP9_peaks[cov_input>6], HP9V5_peaks, "hPrdm9HA", "hPrdm9V5", shift=1e5)

HP9V5_peaks$HP9_enrichment <- foverlaps(HP9V5_peaks, HP9_peaks, mult="first")$enrichment
plot(HP9V5_peaks[cov_input>5]$enrichment, HP9V5_peaks[cov_input>5]$HP9_enrichment, log='xy', pch='.')
cor(HP9V5_peaks[cov_input>5][!is.na(HP9_enrichment)]$enrichment, HP9V5_peaks[cov_input>5][!is.na(HP9_enrichment)]$HP9_enrichment)
```

## Cterm Combo vs V5, HA

```{r}
fraction_overlapN2(HP9combo_peaks[cov_input>5], HP9V5_peaks,  "hPrdm9 Cterm combo", "hPrdm9 Cterm V5", shift=1e5)
fraction_overlapN2(HP9combo_peaks[cov_input>5], HP9_peaks,  "hPrdm9 Cterm combo", "hPrdm9 Cterm HA", shift=1e5)
```


# P9 N vs C term

~ 50% overlap in both directions

```{r}
fraction_overlapN2(HP9N_peaks[cov_input>5], HP9combo_peaks, "hPrdm9 Nterm", "hPrdm9 Cterm", shift=1e5)
fraction_overlapN2(HP9combo_peaks[cov_input>5], HP9N_peaks,  "hPrdm9 Cterm", "hPrdm9 Nterm", shift=1e5)

# HP9N_peaks$HP9_enrichment <- foverlaps(HP9N_peaks, HP9_peaks, mult="first")$enrichment
# HP9N_peaks$HP9combo_enrichment <- foverlaps(HP9N_peaks, HP9combo_peaks, mult="first")$enrichment
# HP9combo_peaks$HP9N_enrichment <- foverlaps(HP9combo_peaks, HP9N_peaks, mult="first")$enrichment
# 
# plot(HP9N_peaks[cov_input>5]$enrichment, HP9N_peaks[cov_input>5]$HP9_enrichment, log='xy', pch='.')
# plot(HP9N_peaks[cov_input>5]$enrichment, HP9N_peaks[cov_input>5]$HP9combo_enrichment, log='xy', pch='.')
# cor(HP9N_peaks[cov_input>5][!is.na(HP9_enrichment)]$enrichment, HP9N_peaks[cov_input>5][!is.na(HP9_enrichment)]$HP9_enrichment)
```

```{r}
UpSetR::upset(HP9N_peaks[,.("Nterm P9"=!is.na(chr),
                            "Cterm HA & V5"=!is.na(HP9combo_enrichment),
                            "C term HA"=!is.na(HP9_enrichment),
                            "P9 H3K4me3"=!is.na(h3k4me3_P9_enrichment),
                            "UT H3K4me3"=!is.na(h3k4me3_UT_enrichment),
                            "Zcw CVC"=!is.na(zcw_CVC_enrichment),
                            "Zcw wP9_cin"=!is.na(zcw_wP9_cin_enrichment),
                            "Zcw"=!is.na(zcw_enrichment))]*1,
              nsets = 8)

UpSetR::upset(dmc1[,.("Nterm P9"=!is.na(HP9N_peaks_enrichment),
                            "Cterm HA & V5"=!is.na(HP9combo_peaks_enrichment),
                            #"Zcw CVC"=!is.na(zcw_CVC_enrichment),
                            "Zcw wP9_cin"=!is.na(zcw_wP9_cin_enrichment),
                            "Zcw"=!is.na(zcw_enrichment))]*1,
              nsets = 4)
```


## Visualise Similar N & C term peaks

```{r}
HP9N_peaks[,NC_similar := (HP9combo_enrichment/enrichment)>exp(0.2*enrichment^1.5) | HP9combo_enrichment/enrichment<exp(-0.2*HP9combo_enrichment^1.5)]

ggplot(HP9N_peaks[cov_input>5], aes(enrichment, HP9combo_enrichment,
                                    colour=NC_similar)) +
  geom_point(pch='.', size=0.1) +
  scale_x_log10() + scale_y_log10() +
  theme(legend.position = 'bottom') + 
  geom_abline(intercept = 0,slope = 1)
```


```{r}
HP9N_peaks[,NC_similar := (FC_hP9combo_enrichment/enrichment)>exp(0.2*enrichment^1.5) | FC_hP9combo_enrichment/enrichment<exp(-0.2*FC_hP9combo_enrichment^1.5)]

ggplot(HP9N_peaks[cov_input>5], aes(enrichment, FC_hP9combo_enrichment,
                                    colour=NC_similar)) +
  geom_point(pch='.', size=0.1) +
  scale_x_log10() + scale_y_log10() +
  theme(legend.position = 'bottom') + 
  geom_abline(intercept = 0,slope = 1)
```

```{r}
rr=rank(HP9N_peaks$enrichment)
rr2=rank(HP9N_peaks$FC_hP9combo_enrichment)
cor(rr,rr2)
# [1] 0.3245829
rr=rr/length(rr)
rr2=rr2/length(rr2)

similar_rank=abs(rr-rr2)<0.1
mean(similar_rank)

quantile(abs(rr-rr2),1/3)
#33.33333% 
#0.1302969 
similar_rank=abs(rr-rr2)<0.1303

similar_rank=abs(rr-rr2)<0.05

ggplot(HP9N_peaks[cov_input>5], aes(enrichment, FC_hP9combo_enrichment, colour=similar_rank[HP9N_peaks$cov_input>5])) +
  geom_point(pch='.', size=0.1) +
  scale_x_log10() + scale_y_log10() +
  theme(legend.position = 'bottom') + 
  geom_abline(intercept = 0,slope = 1)

```


```{r}
h3k4me3_UT_peaks$HP9N_enrichment <- foverlaps(h3k4me3_UT_peaks, HP9N_peaks, mult="first")$enrichment
h3k4me3_P9_peaks$HP9N_enrichment <- foverlaps(h3k4me3_P9_peaks, HP9N_peaks, mult="first")$enrichment
h3k4me3_P9_peaks$HP9N_cov_input <- foverlaps(h3k4me3_P9_peaks, HP9N_peaks, mult="first")$cov_input

fraction_overlapN(h3k4me3_UT_peaks[cov_input>=5], "h3k4me3_UT_peaks", "HP9N_enrichment")
fraction_overlapN(h3k4me3_P9_peaks[cov_input>=5], "h3k4me3_P9_peaks", "HP9N_enrichment")

```


# HP9 vs H3K4

Higher hP9combo -> higher overlap with P9 H3K4me3

Higher hP9combo -> lower overlap with UT H3K4me3,
  But still higher than background overlap, so even strong C term Prdm9 locations are near promoters (matches with GC rich regions).


```{r}
fraction_overlapN2(HP9combo_peaks[cov_input>=5], h3k4me3_UT_peaks, x1="HP9combo", y1="h3k4me3_UT")
fraction_overlapN2(HP9combo_peaks[cov_input>=5][is.na(tss)], h3k4me3_UT_peaks, x1="HP9combo", y1="h3k4me3_UT")
fraction_overlapN2(HP9combo_peaks[cov_input>=5], h3k4me3_P9_peaks, x1="HP9combo", y1="h3k4me3_P9")
fraction_overlapN2(HP9combo_peaks[cov_input>=5][is.na(tss)], h3k4me3_P9_peaks, x1="HP9combo", y1="h3k4me3_P9")
```

Better than C term

higher enriched N peaks, overlap more with transfected H3K4, 
  but less with untransfected H3K4
  

```{r}
fraction_overlapN2(HP9N_peaks[cov_input>=5], h3k4me3_UT_peaks, x1="HP9N", y1="h3k4me3_UT")
fraction_overlapN2(HP9N_peaks[cov_input>=5][is.na(tss)], h3k4me3_UT_peaks, x1="HP9N -TSS", y1="h3k4me3_UT") + ylim(0,0.4)
fraction_overlapN2(HP9N_peaks[cov_input>=5], h3k4me3_P9_peaks, x1="HP9N", y1="h3k4me3_P9")
fraction_overlapN2(HP9N_peaks[cov_input>=5][is.na(tss)], h3k4me3_P9_peaks, x1="HP9N -TSS", y1="h3k4me3_P9")
```



# HP9 vs H3K36me3

```{r}
fraction_overlapN2(HP9N_peaks[cov_input>=5], h3k36me3_P9_peaks, x1="HP9N", y1="h3k36me3_P9")
fraction_overlapN2(HP9N_peaks[cov_input>=5], h3k36me3_UT_peaks, x1="HP9N", y1="h3k36me3_UT")
```


# HP9 vs Zcw

Most (75%) P9combo peaks in Zcw preexisting (-ve cor???) & Zcw+P9

Nice variation from 0.1 to 0.6 with Chip vs Chip comparison

Less linear Variation with Chip vs Chip for N term, only goes to 25%, similar number of peaks.

```{r}
fraction_overlapN2(HP9combo_peaks[cov_input>=5], zcw_peaks, x1="hPrdm9 Cterm combined", y1="Zcw UT", shift = 1e9)
fraction_overlapN2(HP9combo_peaks[cov_input>=5], zcw_wP9_peaks_cin, x1="hPrdm9 Cterm combined", y1="Zcw wP9", shift = 1e9)
fraction_overlapN2(HP9combo_peaks[cov_input>=5], zcw_wP9_vs_chip_Z, x1="hPrdm9 Cterm combined", y1="Zcw CVC", shift = 1e9)

```


```{r}
fraction_overlapN2(HP9N_peaks[cov_input>=5], zcw_peaks, x1="hPrdm9 Nterm", y1="Zcw UT")
fraction_overlapN2(HP9N_peaks[cov_input>=5], zcw_wP9_peaks_cin, x1="hPrdm9 Nterm", y1="Zcw wP9")
fraction_overlapN2(HP9N_peaks[cov_input>=5], zcw_wP9_vs_chip_Z, x1="hPrdm9 Nterm", y1="Zcw CVC")

```



# DMC1 vs HP9

Best HP9combo peaks, overlap DMC 15% of the time

~40% of DMC1 overlap HP9combo (30%-60%)


```{r}
dmc1 <- trim_peaks(dmc1, 800)

plot(dmc1$enrichment, dmc1$width, log="x", pch=".", ylim=c(800,3e3))

plot(HP9N_peaks[cov_input>=5]$enrichment, 
     HP9N_peaks[cov_input>=5][,.(CI_stop-CI_start)]$V1, log="x", pch=".")
```


```{r}
fraction_overlapN2(HP9combo_peaks[cov_input>=5],  dmc1, x1="hP9 Cterm combo", y1="Dcm1")
fraction_overlapN2(dmc1, HP9combo_peaks, x1="Dcm1", y1="hP9 Cterm combo")
```

Much cleaner than C term!

Best HP9N peaks overlap DMC1 50% of the time (vs 15%)

~60% of DMC1 overlap HP9N (vs 40%)

```{r}
fraction_overlapN2(HP9N_peaks[cov_input>=5], dmc1, x1="HP9 N term", y1="Dcm1")
fraction_overlapN2(HP9N_peaks[cov_input>=5], dmc1, x1="HP9 N term", y1="Dcm1", n=75)
fraction_overlapN2(dmc1, HP9N_peaks, x1="Dcm1", y1="HP9 N term")

```

```{r}
fraction_overlapN2(HP9N_peaks[cov_input>=5  & FC_h3k4me3_UT_enrichment>1], dmc1, x1="HP9 N term", y1="Dcm1")

HP9N_peaks$h3k4me3_UT_enrichment <- foverlaps(HP9N_peaks, h3k4me3_UT_peaks, mult="first")$enrichment

fraction_overlapN2(HP9N_peaks[cov_input>=5  & !is.na(h3k4me3_UT_enrichment)], dmc1, x1="HP9 N term", y1="Dcm1")
fraction_overlapN2(HP9N_peaks[cov_input>=5  & is.na(h3k4me3_UT_enrichment)], dmc1, x1="HP9 N term", y1="Dcm1")
```


```{r}
fraction_overlapN2(HP9combo_peaks[cov_input>=5  & FC_h3k4me3_UT_enrichment>1], dmc1, x1="HP9 N term", y1="Dcm1")
fraction_overlapN2(HP9combo_peaks[cov_input>=5  & FC_h3k4me3_UT_enrichment<1], dmc1, x1="HP9 N term", y1="Dcm1")

HP9combo_peaks$h3k4me3_UT_enrichment <- foverlaps(HP9combo_peaks, h3k4me3_UT_peaks, mult="first")$enrichment

fraction_overlapN2(HP9combo_peaks[cov_input>=5  & is.na(h3k4me3_UT_enrichment)], dmc1, x1="HP9 N term", y1="Dcm1")
```


# Zcw vs DMC1

V. few (1%) preexising Zcw become DMC1, but over 50% DMC1 have pre-existing Zcw peaks

V. few (1%, though up to 5%) wP9cotransfected Zcw become DMC1, but 50%+ DMC1 are Zcw_wP9 peaks

Up to 20% of Zcw_CVC peaks become DMC1, Similar fraction of DMC1 have Zcw_CVC peaks


```{r}

fraction_overlapN2(zcw_peaks[cov_input>=5], dmc1, x1="Zcw UT", y1="Dcm1", shift = 1e8)
fraction_overlapN2(dmc1, zcw_peaks, x1="Dcm1", y1="Zcw UT", shift = 1e8)

fraction_overlapN2(zcw_wP9_peaks_cin[cov_input>=5],  dmc1, x1="Zcw wP9", y1="Dcm1", shift = 1e8)
fraction_overlapN2(dmc1, zcw_wP9_peaks_cin, x1="Dcm1", y1="Zcw wP9", shift = 1e8)

fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=5], dmc1, x1="Zcw CVC", y1="Dcm1", shift = 1e8)
fraction_overlapN2(dmc1, zcw_wP9_vs_chip_Z, x1="Dcm1", y1="Zcw CVC", shift = 1e8)

```

Zcw UT peaks cover 32% of the genome (accounting for width of DMC1 peaks @ 500bp)

```{r}
sum(zcw_peaks[,CI_stop_ext-CI_start_ext])/3e9
sum(zcw_peaks[,(CI_stop_ext+250)-(CI_start_ext-250)])/3e9
sum(zcw_peaks[,(CI_stop_ext+400)-(CI_start_ext-400)])/3e9
sum(dmc1[,CI_stop_ext-CI_start_ext])/3e9

hist(dmc1[,CI_stop-CI_start], breaks=100)
mean(dmc1[,CI_stop_ext-CI_start_ext])
```


# Zcw vs HP9

~20% pre-existing Zcw turn into HP9combo peaks, up to 30% for strongest pre-existing peaks

Overall 15% of Zcw_wP9 peaks are H9combo peaks, but up to 80% for strongest Zcw_wP9!

Almost 80% of Zcw_CVC peaks are H9combo peaks


```{r}
fraction_overlapN2(zcw_peaks[cov_input>8], HP9combo_peaks, x1="Zcw UT", y1="hP9 Cterm combo", shift = 1e8)
fraction_overlapN2(zcw_wP9_peaks_cin[cov_input>8], HP9combo_peaks, x1="Zcw wP9", y1="hP9 Cterm combo", n=50, shift = 1e8)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=8], HP9combo_peaks, x1="Zcw CVC", y1="hP9 Cterm combo", shift = 1e7) + ylim(0,1)
```

##  With larger extensions

```{r}
HP9combo_peaks <- prejoin(HP9combo_peaks, ext=300)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=8], HP9combo_peaks, x1="Zcw CVC", y1="hP9 Cterm combo", shift = 1e7) + ylim(0,1)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=10], HP9combo_peaks, x1="Zcw CVC", y1="hP9 Cterm combo", shift = 1e7) + ylim(0.8,1)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=10], HP9combo_peaks, x1="Zcw CVC", y1="hP9 Cterm combo", shift = 1e7) + ylim(0,1)
HP9combo_peaks <- prejoin(HP9combo_peaks, ext=200)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=9], HP9combo_peaks, x1="Zcw CVC", y1="hP9 Cterm combo", shift = 1e7) + ylim(0,1)
HP9combo_peaks <- prejoin(HP9combo_peaks, ext=100)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=9], HP9combo_peaks, x1="Zcw CVC", y1="hP9 Cterm combo", shift = 1e7) + ylim(0,1)
```

## Export Zcw_CVC peaks that don't overlap hP9combo peaks

```{r}

zcw_wP9_vs_chip_Z$HP9combo_enrichment <- foverlaps(zcw_wP9_vs_chip_Z, HP9combo_peaks, mult="first")$enrichment
mean(is.na(zcw_wP9_vs_chip_Z$HP9combo_enrichment))

fwrite(zcw_wP9_vs_chip_Z[is.na(HP9combo_enrichment)][order(enrichment)][,.(chr, center_start, center_stop, enrichment)], "../data/Zcw_CVC_NonOverlapping_HP9combo_peaks.bed", col.names = FALSE, sep = "\t")
fwrite(zcw_wP9_vs_chip_Z[!is.na(HP9combo_enrichment)][order(enrichment)][,.(chr, center_start, center_stop, enrichment)], "../data/Zcw_CVC_Overlapping_HP9combo_peaks.bed", col.names = FALSE, sep = "\t")
```

## Nterm

In all cases worse than combo peaks

~12% pre-existing Zcw turn into HP9N peaks (less than combo), up to 30% for strongest pre-existing peaks

Overall 10% of Zcw_wP9 peaks are H9N peaks, but up to 60% for strongest Zcw_wP9 (less than combo)

Almost 65% of Zcw_CVC peaks are H9N peaks (less than combo), direction *opposite* ??


```{r}
fraction_overlapN2(zcw_peaks[cov_input>8], HP9N_peaks, x1="Zcw UT", y1="HP9 N term", shift=1e8)
fraction_overlapN2(zcw_wP9_peaks_cin[cov_input>10], HP9N_peaks, x1="Zcw wP9", y1="HP9 N term", n=50, shift=1e8)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=9], HP9N_peaks, x1="Zcw CVC", y1="HP9 N term", shift=1e7)
```

```{r}
HP9N_peaks <- prejoin(HP9N_peaks, ext=500)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=9], HP9N_peaks, x1="Zcw CVC", y1="HP9 N term", shift=1e7) + ylim(0,1)
HP9N_peaks <- prejoin(HP9N_peaks, ext=500)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=9], HP9N_peaks, x1="Zcw CVC", y1="HP9 N term", shift=1e7, corrected = T) + ylim(0,1)
HP9N_peaks <- prejoin(HP9N_peaks, ext=300)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=9], HP9N_peaks, x1="Zcw CVC", y1="HP9 N term", shift=1e7, corrected = T) + ylim(0,1)
HP9N_peaks <- prejoin(HP9N_peaks, ext=200)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=9], HP9N_peaks, x1="Zcw CVC", y1="HP9 N term", shift=1e7, corrected = T) + ylim(0,1)
HP9N_peaks <- prejoin(HP9N_peaks, ext=100)
fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>=9], HP9N_peaks, x1="Zcw CVC", y1="HP9 N term", shift=1e7, corrected = T) + ylim(0,1)
```


# Zcw vs Zcw

Most (80%) Pre-existing Zcw peaks, are also Zcw_wP9 peaks (<40% for the weakest)

Most (>90%) Zcw_wP9 peaks are pre-existing Zcw, trends less with higher Zcw_wP9 enrichment

Overall few (5%) Zcw_CVC peaks are Zcw_wP9, but up to 80% for highest enriched Zcw_CVC

Most (90%) Zcw_CVC peaks, are Zcw_wP9 peaks (almost logically nececcary?)

```{r}

fraction_overlapN2(zcw_peaks[cov_input>=5], zcw_wP9_peaks_cin, x1="Zcw UT", y1="Zcw wP9", shift=1e8)

fraction_overlapN2(zcw_wP9_peaks_cin[cov_input>=5], zcw_peaks, x1="Zcw_wP9_cin", y1="zcw_enrichment", shift=1e8)

fraction_overlapN2(zcw_wP9_peaks_cin[cov_input>=5], zcw_wP9_vs_chip_Z, x1="Zcw_wP9_cin", y1="zcw_CVC_enrichment", shift=1e8)
fraction_overlapN2(zcw_wP9_peaks_cin[cov_input>=5], zcw_wP9_vs_chip_Z, x1="Zcw_wP9_cin", y1="zcw_CVC_enrichment",n=200, shift=1e8)

fraction_overlapN2(zcw_wP9_vs_chip_Z[cov_input>5], zcw_wP9_peaks_cin, x1="Zcw_CVC", y1="zcw_wP9_enrichment", shift=1e8)
```



# Zcw vs H3K4

Pre-existing Zcw overlap P9_H3K4me3 5% to 25%

P9 Co-transfected Zcw overlap P9_H3K4me3 10% of the time, up to 50%

Pre-existing Zcw & P9 Co-transfected overlap UT_H3K4me3 < 5%/10% of the time in any case

```{r}
fraction_overlapN2(zcw_peaks[cov_input>=5], h3k4me3_P9_peaks, x1="Zcw UT", y1="h3k4me3_P9_peaks", shift=1e8)
fraction_overlapN2(zcw_wP9_peaks[cov_input>=5], h3k4me3_P9_peaks, x1="Zcw wP9", y1="h3k4me3_P9_peaks", shift=1e8)

fraction_overlapN2(zcw_peaks[cov_input>=5], h3k4me3_UT_peaks, x1="Zcw UT", y1="h3k4me3_UT_peaks", shift=1e8)
fraction_overlapN2(zcw_wP9_peaks[cov_input>=5], h3k4me3_UT_peaks, x1="Zcw wP9", y1="h3k4me3_UT_peaks", shift=1e8)
```

# Zcw vs H3K36

```{r}
fraction_overlapN2(zcw_peaks[cov_input>=5], h3k36me3_P9_peaks, x1="Zcw UT", y1="h3k36me3_P9_peaks", shift=1e8)
fraction_overlapN2(zcw_wP9_peaks[cov_input>=5], h3k36me3_P9_peaks, x1="Zcw wP9", y1="h3k36me3_P9_peaks", shift=1e8)

fraction_overlapN2(zcw_peaks[cov_input>=5], h3k36me3_UT_peaks, x1="Zcw UT", y1="h3k36me3_UT_peaks", shift=1e8)
fraction_overlapN2(zcw_wP9_peaks[cov_input>=5], h3k36me3_UT_peaks, x1="Zcw wP9", y1="h3k36me3_UT_peaks", shift=1e8)
```


# Orientation correlation

```{r}
HP9combo_peaks$zcw_CVC_center_start <- foverlaps(HP9combo_peaks, zcw_wP9_peaks_cin, mult="first")$center_start
HP9combo_peaks$h3k4me3_P9_center_start <- foverlaps(HP9combo_peaks, h3k4me3_P9_peaks, mult="first")$center_start
HP9combo_peaks$h3k36me3_P9_center_start <- foverlaps(HP9combo_peaks, h3k36me3_P9_peaks, mult="first")$center_start

HP9combo_peaks[,dist_to_zcwCVC := center_start - zcw_CVC_center_start]
HP9combo_peaks[,dist_to_h3k4me3 := center_start - h3k4me3_P9_center_start]
HP9combo_peaks[,dist_to_h3k36me3 := center_start - h3k36me3_P9_center_start]

plot(HP9combo_peaks$dist_to_zcwCVC, HP9combo_peaks$dist_to_h3k4me3, pch=".")
cor(HP9combo_peaks$dist_to_zcwCVC, HP9combo_peaks$dist_to_h3k4me3, use="complete.obs")
cor(HP9combo_peaks$dist_to_zcwCVC, HP9combo_peaks$dist_to_h3k36me3, use="complete.obs")

cor(HP9combo_peaks[!is.na(tss)]$dist_to_zcwCVC, HP9combo_peaks[!is.na(tss)]$dist_to_h3k4me3, use="complete.obs")
cor(HP9combo_peaks[is.na(tss)]$dist_to_zcwCVC, HP9combo_peaks[is.na(tss)]$dist_to_h3k4me3, use="complete.obs")
#cor(HP9combo_peaks[FC_h3k4me3_UT_enrichment<1]$dist_to_zcwCVC, HP9combo_peaks[FC_h3k4me3_UT_enrichment<1]$dist_to_h3k4me3, use="complete.obs")

ggplot(HP9combo_peaks, aes(dist_to_zcwCVC, dist_to_h3k4me3)) + geom_point(stroke=0, alpha=0.2, size=0.6) + xlim(-500,500) + ylim(-500,500)

ggplot(HP9combo_peaks[is.na(tss)], aes(dist_to_zcwCVC, dist_to_h3k4me3)) + geom_point(stroke=0, alpha=0.2, size=0.6) + xlim(-500,500) + ylim(-500,500)

#ggplot(HP9combo_peaks[FC_h3k4me3_UT_enrichment<1], aes(dist_to_zcwCVC, dist_to_h3k4me3)) + geom_point(stroke=0, alpha=0.2, size=0.6) + xlim(-500,500) + ylim(-500,500)

ggplot(HP9combo_peaks, aes(dist_to_zcwCVC, dist_to_h3k4me3)) + geom_point(stroke=0, alpha=0.2, size=0.6) + xlim(-500,500) + facet_wrap(~!is.na(tss)) + ylim(-500,500)

ggplot(HP9combo_peaks, aes(dist_to_zcwCVC, dist_to_h3k36me3)) + geom_point(stroke=0, alpha=0.2, size=0.6) + xlim(-500,500) + ylim(-500,500)
```


```{r}
HP9combo_peaks$DMC1_enrichment <- foverlaps(HP9combo_peaks, dmc1, mult="first")$enrichment

HP9combo_peaks$isDMC1 <- !is.na(HP9combo_peaks$DMC1_enrichment)

cor(HP9combo_peaks[isDMC1==T]$dist_to_zcwCVC, HP9combo_peaks[isDMC1==T]$dist_to_h3k4me3, use="complete.obs")
cor(HP9combo_peaks[isDMC1==F]$dist_to_zcwCVC, HP9combo_peaks[isDMC1==F]$dist_to_h3k4me3, use="complete.obs")
ggplot(HP9combo_peaks, aes(dist_to_zcwCVC, dist_to_h3k4me3)) + geom_point(stroke=0, alpha=0.2, size=1.5) + xlim(-500,500) + ylim(-500,500) + facet_wrap(~isDMC1)


fraction_overlap_mean <- function(dt,x,y,n=30, returndf=F){
  setorder(dt, dist_to_zcwCVC)
  
  dt$enrichment_bin <- cut(1:nrow(dt), n)
  
  tmp <- dt[,.(mean=mean(get(x)),
               num=.N,
               dist=mean(get(y)),
               sem=sd((get(y)))/sqrt(length(get(y))),
            sem_mean=sd((get(x)))/sqrt(length(get(x)))),
               by=enrichment_bin]
  
  if(returndf){
    return(tmp)
  }else{
  return(
  ggplot(tmp, aes(mean, dist)) +
    geom_pointrange(aes(ymax = dist + 2*sem, ymin = dist - 2*sem)) +
    geom_errorbarh(aes(xmax= mean + 2*sem_mean, xmin = mean - 2*sem_mean), colour='red') +
    ylab(paste("Mean",y)) + xlab("Binned")
  )
  }
}

fraction_overlap_mean(HP9combo_peaks[isDMC1==T], "dist_to_zcwCVC", "dist_to_h3k4me3")
fraction_overlap_mean(HP9combo_peaks[isDMC1==F], "dist_to_zcwCVC", "dist_to_h3k4me3")

fraction_overlap_mean(HP9combo_peaks[!is.na(dist_to_zcwCVC) & !is.na(dist_to_h3k4me3)], "dist_to_zcwCVC", "dist_to_h3k4me3")

# horizontal bars are small
# fraction_overlap_mean(HP9combo_peaks[!is.na(dist_to_zcwCVC) & !is.na(dist_to_h3k4me3)], "dist_to_zcwCVC", "dist_to_h3k4me3", n=30) + xlim(-3,-1)
# fraction_overlap_mean(HP9combo_peaks[!is.na(dist_to_zcwCVC) & !is.na(dist_to_h3k4me3)], "dist_to_zcwCVC", "dist_to_h3k4me3", n=30, returndf = T)
```



# Sandbox

```{r}
ctcf <- fread("~/Downloads/hglft_genome_1dd9_39f220.bed")
setnames(ctcf,c("chr",'CI_start','CI_stop',"enrichment","B","C"))
ctcf <- prejoin(ctcf)
ctcf

zcw_wP9_peaks_JOIN_ctcf <- foverlaps(zcw_wP9_peaks, ctcf, by.x = c("chr","CI_start_ext","CI_stop_ext"))
fraction_overlap(zcw_wP9_peaks_JOIN_ctcf, x="Zcwpw1", y="CTCF", n = 150)

zcw_wP9_peaks_JOIN_ctcf <- foverlaps(zcw_wP9_peaks, ctcf, by.x = c("chr","CI_start_ext","CI_stop_ext"))
fraction_overlap(zcw_wP9_peaks_JOIN_ctcf, x="Zcwpw1", y="H3K36me3_P9", n = 400)
```

```{r}
readBED <- function(bedfile="~/Downloads/hglft_genome_6542_3a6b10.bed"){
  ctcf <- fread(bedfile)
  setnames(ctcf,c("chr",'CI_start','CI_stop',"enrichment","B","C"))
  ctcf <- prejoin(ctcf)
  ctcf <- foverlaps(zcw_wP9_peaks, ctcf, by.x = c("chr","CI_start_ext","CI_stop_ext"))
  return(ctcf)
}


h4k20me1 <- readBED("~/Downloads/h4k20me1_bernstein.bed")
h3k27ac <- readBED("~/Downloads/h3k27ac_bernstein.bed")
h3k27me3 <- readBED("~/Downloads/h3k27me3_bernstein.bed")

ctcf <- readBED("~/Downloads/Bernstein_NHEK_CTCF_hg38.bed")
pol2b <- readBED("~/Downloads/Bernstein_NHEK_Pol2b_hg38.bed")
h3k4me3_bern <- readBED("~/Downloads/Bernstein_NHEK_H3K4me3_hg38.bed")
h3k36me3_bern <- readBED("~/Downloads/Bernstein_NHEK_H3K36me3_hg38.bed")
h3k9me3_broad <- readBED("~/Downloads/Broad_ChipSeq_NHEK_H3K9me3_hg38.bed")

fraction_overlap(h4k20me1, x="Zcwpw1", y="h4k20me1", n = 150)
fraction_overlap(h3k27ac, x="Zcwpw1", y="h3k27ac", n = 150)
fraction_overlap(h3k27me3, x="Zcwpw1", y="h3k27me3", n = 150)

fraction_overlap(ctcf, x="Zcwpw1", y="ctcf", n = 150)
fraction_overlap(pol2b, x="Zcwpw1", y="pol2b", n = 150)
fraction_overlap(h3k4me3_bern, x="Zcwpw1", y="h3k4me3_bern", n = 150)
fraction_overlap(h3k36me3_bern, x="Zcwpw1", y="h3k36me3_bern", n = 150)
fraction_overlap(h3k9me3_broad, x="Zcwpw1", y="h3k9me3_broad", n = 150)


```


```{r}
nearest_peak <- function(x, A=zcw_wP9_peaks, B=h3k4me3_P9_peaks) min(abs(A[chr=="chr16" & center_start<3.1e+07][x]$center_start - B[chr=="chr16"]$center_start))

set.seed(42)
uniform_dist <- sample(1:max(zcw_wP9_peaks[chr=="chr16" & center_start<3.1e+07]$center_start), 10000)

no <- sort(uniform_dist)[diff(sort(uniform_dist))<250]
uniform_dist <- uniform_dist[!uniform_dist %in% no][1:5000]

nearest_peak_null <- function(x, B=h3k4me3_P9_peaks) min(abs(uniform_dist[x] - B[chr=="chr16"]$center_start))

plot_cum_nearest <- function(nearest, nearest_null, to=10e4, by=100){
  cum_nearest <- function(x, vec=nearest) sum(vec<x)/length(vec)
  dt <- data.table(dist=seq(1,to,by),
             overlap=sapply(seq(1,to,by), cum_nearest),
             overlap_null=sapply(seq(1,to,by), function(x) cum_nearest(x, vec=nearest_null)))
  ggplot(dt, aes(dist, overlap)) +
    geom_line() +
      xlab("Distance to peak") + ylab("Cumulative fraction of peaks") + 
    geom_line(aes(y=overlap_null), colour='red')
}

set.seed(42)
indexes <- sample(1:nrow(zcw_wP9_peaks[chr=="chr16" & center_start<3.1e+07]), 5000)

nearest <- sapply(indexes, nearest_peak)
nearest_null <- sapply(seq_along(uniform_dist), nearest_peak_null)


nearest_k36 <- sapply(indexes, function(x) nearest_peak(x, B=h3k36me3_P9_peaks))
nearest_null_k36 <- sapply(seq_along(uniform_dist), function(x) nearest_peak_null(x, B=h3k36me3_P9_peaks))

nearest_hP9 <- sapply(indexes, function(x) nearest_peak(x, B=HP9_peaks))
nearest_null_hP9 <- sapply(seq_along(uniform_dist), function(x) nearest_peak_null(x, B=HP9_peaks))



plot_cum_nearest(nearest, nearest_null, to=1e4)
plot_cum_nearest(nearest_k36, nearest_null_k36, to=1e4)
plot_cum_nearest(nearest_hP9, nearest_null_hP9, to=1e4)

# debugging plot
# plot(sort(uniform_dist)[100:140], rep(1,41), col='blue')
# points(zcw_wP9_peaks[chr=="chr16"]$center_start[indexes], rep(0.98,length(indexes)))
# points(h3k4me3_P9_peaks[chr=="chr16"]$center_start, rep(0.96,nrow(h3k4me3_P9_peaks[chr=="chr16"])), col='red')


hist(nearest, breaks=10000, xlim=c(0,10000), ylim=c(0,100))
hist(nearest_null, breaks=10000, xlim=c(0,10000), ylim=c(0,100))

hist(nearest_k36, breaks=10000, xlim=c(0,10000), ylim=c(0,100))
hist(nearest_null_k36, breaks=10000, xlim=c(0,10000), ylim=c(0,100))

```


```{r}

library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
library(clusterProfiler)

enriches <- enrichPeakOverlap(queryPeak = zcw_wP9_peaks_GR,
                  targetPeak    = list(h3k36me3_P9_peaks_GR),
                  TxDb          = txdb,
                  pAdjustMethod = "BH",
                  nShuffle      = 1000,
                  chainFile     = NULL,
                  verbose       = FALSE,
                  pool = FALSE)


zcw_wP9_peaks_GR <- makeGRangesFromDataFrame(zcw_wP9_peaks, ignore.strand = T, seqnames.field = "chr", start.field = "CI_start", end.field = "CI_stop", keep.extra.columns = T)
zcw_wP9_peaks_GR

h3k36me3_P9_peaks_GR <- makeGRangesFromDataFrame(h3k36me3_P9_peaks, ignore.strand = T, seqnames.field = "chr", start.field = "CI_start", end.field = "CI_stop", keep.extra.columns = T)

ctcf <- readPeakFile("~/Downloads/hglft_genome_1dd9_39f220.bed")

hg19 <- getGEOInfo(genome="hg19", simplify=TRUE)
head(hg19)

data.table(hg19)[grepl("HEK",hg19$title)]

wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM749nnn/GSM749668/suppl/GSM749668_hg19_wgEncodeUwTfbsHek293CtcfStdHotspotsRep1.broadPeak.gz
cut -f 1-6 GSM749668_hg19_wgEncodeUwTfbsHek293CtcfStdHotspotsRep1.broadPeak > GSM749668_hg19_wgEncodeUwTfbsHek293CtcfStdHotspotsRep1.bed

wget  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM749nnn/GSM749668/suppl/GSM749668_hg19_wgEncodeUwTfbsHek293CtcfStdPkRep1.narrowPeak.gz
cut -f 1-6 GSM749668_hg19_wgEncodeUwTfbsHek293CtcfStdPkRep1.narrowPeak > GSM749668_hg19_wgEncodeUwTfbsHek293CtcfStdPkRep1.bed

Broad_ChipSeq_NHEK_H3K79me2
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1003nnn/GSM1003527/suppl/GSM1003527_hg19_wgEncodeBroadHistoneNhekH3k79me2Pk.broadPeak.gz
cut -f 1-6 GSM1003527_hg19_wgEncodeBroadHistoneNhekH3k79me2Pk.broadPeak > GSM1003527_hg19_wgEncodeBroadHistoneNhekH3k79me2Pk.bed

Bernstein_NHEK_H3K27me3
curl ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM733nnn/GSM733701/suppl/GSM733701_hg19_wgEncodeBroadHistoneNhekH3k27me3StdPk.broadPeak.gz | zcat | cut -f 1-6 > GSM733701_hg19_wgEncodeBroadHistoneNhekH3k27me3StdPk.bed

#Bernstein_NHEK_H3K27ac
curl ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM733nnn/GSM733674/suppl/GSM733674_hg19_wgEncodeBroadHistoneNhekH3k27acStdPk.broadPeak.gz | zcat | cut -f 1-6 > GSM733674_hg19_wgEncodeBroadHistoneNhekH3k27acStdPk.bed

#Bernstein_NHEK_H4K20me1
curl ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM733nnn/GSM733728/suppl/GSM733728_hg19_wgEncodeBroadHistoneNhekH4k20me1StdPk.broadPeak.gz | zcat | cut -f 1-6 > GSM733728_hg19_wgEncodeBroadHistoneNhekH4k20me1StdPk.bed

#Bernstein_NHEK_CTCF
curl ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM733nnn/GSM733636/suppl/GSM733636_hg19_wgEncodeBroadHistoneNhekCtcfStdPk.broadPeak.gz | zcat | cut -f 1-6 > Bernstein_NHEK_CTCF_hg19.bed

#Bernstein_NHEK_Pol2(b)
curl ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM733nnn/GSM733671/suppl/GSM733671_hg19_wgEncodeBroadHistoneNhekPol2bStdPk.broadPeak.gz  | zcat | cut -f 1-6 > Bernstein_NHEK_Pol2b_hg19.bed

#Bernstein_NHEK_H3K4me3
curl ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM733nnn/GSM733720/suppl/GSM733720_hg19_wgEncodeBroadHistoneNhekH3k4me3StdPk.broadPeak.gz | zcat | cut -f 1-6 > Bernstein_NHEK_H3K4me3_hg19.bed

#Bernstein_NHEK_H3K36me3
curl ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM733nnn/GSM733726/suppl/GSM733726_hg19_wgEncodeBroadHistoneNhekH3k36me3StdPk.broadPeak.gz | zcat | cut -f 1-6 > Bernstein_NHEK_H3K36me3_hg19.bed

#Broad_ChipSeq_NHEK_H3K9me3
curl ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1003nnn/GSM1003528/suppl/GSM1003528_hg19_wgEncodeBroadHistoneNhekH3k09me3Pk.broadPeak.gz | zcat | cut -f 1-6 > Broad_ChipSeq_NHEK_H3K9me3_hg19.bed

#Bernstein_NHEK_H3K9ac
curl ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM733nnn/GSM733665/suppl/GSM733665_hg19_wgEncodeBroadHistoneNhekH3k9acStdPk.broadPeak.gz | zcat | cut -f 1-6 > Bernstein_NHEK_H3K9ac_hg19.bed


enriches <- enrichPeakOverlap(queryPeak = zcw_wP9_peaks_GR,
                  targetPeak    = "~/Downloads/hglft_genome_1dd9_39f220.bed",
                  TxDb          = txdb,
                  pAdjustMethod = "BH",
                  nShuffle      = 1000,
                  chainFile     = NULL,
                  verbose       = FALSE,
                  pool = FALSE)

enriches

hg38 <- getGEOInfo(genome="hg38", simplify=TRUE)
head(hg38)

#GSM749668 hek ctcf

downloadGEObedFiles(genome="hg38", destDir="~/chipseeker")

GEO_beds <- list.files("~/chipseeker", full.names = T, pattern = "narrowPeak")

# Takes too long to run!
# enriches <- enrichPeakOverlap(queryPeak = zcw_wP9_peaks_GR,
#                   targetPeak    = GEO_beds,
#                   TxDb          = txdb,
#                   pAdjustMethod = "BH",
#                   nShuffle      = 1000,
#                   chainFile     = NULL,
#                   verbose       = FALSE,
#                   pool=FALSE)

enriches
```



```{r}
 hist(Zcw_CVC_AT_hP9$pvalue[Zcw_CVC_AT_hP9$cov_input>=30 & HP9_peaks$enrichment>=10 & HP9_peaks$cov_input>=5],n=100)
aa=Zcw_CVC_AT_hP9$cov_r1+Zcw_CVC_AT_hP9$cov_r2
bb=Zcw_CVC_AT_hP9$cov_input
cc=Zcw_CVC_AT_hP9$enrichment
dd=cc*bb
summary(lm(dd~aa+bb))

enrich*bg=(0.5*(cov1+cov2)-0.3552*bg)
Error: object 'cov1' not found
> summary(lm(dd~0+aa+bb))

enrich*bg=(0.5*(cov1+cov2)-0.3525*bg)
Error: object 'cov1' not found
> predcc=(0.5*aa-0.325*bb)/bb
> plot(predcc,cc)
> summary(lm(dd[dd>0]~0+aa[dd>0]+bb[dd>0]))

predcc=(0.5*aa-0.3555*bb)/bb
> summary(lm(dd[dd>0]~aa[dd>0]+bb[dd>0]))

plot(predcc,cc)
abline(0,1)
cc2=HP9_peaks$enrichment
bb2=HP9_peaks$cov_input
plot(cc2[bb>=5 & bb2>=5],aa[bb>=5 & bb2>=5],pch=".",ylim=c(0.1,200),log="xy")
plot(cc2[bb>=5 & bb2>=5],cc[bb>=5 & bb2>=5],pch=".",ylim=c(0.01,5),log="xy")
mean(cc[bb>=5 & bb2>=5 & cc2>5])
ee=aa-0.355*bb
mean(aa[bb>=5 & bb2>=5 & cc2>2])
hist(Zcw_CVC_AT_hP9$pvalue[Zcw_CVC_AT_hP9$cov_input>=5 & HP9_peaks$enrichment>=5 & HP9_peaks$cov_input>=5],n=100)
mean(Zcw_CVC_AT_hP9$pvalue[HP9_peaks$enrichment>=10 & HP9_peaks$cov_input>=5]<=1e-6)



```

